{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Genetic determinants of incident CH"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "setwd(\"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/rg2022/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes 'data.table' and 'data.frame':\t57 obs. of  51 variables:\n",
      " $ varID            : chr  \"1:226374920:A:C\" \"10:103908421:C:CTT\" \"11:108043157:T:C\" \"11:108435509:C:A\" ...\n",
      " $ chr_hg19         : int  1 10 11 11 11 12 14 17 17 17 ...\n",
      " $ POS_hg19         : int  226562621 105668179 107913884 108306236 108315606 26589770 101178715 55463857 7571752 7731041 ...\n",
      " $ locus            : chr  \"1-1\" \"10-1\" \"11-1\" \"11-1\" ...\n",
      " $ Chr              : int  1 10 11 11 11 12 14 17 17 17 ...\n",
      " $ Pos              : int  226374920 103908421 108043157 108435509 108444879 26436837 100712378 57386496 7668434 7827723 ...\n",
      " $ Ref              : chr  \"A\" \"C\" \"T\" \"C\" ...\n",
      " $ Alt              : chr  \"C\" \"CTT\" \"C\" \"A\" ...\n",
      " $ Trait            : chr  \"CHIP_inclusive\" \"CHIP_inclusive\" \"CHIP_inclusive\" \"CHIP_inclusive\" ...\n",
      " $ Model            : chr  \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" ...\n",
      " $ Effect_UKB       : num  1.115 0.914 0.921 1.12 1.228 ...\n",
      " $ LCI_Effect_UKB   : num  1.086 0.887 0.896 1.099 1.162 ...\n",
      " $ UCI_Effect_UKB   : num  1.144 0.943 0.948 1.141 1.298 ...\n",
      " $ Pval_UKB         : num  2.41e-16 8.41e-09 9.92e-09 1.43e-32 3.97e-13 ...\n",
      " $ AAF_UKB          : num  0.8416 0.9012 0.1282 0.4077 0.0255 ...\n",
      " $ Num_Cases_UKB    : int  25657 25657 25657 25418 25657 25657 25657 25657 25657 25657 ...\n",
      " $ Cases_Ref_UKB    : int  566 298 19807 8204 24135 21744 19385 23544 24959 12301 ...\n",
      " $ Cases_Het_UKB    : int  6173 4697 5474 12517 1497 3776 5853 2067 692 10880 ...\n",
      " $ Cases_Alt_UKB    : int  18918 20662 376 4697 25 137 419 46 6 2476 ...\n",
      " $ Num_Controls_UKB : int  342869 342869 342869 339848 342869 342869 342869 342869 342869 342869 ...\n",
      " $ Controls_Ref_UKB : int  8265 3155 259616 119601 325801 297827 254620 309598 335330 156840 ...\n",
      " $ Controls_Het_UKB : int  89846 59068 77600 164536 16822 43334 81656 32424 7489 149972 ...\n",
      " $ Controls_Alt_UKB : int  244758 280646 5653 55711 246 1708 6593 847 50 36057 ...\n",
      " $ Model_GHS        : chr  \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" ...\n",
      " $ Effect_GHS       : num  1.077 0.944 0.988 1.04 1.044 ...\n",
      " $ LCI_Effect_GHS   : num  1.037 0.902 0.946 1.011 0.955 ...\n",
      " $ UCI_Effect_GHS   : num  1.119 0.989 1.031 1.07 1.142 ...\n",
      " $ Pval_GHS         : num  0.000132 0.014162 0.565679 0.006192 0.343956 ...\n",
      " $ AAF_GHS          : num  0.8314 0.8921 0.1217 0.4289 0.0249 ...\n",
      " $ Num_Cases_GHS    : int  11821 11821 11821 11821 11821 11821 11821 11821 11821 11821 ...\n",
      " $ Cases_Ref_GHS    : int  312 133 9119 3730 11206 9976 8784 10820 11510 5491 ...\n",
      " $ Cases_Het_GHS    : int  3097 2276 2512 5742 611 1764 2811 988 311 5045 ...\n",
      " $ Cases_Alt_GHS    : int  8412 9412 190 2349 4 81 226 13 0 1285 ...\n",
      " $ Num_Controls_GHS : int  135106 135106 135106 135106 135106 135106 135106 135106 135106 135106 ...\n",
      " $ Controls_Ref_GHS : int  3667 1410 103948 43560 128484 116191 100345 122598 131863 60973 ...\n",
      " $ Controls_Het_GHS : int  37363 24809 29023 66395 6530 18193 32243 12213 3220 59616 ...\n",
      " $ Controls_Alt_GHS : int  94076 108887 2135 25151 92 722 2518 295 23 14517 ...\n",
      " $ nearestGene      : chr  \"PARP1\" \"OBFC1\" \"CUL5\" \"C11orf65\" ...\n",
      " $ dist_nearestGene : int  0 0 0 0 0 0 14691 0 0 0 ...\n",
      " $ variantEffect    : chr  \"intronic\" \"intronic\" \"intronic\" \"intronic\" ...\n",
      " $ CADD             : num  1.661 NA 2.936 0.995 1.713 ...\n",
      " $ maxCADD_inLD     : num  27.9 NA 13.8 11.5 28.3 ...\n",
      " $ Lof_Missense_inLD: chr  \"rs1136410:PARP1:missense:0.99\" \"rs10786775:OBFC1:missense:0.98,rs2487999:OBFC1:missense:0.98\" NA NA ...\n",
      " $ Same_Direction   : int  1 1 1 1 1 1 0 1 1 1 ...\n",
      " $ Effect_adj       : num  1.114 0.955 0.962 1.12 1.221 ...\n",
      " $ LCI_Effect_adj   : num  1.086 0.926 0.936 1.099 1.155 ...\n",
      " $ UCI_Effect_adj   : num  1.143 0.985 0.989 1.141 1.291 ...\n",
      " $ CHROM            : chr  \"chr1\" \"chr10\" \"chr11\" \"chr11\" ...\n",
      " $ varID_hg19       : chr  \"1:226562621:A:C\" \"10:105668179:C:CTT\" \"11:107913884:T:C\" \"11:108306236:C:A\" ...\n",
      " $ Start_500kb      : int  226062621 105168179 107413884 107806236 107815606 26089770 100678715 54963857 7071752 7231041 ...\n",
      " $ End_500kb        : int  227062621 106168179 108413884 108806236 108815606 27089770 101678715 55963857 8071752 8231041 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n",
      "Classes 'data.table' and 'data.frame':\t53 obs. of  48 variables:\n",
      " $ varID            : chr  \"1:202860408:C:G\" \"1:226405440:A:C\" \"10:103911351:T:G\" \"11:108217120:G:T\" ...\n",
      " $ chr_hg19         : int  1 1 10 11 11 12 13 14 17 17 ...\n",
      " $ POS_hg19         : int  202829536 226593141 105671109 108087847 108315606 26586829 28604007 96180695 55463341 56669049 ...\n",
      " $ locus            : chr  \"1-1\" \"1-2\" \"10-1\" \"11-1\" ...\n",
      " $ chr              : int  1 1 10 11 11 12 13 14 17 17 ...\n",
      " $ pos              : int  202860408 226405440 103911351 108217120 108444879 26433896 28029870 95714358 57385980 58591688 ...\n",
      " $ ref              : chr  \"C\" \"A\" \"T\" \"G\" ...\n",
      " $ alt              : chr  \"G\" \"C\" \"G\" \"T\" ...\n",
      " $ trait            : chr  \"DNMT3A\" \"DNMT3A\" \"DNMT3A\" \"DNMT3A\" ...\n",
      " $ model            : chr  \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" ...\n",
      " $ effect_UKB       : num  1.07 0.87 0.89 1.14 1.23 ...\n",
      " $ lci_effect_UKB   : num  1.045 0.843 0.857 1.111 1.146 ...\n",
      " $ uci_effect_UKB   : num  1.094 0.899 0.924 1.163 1.313 ...\n",
      " $ pval_UKB         : num  1.59e-08 2.70e-17 8.88e-10 4.76e-28 4.06e-09 ...\n",
      " $ aaf_UKB          : num  0.3987 0.1532 0.9025 0.4073 0.0254 ...\n",
      " $ num_cases_UKB    : int  16219 16219 16219 16219 16219 16175 16219 16219 16219 16219 ...\n",
      " $ cases_ref_UKB    : int  5547 12086 193 5112 15251 13611 15675 9601 14896 10645 ...\n",
      " $ cases_het_UKB    : int  7853 3807 3035 8044 955 2468 539 5720 1297 5025 ...\n",
      " $ cases_alt_UKB    : int  2819 326 12991 3063 13 96 5 898 26 549 ...\n",
      " $ num_controls_UKB : int  342869 342869 342869 342869 342869 341948 342869 342869 342869 342869 ...\n",
      " $ controls_ref_UKB : int  123139 245386 3140 120557 325801 296923 333825 215099 309608 232910 ...\n",
      " $ controls_het_UKB : int  164814 89351 58930 165402 16822 43340 8981 112789 32415 99417 ...\n",
      " $ controls_alt_UKB : int  54916 8132 280799 56910 246 1685 63 14981 846 10542 ...\n",
      " $ model_GHS        : chr  \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" ...\n",
      " $ effect_GHS       : num  1.022 0.898 0.913 1.093 1.033 ...\n",
      " $ lci_effect_GHS   : num  0.983 0.852 0.859 1.051 0.913 ...\n",
      " $ uci_effect_GHS   : num  1.062 0.947 0.971 1.135 1.169 ...\n",
      " $ pval_GHS         : num  2.81e-01 6.44e-05 3.78e-03 7.00e-06 6.06e-01 ...\n",
      " $ aaf_GHS          : num  0.402 0.1624 0.8925 0.4282 0.0249 ...\n",
      " $ num_cases_GHS    : int  5991 5991 5991 5991 5991 5991 5991 5991 5991 5991 ...\n",
      " $ cases_ref_GHS    : int  2071 4301 81 1824 5691 5023 5766 3418 5491 3849 ...\n",
      " $ cases_het_GHS    : int  2928 1568 1170 2913 298 930 220 2194 492 1921 ...\n",
      " $ cases_alt_GHS    : int  992 122 4740 1254 2 38 5 379 8 221 ...\n",
      " $ num_controls_GHS : int  135106 135106 135106 135106 135106 135106 135106 135106 135106 135106 ...\n",
      " $ controls_ref_GHS : int  47819 94539 1433 43679 128484 116198 130955 84527 122599 89158 ...\n",
      " $ controls_het_GHS : int  65097 36986 24903 66371 6530 18182 4123 44690 12213 41236 ...\n",
      " $ controls_alt_GHS : int  22190 3581 108770 25056 92 726 28 5889 294 4712 ...\n",
      " $ nearestGene      : chr  \"RABIF\" \"PARP1\" \"OBFC1\" \"NPAT\" ...\n",
      " $ dist_nearestGene : int  20570 0 0 0 0 0 0 293 0 0 ...\n",
      " $ variantEffect    : chr  NA \"intronic\" \"intronic\" \"intronic\" ...\n",
      " $ CADD             : num  0.063 2.239 4.846 0.324 1.713 ...\n",
      " $ maxCADD_inLD     : num  9.77 27.9 10.33 11.47 28.3 ...\n",
      " $ Lof_Missense_inLD: chr  NA \"rs1136410:PARP1:missense:0.99\" \"rs10786775:OBFC1:missense:1.00,rs2487999:OBFC1:missense:1.00\" NA ...\n",
      " $ Same_Direction   : int  1 1 1 1 1 1 1 1 1 1 ...\n",
      " $ CHROM            : chr  \"chr1\" \"chr1\" \"chr10\" \"chr11\" ...\n",
      " $ varID_hg19       : chr  \"1:202829536:C:G\" \"1:226593141:A:C\" \"10:105671109:T:G\" \"11:108087847:G:T\" ...\n",
      " $ Start_500kb      : int  202329536 226093141 105171109 107587847 107815606 26086829 28104007 95680695 54963341 56169049 ...\n",
      " $ End_500kb        : int  203329536 227093141 106171109 108587847 108815606 27086829 29104007 96680695 55963341 57169049 ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n",
      "Classes 'data.table' and 'data.frame':\t8 obs. of  47 variables:\n",
      " $ varID            : chr  \"11:108306370:T:A\" \"14:95687472:CTAA:C\" \"14:95714348:T:G\" \"17:7668434:T:G\" ...\n",
      " $ chr_hg19         : int  11 14 14 17 3 3 5 5\n",
      " $ POS_hg19         : int  108177097 96153809 96180685 7571752 128182074 24343330 1285974 1317481\n",
      " $ chr              : int  11 14 14 17 3 3 5 5\n",
      " $ pos              : int  108306370 95687472 95714348 7668434 128463231 24301839 1285859 1317366\n",
      " $ ref              : chr  \"T\" \"CTAA\" \"T\" \"T\" ...\n",
      " $ alt              : chr  \"A\" \"C\" \"G\" \"G\" ...\n",
      " $ trait            : chr  \"TET2\" \"TET2\" \"TET2\" \"TET2\" ...\n",
      " $ model            : chr  \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" ...\n",
      " $ effect_UKB       : num  1.164 0.759 0.762 1.738 0.854 ...\n",
      " $ lci_effect_UKB   : num  1.112 0.688 0.721 1.441 0.81 ...\n",
      " $ uci_effect_UKB   : num  1.218 0.837 0.805 2.095 0.9 ...\n",
      " $ pval_UKB         : num  5.72e-11 3.41e-08 4.54e-22 7.15e-09 4.11e-09 ...\n",
      " $ aaf_UKB          : num  0.4057 0.0653 0.2318 0.0111 0.2548 ...\n",
      " $ num_cases_UKB    : int  3918 3918 3918 3918 3918 3913 3918 3918\n",
      " $ cases_ref_UKB    : int  1200 3521 2556 3778 2341 421 1516 2502\n",
      " $ cases_het_UKB    : int  1981 390 1234 140 1381 1661 1859 1267\n",
      " $ cases_alt_UKB    : int  737 7 128 0 196 1831 543 149\n",
      " $ num_controls_UKB : int  342869 342869 342869 342869 342869 342262 342869 342869\n",
      " $ controls_ref_UKB : int  120730 299379 201989 335330 190144 41930 158636 204407\n",
      " $ controls_het_UKB : int  165328 41963 122252 7489 130499 156145 148957 120638\n",
      " $ controls_alt_UKB : int  56811 1527 18628 50 22226 144187 35276 17824\n",
      " $ model_GHS        : chr  \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" \"ADD-WGR-FIRTH\" ...\n",
      " $ effect_GHS       : num  1.058 0.884 0.813 1.119 0.964 ...\n",
      " $ lci_effect_GHS   : num  0.989 0.773 0.749 0.822 0.894 ...\n",
      " $ uci_effect_GHS   : num  1.131 1.01 0.881 1.524 1.039 ...\n",
      " $ pval_GHS         : num  1.00e-01 6.98e-02 5.06e-07 4.74e-01 3.34e-01 ...\n",
      " $ aaf_GHS          : num  0.4269 0.0673 0.231 0.0121 0.2656 ...\n",
      " $ num_cases_GHS    : int  1888 1888 1888 1888 1888 1888 1888 1888\n",
      " $ cases_ref_GHS    : int  582 1666 1198 1839 1033 182 685 1223\n",
      " $ cases_het_GHS    : int  929 217 635 49 728 796 906 605\n",
      " $ cases_alt_GHS    : int  377 5 55 0 127 910 297 60\n",
      " $ num_controls_GHS : int  135106 135106 135106 135106 135106 135106 135106 135106\n",
      " $ controls_ref_GHS : int  43761 117524 79484 131863 72823 16801 61404 81859\n",
      " $ controls_het_GHS : int  66338 16985 48313 3220 52777 61441 59332 46623\n",
      " $ controls_alt_GHS : int  25007 597 7309 23 9506 56864 14370 6624\n",
      " $ nearestGene      : chr  \"ATM\" \"TCL1B\" \"TCL1A\" \"TP53\" ...\n",
      " $ dist_nearestGene : int  0 0 283 0 0 72881 0 1003\n",
      " $ variantEffect    : chr  \"downstream,intronic\" \"intronic\" \"upstream\" \"3_prime_UTR\" ...\n",
      " $ CADD             : num  1.65 NA 1.71 15.7 16.71 ...\n",
      " $ maxCADD_inLD     : num  11.47 NA 4.01 15.7 16.71 ...\n",
      " $ Lof_Missense_inLD: logi  NA NA NA NA NA NA ...\n",
      " $ Same_Direction   : int  1 1 1 1 1 1 1 1\n",
      " $ CHROM            : chr  \"chr11\" \"chr14\" \"chr14\" \"chr17\" ...\n",
      " $ varID_hg19       : chr  \"11:108177097:T:A\" \"14:96153809:CTAA:C\" \"14:96180685:T:G\" \"17:7571752:T:G\" ...\n",
      " $ Start_500kb      : int  107677097 95653809 95680685 7071752 127682074 23843330 785974 817481\n",
      " $ End_500kb        : int  108677097 96653809 96680685 8071752 128682074 24843330 1785974 1817481\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n"
     ]
    }
   ],
   "source": [
    "# load prev. CH GWAS loci \n",
    "chip_hg38tohg19 <- fread(\"ch_rg22.hg19.csv\", header=T)\n",
    "str(chip_hg38tohg19)  \n",
    "\n",
    "dnmt3a_hg38tohg19 <- fread(\"dnmt3a_rg22.hg19.csv\", header=T)\n",
    "str(dnmt3a_hg38tohg19)\n",
    "\n",
    "tet2_hg38tohg19 <- fread(\"tet2_rg22.hg19.csv\", header=T)\n",
    "str(tet2_hg38tohg19)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "## Load phenotypes\n",
    "aric_baseline_n_v05 <- fread(\"../../pheno/aric_baseline_n_v05_N4189.pheno_ch_status_trajectory.23Mar2023.csv\", header=T)\n",
    "aric_baseline_n_v05$dAge <- aric_baseline_n_v05$Age - aric_baseline_n_v05$age_base"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "TRUE \n",
       "4189 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# dbGaP linker\n",
    "aric_dbgap_linker <- fread(\"../../pheno/phs000280.v7.pht001441.v7.p1.ARIC_Sample.MULTI.txt.gz\", \n",
    "                           skip = 10, fill = TRUE, header = T)\n",
    "\n",
    "aric_dbgap_linker <- aric_dbgap_linker[,c(1,2,4,5)]\n",
    "\n",
    "aric_dbgap_linker <- subset(aric_dbgap_linker, !duplicated(aric_dbgap_linker$SAMPLE_ID))\n",
    "\n",
    "table(aric_baseline_n_v05$GWAS_ID %in% aric_dbgap_linker$SAMPLE_ID, exclude = NULL)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 110</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>SAMPLE_ID</th><th scope=col>dbGaP_Subject_ID</th><th scope=col>dbGaP_Sample_ID</th><th scope=col>SUBJECT_ID</th><th scope=col>ARIC_ID</th><th scope=col>GWASID_Visit</th><th scope=col>cram_id</th><th scope=col>Visit.x</th><th scope=col>Gender</th><th scope=col>Age</th><th scope=col>...</th><th scope=col>CH_SF</th><th scope=col>CH_DDR</th><th scope=col>CH_DNMT3A</th><th scope=col>CH_TET2</th><th scope=col>CH_ASXL1</th><th scope=col>nonHDL_base_std</th><th scope=col>age_base_sqr</th><th scope=col>Center</th><th scope=col>incident_CH_or_growingClones</th><th scope=col>dAge</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>...</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>A00004</td><td>288368</td><td>1786858</td><td>131031724</td><td>M102321</td><td>A00004_V05</td><td>H5TH7DSX2-2-IDUDI0072</td><td>V05</td><td>F</td><td>71</td><td>...</td><td>NA</td><td>NA</td><td> 1</td><td>NA</td><td>NA</td><td>-0.06472577</td><td>2601</td><td>M  </td><td>1</td><td>20</td></tr>\n",
       "\t<tr><td>A00009</td><td>227213</td><td>1786861</td><td>131051876</td><td>F101210</td><td>A00009_V05</td><td>HNTCYDSX2-2-IDUDI0060</td><td>V05</td><td>F</td><td>76</td><td>...</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td>-0.16103907</td><td>3249</td><td>F_J</td><td>0</td><td>19</td></tr>\n",
       "\t<tr><td>A00012</td><td>232161</td><td>1786863</td><td>131188895</td><td>W121907</td><td>A00012_V05</td><td>HNTGWDSX2-1-IDUDI0006</td><td>V05</td><td>F</td><td>79</td><td>...</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td>-0.16966414</td><td>3249</td><td>W  </td><td>0</td><td>22</td></tr>\n",
       "\t<tr><td>A00013</td><td>222837</td><td>1387960</td><td>131123798</td><td>M263966</td><td>A00013_V05</td><td>HKNWTDSX2-3-IDUDI0063</td><td>V05</td><td>M</td><td>82</td><td>...</td><td>NA</td><td>NA</td><td>NA</td><td> 1</td><td>NA</td><td>-0.07766338</td><td>3844</td><td>M  </td><td>1</td><td>20</td></tr>\n",
       "\t<tr><td>A00015</td><td>228166</td><td>1786865</td><td>131154692</td><td>F101837</td><td>A00015_V05</td><td>H252LDSX3-1-IDUDI0036</td><td>V05</td><td>M</td><td>82</td><td>...</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td>-0.13947639</td><td>3969</td><td>F_J</td><td>0</td><td>19</td></tr>\n",
       "\t<tr><td>A00017</td><td>224264</td><td>1786866</td><td>131017525</td><td>M132251</td><td>A00017_V05</td><td>HY2KGDSX2-1-IDUDI0021</td><td>V05</td><td>M</td><td>83</td><td>...</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td> 0</td><td>-0.07047582</td><td>3721</td><td>M  </td><td>0</td><td>22</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 110\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " SAMPLE\\_ID & dbGaP\\_Subject\\_ID & dbGaP\\_Sample\\_ID & SUBJECT\\_ID & ARIC\\_ID & GWASID\\_Visit & cram\\_id & Visit.x & Gender & Age & ... & CH\\_SF & CH\\_DDR & CH\\_DNMT3A & CH\\_TET2 & CH\\_ASXL1 & nonHDL\\_base\\_std & age\\_base\\_sqr & Center & incident\\_CH\\_or\\_growingClones & dAge\\\\\n",
       " <chr> & <int> & <int> & <dbl> & <chr> & <chr> & <chr> & <chr> & <chr> & <int> & ... & <int> & <int> & <int> & <int> & <int> & <dbl> & <int> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t A00004 & 288368 & 1786858 & 131031724 & M102321 & A00004\\_V05 & H5TH7DSX2-2-IDUDI0072 & V05 & F & 71 & ... & NA & NA &  1 & NA & NA & -0.06472577 & 2601 & M   & 1 & 20\\\\\n",
       "\t A00009 & 227213 & 1786861 & 131051876 & F101210 & A00009\\_V05 & HNTCYDSX2-2-IDUDI0060 & V05 & F & 76 & ... &  0 &  0 &  0 &  0 &  0 & -0.16103907 & 3249 & F\\_J & 0 & 19\\\\\n",
       "\t A00012 & 232161 & 1786863 & 131188895 & W121907 & A00012\\_V05 & HNTGWDSX2-1-IDUDI0006 & V05 & F & 79 & ... &  0 &  0 &  0 &  0 &  0 & -0.16966414 & 3249 & W   & 0 & 22\\\\\n",
       "\t A00013 & 222837 & 1387960 & 131123798 & M263966 & A00013\\_V05 & HKNWTDSX2-3-IDUDI0063 & V05 & M & 82 & ... & NA & NA & NA &  1 & NA & -0.07766338 & 3844 & M   & 1 & 20\\\\\n",
       "\t A00015 & 228166 & 1786865 & 131154692 & F101837 & A00015\\_V05 & H252LDSX3-1-IDUDI0036 & V05 & M & 82 & ... &  0 &  0 &  0 &  0 &  0 & -0.13947639 & 3969 & F\\_J & 0 & 19\\\\\n",
       "\t A00017 & 224264 & 1786866 & 131017525 & M132251 & A00017\\_V05 & HY2KGDSX2-1-IDUDI0021 & V05 & M & 83 & ... &  0 &  0 &  0 &  0 &  0 & -0.07047582 & 3721 & M   & 0 & 22\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 110\n",
       "\n",
       "| SAMPLE_ID &lt;chr&gt; | dbGaP_Subject_ID &lt;int&gt; | dbGaP_Sample_ID &lt;int&gt; | SUBJECT_ID &lt;dbl&gt; | ARIC_ID &lt;chr&gt; | GWASID_Visit &lt;chr&gt; | cram_id &lt;chr&gt; | Visit.x &lt;chr&gt; | Gender &lt;chr&gt; | Age &lt;int&gt; | ... ... | CH_SF &lt;int&gt; | CH_DDR &lt;int&gt; | CH_DNMT3A &lt;int&gt; | CH_TET2 &lt;int&gt; | CH_ASXL1 &lt;int&gt; | nonHDL_base_std &lt;dbl&gt; | age_base_sqr &lt;int&gt; | Center &lt;chr&gt; | incident_CH_or_growingClones &lt;int&gt; | dAge &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| A00004 | 288368 | 1786858 | 131031724 | M102321 | A00004_V05 | H5TH7DSX2-2-IDUDI0072 | V05 | F | 71 | ... | NA | NA |  1 | NA | NA | -0.06472577 | 2601 | M   | 1 | 20 |\n",
       "| A00009 | 227213 | 1786861 | 131051876 | F101210 | A00009_V05 | HNTCYDSX2-2-IDUDI0060 | V05 | F | 76 | ... |  0 |  0 |  0 |  0 |  0 | -0.16103907 | 3249 | F_J | 0 | 19 |\n",
       "| A00012 | 232161 | 1786863 | 131188895 | W121907 | A00012_V05 | HNTGWDSX2-1-IDUDI0006 | V05 | F | 79 | ... |  0 |  0 |  0 |  0 |  0 | -0.16966414 | 3249 | W   | 0 | 22 |\n",
       "| A00013 | 222837 | 1387960 | 131123798 | M263966 | A00013_V05 | HKNWTDSX2-3-IDUDI0063 | V05 | M | 82 | ... | NA | NA | NA |  1 | NA | -0.07766338 | 3844 | M   | 1 | 20 |\n",
       "| A00015 | 228166 | 1786865 | 131154692 | F101837 | A00015_V05 | H252LDSX3-1-IDUDI0036 | V05 | M | 82 | ... |  0 |  0 |  0 |  0 |  0 | -0.13947639 | 3969 | F_J | 0 | 19 |\n",
       "| A00017 | 224264 | 1786866 | 131017525 | M132251 | A00017_V05 | HY2KGDSX2-1-IDUDI0021 | V05 | M | 83 | ... |  0 |  0 |  0 |  0 |  0 | -0.07047582 | 3721 | M   | 0 | 22 |\n",
       "\n"
      ],
      "text/plain": [
       "  SAMPLE_ID dbGaP_Subject_ID dbGaP_Sample_ID SUBJECT_ID ARIC_ID GWASID_Visit\n",
       "1 A00004    288368           1786858         131031724  M102321 A00004_V05  \n",
       "2 A00009    227213           1786861         131051876  F101210 A00009_V05  \n",
       "3 A00012    232161           1786863         131188895  W121907 A00012_V05  \n",
       "4 A00013    222837           1387960         131123798  M263966 A00013_V05  \n",
       "5 A00015    228166           1786865         131154692  F101837 A00015_V05  \n",
       "6 A00017    224264           1786866         131017525  M132251 A00017_V05  \n",
       "  cram_id               Visit.x Gender Age ... CH_SF CH_DDR CH_DNMT3A CH_TET2\n",
       "1 H5TH7DSX2-2-IDUDI0072 V05     F      71  ... NA    NA      1        NA     \n",
       "2 HNTCYDSX2-2-IDUDI0060 V05     F      76  ...  0     0      0         0     \n",
       "3 HNTGWDSX2-1-IDUDI0006 V05     F      79  ...  0     0      0         0     \n",
       "4 HKNWTDSX2-3-IDUDI0063 V05     M      82  ... NA    NA     NA         1     \n",
       "5 H252LDSX3-1-IDUDI0036 V05     M      82  ...  0     0      0         0     \n",
       "6 HY2KGDSX2-1-IDUDI0021 V05     M      83  ...  0     0      0         0     \n",
       "  CH_ASXL1 nonHDL_base_std age_base_sqr Center incident_CH_or_growingClones\n",
       "1 NA       -0.06472577     2601         M      1                           \n",
       "2  0       -0.16103907     3249         F_J    0                           \n",
       "3  0       -0.16966414     3249         W      0                           \n",
       "4 NA       -0.07766338     3844         M      1                           \n",
       "5  0       -0.13947639     3969         F_J    0                           \n",
       "6  0       -0.07047582     3721         M      0                           \n",
       "  dAge\n",
       "1 20  \n",
       "2 19  \n",
       "3 22  \n",
       "4 20  \n",
       "5 19  \n",
       "6 22  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## link dbGaP ids with GWASID\n",
    "aric_baseline_n_v05 <- merge(aric_dbgap_linker, \n",
    "                              aric_baseline_n_v05,\n",
    "                              by.x = \"SAMPLE_ID\",\n",
    "                              by.y = \"GWAS_ID\")\n",
    "\n",
    "aric_baseline_n_v05$SUBJECT_ID <- as.numeric(aric_baseline_n_v05$SUBJECT_ID)\n",
    "\n",
    "head(aric_baseline_n_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE \n",
       " 1497  2692 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## PCA data\n",
    "pca_ea <- fread(\"../../pca/ARIC_EurAm_chr1_22.aray_snps.eigenvec\", header = T)\n",
    "# pca_ea$IID <- as.character(pca_ea$IID)\n",
    "pca_aa <- fread(\"../../pca/ARIC_AfrAm_chr1_22.aray_snps.eigenvec\", header = T)\n",
    "# pca_ea$IID <- as.character(pca_ea$IID)\n",
    "table(aric_baseline_n_v05$SUBJECT_ID %in% pca_ea$IID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "2692"
      ],
      "text/latex": [
       "2692"
      ],
      "text/markdown": [
       "2692"
      ],
      "text/plain": [
       "[1] 2692"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# EA: 2692\n",
    "aric_baseline_n_v05_ea <- merge(aric_baseline_n_v05, pca_ea, \n",
    "                                by.x=\"SUBJECT_ID\", by.y=\"IID\")\n",
    "\n",
    "nrow(aric_baseline_n_v05_ea)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "701"
      ],
      "text/latex": [
       "701"
      ],
      "text/markdown": [
       "701"
      ],
      "text/plain": [
       "[1] 701"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# EA: 2692\n",
    "aric_baseline_n_v05_aa <- merge(aric_baseline_n_v05, pca_aa, \n",
    "                                by.x=\"SUBJECT_ID\", by.y=\"IID\")\n",
    "\n",
    "nrow(aric_baseline_n_v05_aa)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SUBJECT_ID'</li><li>'SUBJECT_ID'</li><li>'SAMPLE_ID'</li><li>'GWASID_Visit'</li><li>'dbGaP_Subject_ID'</li><li>'dbGaP_Sample_ID'</li><li>'ARIC_ID'</li><li>'Visit.y'</li><li>'Gender'</li><li>'Age'</li><li>'CH_baseline'</li><li>'CH_v05'</li><li>'incident_CH'</li><li>'incident_CHvaf05'</li><li>'incident_CHvaf10'</li><li>'incident_DNMT3A'</li><li>'incident_TET2'</li><li>'incident_ASXL1'</li><li>'incident_DTA'</li><li>'incident_SF'</li><li>'incident_DDR'</li><li>'incident_CH_or_growingClones'</li><li>'age_base'</li><li>'age_base_sqr'</li><li>'Center'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'PC8'</li><li>'PC9'</li><li>'PC10'</li><li>'ever_smoke'</li><li>'v2_vs_other'</li><li>'Sex'</li><li>'BMI_cat'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SUBJECT\\_ID'\n",
       "\\item 'SUBJECT\\_ID'\n",
       "\\item 'SAMPLE\\_ID'\n",
       "\\item 'GWASID\\_Visit'\n",
       "\\item 'dbGaP\\_Subject\\_ID'\n",
       "\\item 'dbGaP\\_Sample\\_ID'\n",
       "\\item 'ARIC\\_ID'\n",
       "\\item 'Visit.y'\n",
       "\\item 'Gender'\n",
       "\\item 'Age'\n",
       "\\item 'CH\\_baseline'\n",
       "\\item 'CH\\_v05'\n",
       "\\item 'incident\\_CH'\n",
       "\\item 'incident\\_CHvaf05'\n",
       "\\item 'incident\\_CHvaf10'\n",
       "\\item 'incident\\_DNMT3A'\n",
       "\\item 'incident\\_TET2'\n",
       "\\item 'incident\\_ASXL1'\n",
       "\\item 'incident\\_DTA'\n",
       "\\item 'incident\\_SF'\n",
       "\\item 'incident\\_DDR'\n",
       "\\item 'incident\\_CH\\_or\\_growingClones'\n",
       "\\item 'age\\_base'\n",
       "\\item 'age\\_base\\_sqr'\n",
       "\\item 'Center'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'PC8'\n",
       "\\item 'PC9'\n",
       "\\item 'PC10'\n",
       "\\item 'ever\\_smoke'\n",
       "\\item 'v2\\_vs\\_other'\n",
       "\\item 'Sex'\n",
       "\\item 'BMI\\_cat'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SUBJECT_ID'\n",
       "2. 'SUBJECT_ID'\n",
       "3. 'SAMPLE_ID'\n",
       "4. 'GWASID_Visit'\n",
       "5. 'dbGaP_Subject_ID'\n",
       "6. 'dbGaP_Sample_ID'\n",
       "7. 'ARIC_ID'\n",
       "8. 'Visit.y'\n",
       "9. 'Gender'\n",
       "10. 'Age'\n",
       "11. 'CH_baseline'\n",
       "12. 'CH_v05'\n",
       "13. 'incident_CH'\n",
       "14. 'incident_CHvaf05'\n",
       "15. 'incident_CHvaf10'\n",
       "16. 'incident_DNMT3A'\n",
       "17. 'incident_TET2'\n",
       "18. 'incident_ASXL1'\n",
       "19. 'incident_DTA'\n",
       "20. 'incident_SF'\n",
       "21. 'incident_DDR'\n",
       "22. 'incident_CH_or_growingClones'\n",
       "23. 'age_base'\n",
       "24. 'age_base_sqr'\n",
       "25. 'Center'\n",
       "26. 'PC1'\n",
       "27. 'PC2'\n",
       "28. 'PC3'\n",
       "29. 'PC4'\n",
       "30. 'PC5'\n",
       "31. 'PC6'\n",
       "32. 'PC7'\n",
       "33. 'PC8'\n",
       "34. 'PC9'\n",
       "35. 'PC10'\n",
       "36. 'ever_smoke'\n",
       "37. 'v2_vs_other'\n",
       "38. 'Sex'\n",
       "39. 'BMI_cat'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SUBJECT_ID\"                   \"SUBJECT_ID\"                  \n",
       " [3] \"SAMPLE_ID\"                    \"GWASID_Visit\"                \n",
       " [5] \"dbGaP_Subject_ID\"             \"dbGaP_Sample_ID\"             \n",
       " [7] \"ARIC_ID\"                      \"Visit.y\"                     \n",
       " [9] \"Gender\"                       \"Age\"                         \n",
       "[11] \"CH_baseline\"                  \"CH_v05\"                      \n",
       "[13] \"incident_CH\"                  \"incident_CHvaf05\"            \n",
       "[15] \"incident_CHvaf10\"             \"incident_DNMT3A\"             \n",
       "[17] \"incident_TET2\"                \"incident_ASXL1\"              \n",
       "[19] \"incident_DTA\"                 \"incident_SF\"                 \n",
       "[21] \"incident_DDR\"                 \"incident_CH_or_growingClones\"\n",
       "[23] \"age_base\"                     \"age_base_sqr\"                \n",
       "[25] \"Center\"                       \"PC1\"                         \n",
       "[27] \"PC2\"                          \"PC3\"                         \n",
       "[29] \"PC4\"                          \"PC5\"                         \n",
       "[31] \"PC6\"                          \"PC7\"                         \n",
       "[33] \"PC8\"                          \"PC9\"                         \n",
       "[35] \"PC10\"                         \"ever_smoke\"                  \n",
       "[37] \"v2_vs_other\"                  \"Sex\"                         \n",
       "[39] \"BMI_cat\"                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes 'data.table' and 'data.frame':\t2692 obs. of  39 variables:\n",
      " $ FID                         : int  131000113 131000147 131000179 131000253 131000354 131000401 131000455 131000596 131000641 131000672 ...\n",
      " $ IID                         : int  131000113 131000147 131000179 131000253 131000354 131000401 131000455 131000596 131000641 131000672 ...\n",
      " $ GWAS_ID                     : chr  \"A15057\" \"A04826\" \"A02787\" \"A07971\" ...\n",
      " $ GWASID_Visit                : chr  \"A15057_V05\" \"A04826_V05\" \"A02787_V05\" \"A07971_V05\" ...\n",
      " $ dbGaP_Subject_ID            : int  228789 232712 230453 231733 224837 229698 234835 230455 228925 229174 ...\n",
      " $ dbGaP_Sample_ID             : int  1336383 1788922 1788029 1389147 1787805 1388504 1787033 1388831 1336156 1334770 ...\n",
      " $ ARIC_ID                     : chr  \"W293092\" \"M127886\" \"W235739\" \"M288311\" ...\n",
      " $ Visit.y                     : chr  \"V02\" \"V02\" \"V02\" \"V02\" ...\n",
      " $ Gender                      : chr  \"M\" \"M\" \"F\" \"M\" ...\n",
      " $ Age                         : int  74 80 74 67 81 77 87 84 85 84 ...\n",
      " $ CH_baseline                 : int  1 0 0 0 0 0 0 0 0 0 ...\n",
      " $ CH_v05                      : int  1 0 0 0 0 0 0 1 1 0 ...\n",
      " $ incident_CH                 : int  NA 0 0 0 0 0 0 1 1 0 ...\n",
      " $ incident_CHvaf05            : int  NA 0 0 0 0 0 0 NA 1 0 ...\n",
      " $ incident_CHvaf10            : int  NA 0 0 0 0 0 0 NA NA 0 ...\n",
      " $ incident_DNMT3A             : int  NA 0 0 0 0 0 0 0 0 0 ...\n",
      " $ incident_TET2               : int  NA 0 0 0 0 0 0 1 0 0 ...\n",
      " $ incident_ASXL1              : int  NA 0 0 0 0 0 0 0 0 0 ...\n",
      " $ incident_DTA                : int  NA 0 0 0 0 0 0 1 0 0 ...\n",
      " $ incident_SF                 : int  NA 0 0 0 0 0 0 0 1 0 ...\n",
      " $ incident_DDR                : int  NA 0 0 0 0 0 0 0 0 0 ...\n",
      " $ incident_CH_or_growingClones: int  1 0 0 0 0 0 0 1 1 0 ...\n",
      " $ age_base                    : int  55 58 54 48 61 56 70 65 67 62 ...\n",
      " $ age_base_sqr                : int  3025 3364 2916 2304 3721 3136 4900 4225 4489 3844 ...\n",
      " $ Center                      : chr  \"W\" \"M\" \"W\" \"M\" ...\n",
      " $ PC1                         : num  -0.0475 -0.00841 0.00249 0.0074 0.0022 ...\n",
      " $ PC2                         : num  -0.01544 -0.00766 -0.00469 0.00312 -0.00599 ...\n",
      " $ PC3                         : num  -0.01336 -0.00272 -0.02079 0.0167 -0.01534 ...\n",
      " $ PC4                         : num  0.01126 0.00338 -0.00219 0.03581 -0.00324 ...\n",
      " $ PC5                         : num  -0.007131 -0.000637 0.000839 -0.006886 -0.000967 ...\n",
      " $ PC6                         : num  -0.006007 0.005957 0.000279 -0.00311 0.003249 ...\n",
      " $ PC7                         : num  -2.41e-05 3.18e-03 4.24e-03 -2.39e-03 1.44e-03 ...\n",
      " $ PC8                         : num  0.00733 0.00565 -0.0055 -0.01742 -0.00095 ...\n",
      " $ PC9                         : num  0.00636 0.00583 0.01091 0.00688 0.00335 ...\n",
      " $ PC10                        : num  -0.00978 -0.00586 0.01042 -0.00981 0.00647 ...\n",
      " $ ever_smoke                  : int  0 1 1 1 1 0 0 1 1 0 ...\n",
      " $ v2_vs_other                 : chr  \"V02\" \"V02\" \"V02\" \"V02\" ...\n",
      " $ Sex                         : chr  \"M\" \"M\" \"F\" \"M\" ...\n",
      " $ BMI_cat                     : chr  \">30\" \"<=25\" \"25-30\" \"25-30\" ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n",
      " - attr(*, \"sorted\")= chr \"IID\"\n"
     ]
    }
   ],
   "source": [
    "names(aric_baseline_n_v05_ea[, c(1,1,2,6,3:5,55,9:21,109, 61,107,108,112:121,85,86,87,89)])\n",
    "aric_baseline_n_v05_ea_gwas <- aric_baseline_n_v05_ea[, c(1,1,2,6,3:5,55,9:21,109, 61,107,108,112:121,85,86,87,89)]\n",
    "names(aric_baseline_n_v05_ea_gwas) <- c(\"FID\",\"IID\",\"GWAS_ID\", names(aric_baseline_n_v05_ea_gwas)[4:39])\n",
    "str(aric_baseline_n_v05_ea_gwas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'SUBJECT_ID'</li><li>'SUBJECT_ID'</li><li>'SAMPLE_ID'</li><li>'GWASID_Visit'</li><li>'dbGaP_Subject_ID'</li><li>'dbGaP_Sample_ID'</li><li>'ARIC_ID'</li><li>'Visit.y'</li><li>'Gender'</li><li>'Age'</li><li>'CH_baseline'</li><li>'CH_v05'</li><li>'incident_CH'</li><li>'incident_CHvaf05'</li><li>'incident_CHvaf10'</li><li>'incident_DNMT3A'</li><li>'incident_TET2'</li><li>'incident_ASXL1'</li><li>'incident_DTA'</li><li>'incident_SF'</li><li>'incident_DDR'</li><li>'incident_CH_or_growingClones'</li><li>'age_base'</li><li>'age_base_sqr'</li><li>'Center'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'PC8'</li><li>'PC9'</li><li>'PC10'</li><li>'ever_smoke'</li><li>'v2_vs_other'</li><li>'Sex'</li><li>'BMI_cat'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'SUBJECT\\_ID'\n",
       "\\item 'SUBJECT\\_ID'\n",
       "\\item 'SAMPLE\\_ID'\n",
       "\\item 'GWASID\\_Visit'\n",
       "\\item 'dbGaP\\_Subject\\_ID'\n",
       "\\item 'dbGaP\\_Sample\\_ID'\n",
       "\\item 'ARIC\\_ID'\n",
       "\\item 'Visit.y'\n",
       "\\item 'Gender'\n",
       "\\item 'Age'\n",
       "\\item 'CH\\_baseline'\n",
       "\\item 'CH\\_v05'\n",
       "\\item 'incident\\_CH'\n",
       "\\item 'incident\\_CHvaf05'\n",
       "\\item 'incident\\_CHvaf10'\n",
       "\\item 'incident\\_DNMT3A'\n",
       "\\item 'incident\\_TET2'\n",
       "\\item 'incident\\_ASXL1'\n",
       "\\item 'incident\\_DTA'\n",
       "\\item 'incident\\_SF'\n",
       "\\item 'incident\\_DDR'\n",
       "\\item 'incident\\_CH\\_or\\_growingClones'\n",
       "\\item 'age\\_base'\n",
       "\\item 'age\\_base\\_sqr'\n",
       "\\item 'Center'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'PC8'\n",
       "\\item 'PC9'\n",
       "\\item 'PC10'\n",
       "\\item 'ever\\_smoke'\n",
       "\\item 'v2\\_vs\\_other'\n",
       "\\item 'Sex'\n",
       "\\item 'BMI\\_cat'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'SUBJECT_ID'\n",
       "2. 'SUBJECT_ID'\n",
       "3. 'SAMPLE_ID'\n",
       "4. 'GWASID_Visit'\n",
       "5. 'dbGaP_Subject_ID'\n",
       "6. 'dbGaP_Sample_ID'\n",
       "7. 'ARIC_ID'\n",
       "8. 'Visit.y'\n",
       "9. 'Gender'\n",
       "10. 'Age'\n",
       "11. 'CH_baseline'\n",
       "12. 'CH_v05'\n",
       "13. 'incident_CH'\n",
       "14. 'incident_CHvaf05'\n",
       "15. 'incident_CHvaf10'\n",
       "16. 'incident_DNMT3A'\n",
       "17. 'incident_TET2'\n",
       "18. 'incident_ASXL1'\n",
       "19. 'incident_DTA'\n",
       "20. 'incident_SF'\n",
       "21. 'incident_DDR'\n",
       "22. 'incident_CH_or_growingClones'\n",
       "23. 'age_base'\n",
       "24. 'age_base_sqr'\n",
       "25. 'Center'\n",
       "26. 'PC1'\n",
       "27. 'PC2'\n",
       "28. 'PC3'\n",
       "29. 'PC4'\n",
       "30. 'PC5'\n",
       "31. 'PC6'\n",
       "32. 'PC7'\n",
       "33. 'PC8'\n",
       "34. 'PC9'\n",
       "35. 'PC10'\n",
       "36. 'ever_smoke'\n",
       "37. 'v2_vs_other'\n",
       "38. 'Sex'\n",
       "39. 'BMI_cat'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"SUBJECT_ID\"                   \"SUBJECT_ID\"                  \n",
       " [3] \"SAMPLE_ID\"                    \"GWASID_Visit\"                \n",
       " [5] \"dbGaP_Subject_ID\"             \"dbGaP_Sample_ID\"             \n",
       " [7] \"ARIC_ID\"                      \"Visit.y\"                     \n",
       " [9] \"Gender\"                       \"Age\"                         \n",
       "[11] \"CH_baseline\"                  \"CH_v05\"                      \n",
       "[13] \"incident_CH\"                  \"incident_CHvaf05\"            \n",
       "[15] \"incident_CHvaf10\"             \"incident_DNMT3A\"             \n",
       "[17] \"incident_TET2\"                \"incident_ASXL1\"              \n",
       "[19] \"incident_DTA\"                 \"incident_SF\"                 \n",
       "[21] \"incident_DDR\"                 \"incident_CH_or_growingClones\"\n",
       "[23] \"age_base\"                     \"age_base_sqr\"                \n",
       "[25] \"Center\"                       \"PC1\"                         \n",
       "[27] \"PC2\"                          \"PC3\"                         \n",
       "[29] \"PC4\"                          \"PC5\"                         \n",
       "[31] \"PC6\"                          \"PC7\"                         \n",
       "[33] \"PC8\"                          \"PC9\"                         \n",
       "[35] \"PC10\"                         \"ever_smoke\"                  \n",
       "[37] \"v2_vs_other\"                  \"Sex\"                         \n",
       "[39] \"BMI_cat\"                     "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Classes 'data.table' and 'data.frame':\t701 obs. of  39 variables:\n",
      " $ FID                         : int  131000053 131000094 131000655 131001064 131001173 131001245 131001306 131001309 131001392 131001460 ...\n",
      " $ IID                         : int  131000053 131000094 131000655 131001064 131001173 131001245 131001306 131001309 131001392 131001460 ...\n",
      " $ GWAS_ID                     : chr  \"A05162\" \"A14961\" \"A12667\" \"A11283\" ...\n",
      " $ GWASID_Visit                : chr  \"A05162_V05\" \"A14961_V05\" \"A12667_V05\" \"A11283_V05\" ...\n",
      " $ dbGaP_Subject_ID            : int  230296 229859 224585 228563 229539 224645 226634 232824 229013 231474 ...\n",
      " $ dbGaP_Sample_ID             : int  699030 701005 1792295 1389627 699670 699409 699581 701049 699835 698355 ...\n",
      " $ ARIC_ID                     : chr  \"J187560\" \"J124491\" \"J307167\" \"J304595\" ...\n",
      " $ Visit.y                     : chr  \"V02\" \"V02\" \"V01\" \"V02\" ...\n",
      " $ Gender                      : chr  \"M\" \"F\" \"F\" \"M\" ...\n",
      " $ Age                         : int  72 69 67 70 84 69 70 77 71 79 ...\n",
      " $ CH_baseline                 : int  0 0 0 0 0 0 0 0 1 0 ...\n",
      " $ CH_v05                      : int  0 0 0 0 0 0 0 1 1 0 ...\n",
      " $ incident_CH                 : int  0 0 0 0 0 0 0 1 NA 0 ...\n",
      " $ incident_CHvaf05            : int  0 0 0 0 0 0 0 1 NA 0 ...\n",
      " $ incident_CHvaf10            : int  0 0 0 0 0 0 0 1 NA 0 ...\n",
      " $ incident_DNMT3A             : int  0 0 0 0 0 0 0 0 NA 0 ...\n",
      " $ incident_TET2               : int  0 0 0 0 0 0 0 0 NA 0 ...\n",
      " $ incident_ASXL1              : int  0 0 0 0 0 0 0 0 NA 0 ...\n",
      " $ incident_DTA                : int  0 0 0 0 0 0 0 0 NA 0 ...\n",
      " $ incident_SF                 : int  0 0 0 0 0 0 0 0 NA 0 ...\n",
      " $ incident_DDR                : int  0 0 0 0 0 0 0 0 NA 0 ...\n",
      " $ incident_CH_or_growingClones: int  0 0 0 0 0 0 0 1 NA 0 ...\n",
      " $ age_base                    : int  51 49 45 51 64 50 51 55 51 57 ...\n",
      " $ age_base_sqr                : int  2601 2401 2025 2601 4096 2500 2601 3025 2601 3249 ...\n",
      " $ Center                      : chr  \"F_J\" \"F_J\" \"F_J\" \"F_J\" ...\n",
      " $ PC1                         : num  0.01524 0.03472 -0.02298 -0.00737 -0.01231 ...\n",
      " $ PC2                         : num  -0.05 0.029 -0.00942 -0.01432 -0.00609 ...\n",
      " $ PC3                         : num  -0.00265 0.01182 -0.02226 -0.00304 -0.02567 ...\n",
      " $ PC4                         : num  -0.0161 -0.0203 -0.0132 -0.0167 0.0165 ...\n",
      " $ PC5                         : num  -0.02319 0.00267 0.00583 -0.02798 -0.00476 ...\n",
      " $ PC6                         : num  -0.00254 -0.01573 -0.0078 -0.01208 0.02233 ...\n",
      " $ PC7                         : num  -0.00183 -0.00157 -0.01494 0.00344 -0.01398 ...\n",
      " $ PC8                         : num  -0.03706 0.01032 -0.00611 -0.01596 0.00148 ...\n",
      " $ PC9                         : num  -0.00436 0.04678 0.01741 0.01301 -0.02278 ...\n",
      " $ PC10                        : num  0.020383 -0.003932 0.002619 -0.028102 -0.000707 ...\n",
      " $ ever_smoke                  : int  1 0 0 0 1 0 1 0 0 0 ...\n",
      " $ v2_vs_other                 : chr  \"V02\" \"V02\" \"Others\" \"V02\" ...\n",
      " $ Sex                         : chr  \"M\" \"F\" \"F\" \"M\" ...\n",
      " $ BMI_cat                     : chr  \">30\" \">30\" \"<=25\" \"25-30\" ...\n",
      " - attr(*, \".internal.selfref\")=<externalptr> \n",
      " - attr(*, \"sorted\")= chr \"IID\"\n"
     ]
    }
   ],
   "source": [
    "names(aric_baseline_n_v05_aa[, c(1,1,2,6,3:5,55,9:21,109, 61,107,108,112:121,85,86,87,89)])\n",
    "aric_baseline_n_v05_aa_gwas <- aric_baseline_n_v05_aa[, c(1,1,2,6,3:5,55,9:21,109, 61,107,108,112:121,85,86,87,89)]\n",
    "names(aric_baseline_n_v05_aa_gwas) <- c(\"FID\",\"IID\",\"GWAS_ID\", names(aric_baseline_n_v05_aa_gwas)[4:39])\n",
    "str(aric_baseline_n_v05_aa_gwas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "## GWAS_ID==SAMPLE_ID\n",
    "# fwrite(aric_baseline_n_v05_ea, \"../aric_baseline_n_v05_ea_PCA.2023Apr3.csv\", \n",
    "  #      row.names = F, col.names = T, sep=\",\", na=\"NA\")\n",
    "# fwrite(aric_baseline_n_v05_ea_gwas, \"../aric_baseline_n_v05_ea_PCA_GWAS.2023Apr3.tsv\",\n",
    "  #      row.names = F, col.names = T, sep=\"\\t\", na=\"NA\", quote = F)\n",
    "# \n",
    "# fwrite(aric_baseline_n_v05_aa, \"../aric_baseline_n_v05_aa_PCA.2023Apr3.csv\", \n",
    "  #      row.names = F, col.names = T, sep=\",\", na=\"NA\")\n",
    "# fwrite(aric_baseline_n_v05_aa_gwas, \"../aric_baseline_n_v05_aa_PCA_GWAS.2023Apr3.tsv\",\n",
    "  #      row.names = F, col.names = T, sep=\"\\t\", na=\"NA\", quote = F)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 38</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>FID</th><th scope=col>IID</th><th scope=col>GWAS_ID</th><th scope=col>GWASID_Visit</th><th scope=col>dbGaP_Subject_ID</th><th scope=col>dbGaP_Sample_ID</th><th scope=col>ARIC_ID</th><th scope=col>Visit.y</th><th scope=col>Gender</th><th scope=col>Age</th><th scope=col>...</th><th scope=col>PC4</th><th scope=col>PC5</th><th scope=col>PC6</th><th scope=col>PC7</th><th scope=col>PC8</th><th scope=col>PC9</th><th scope=col>PC10</th><th scope=col>ever_smoke</th><th scope=col>v2_vs_other</th><th scope=col>Sex</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>...</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>131000053</td><td>131000053</td><td>A05162</td><td>A05162_V05</td><td>230296</td><td> 699030</td><td>J187560</td><td>V02</td><td>M</td><td>72</td><td>...</td><td>-0.0160964</td><td>-0.02318640</td><td>-0.00253648</td><td>-0.00182851</td><td>-0.03706380</td><td>-0.00436290</td><td> 0.020383400</td><td>1</td><td>V02   </td><td>M</td></tr>\n",
       "\t<tr><td>131000094</td><td>131000094</td><td>A14961</td><td>A14961_V05</td><td>229859</td><td> 701005</td><td>J124491</td><td>V02</td><td>F</td><td>69</td><td>...</td><td>-0.0203355</td><td> 0.00267189</td><td>-0.01572940</td><td>-0.00156514</td><td> 0.01031670</td><td> 0.04677960</td><td>-0.003931650</td><td>0</td><td>V02   </td><td>F</td></tr>\n",
       "\t<tr><td>131000655</td><td>131000655</td><td>A12667</td><td>A12667_V05</td><td>224585</td><td>1792295</td><td>J307167</td><td>V01</td><td>F</td><td>67</td><td>...</td><td>-0.0132270</td><td> 0.00583279</td><td>-0.00779501</td><td>-0.01494340</td><td>-0.00611278</td><td> 0.01740740</td><td> 0.002618600</td><td>0</td><td>Others</td><td>F</td></tr>\n",
       "\t<tr><td>131001064</td><td>131001064</td><td>A11283</td><td>A11283_V05</td><td>228563</td><td>1389627</td><td>J304595</td><td>V02</td><td>M</td><td>70</td><td>...</td><td>-0.0166595</td><td>-0.02797620</td><td>-0.01208170</td><td> 0.00343874</td><td>-0.01595520</td><td> 0.01301230</td><td>-0.028101800</td><td>0</td><td>V02   </td><td>M</td></tr>\n",
       "\t<tr><td>131001173</td><td>131001173</td><td>A08339</td><td>A08339_V05</td><td>229539</td><td> 699670</td><td>J214024</td><td>V02</td><td>F</td><td>84</td><td>...</td><td> 0.0164780</td><td>-0.00476401</td><td> 0.02233330</td><td>-0.01398200</td><td> 0.00148152</td><td>-0.02277820</td><td>-0.000707081</td><td>1</td><td>V02   </td><td>F</td></tr>\n",
       "\t<tr><td>131001245</td><td>131001245</td><td>A07099</td><td>A07099_V05</td><td>224645</td><td> 699409</td><td>J291749</td><td>V02</td><td>F</td><td>69</td><td>...</td><td> 0.0132752</td><td>-0.01870220</td><td>-0.02439060</td><td>-0.00570036</td><td> 0.01178660</td><td> 0.00763499</td><td> 0.007431030</td><td>0</td><td>V02   </td><td>F</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 38\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " FID & IID & GWAS\\_ID & GWASID\\_Visit & dbGaP\\_Subject\\_ID & dbGaP\\_Sample\\_ID & ARIC\\_ID & Visit.y & Gender & Age & ... & PC4 & PC5 & PC6 & PC7 & PC8 & PC9 & PC10 & ever\\_smoke & v2\\_vs\\_other & Sex\\\\\n",
       " <int> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & ... & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 131000053 & 131000053 & A05162 & A05162\\_V05 & 230296 &  699030 & J187560 & V02 & M & 72 & ... & -0.0160964 & -0.02318640 & -0.00253648 & -0.00182851 & -0.03706380 & -0.00436290 &  0.020383400 & 1 & V02    & M\\\\\n",
       "\t 131000094 & 131000094 & A14961 & A14961\\_V05 & 229859 &  701005 & J124491 & V02 & F & 69 & ... & -0.0203355 &  0.00267189 & -0.01572940 & -0.00156514 &  0.01031670 &  0.04677960 & -0.003931650 & 0 & V02    & F\\\\\n",
       "\t 131000655 & 131000655 & A12667 & A12667\\_V05 & 224585 & 1792295 & J307167 & V01 & F & 67 & ... & -0.0132270 &  0.00583279 & -0.00779501 & -0.01494340 & -0.00611278 &  0.01740740 &  0.002618600 & 0 & Others & F\\\\\n",
       "\t 131001064 & 131001064 & A11283 & A11283\\_V05 & 228563 & 1389627 & J304595 & V02 & M & 70 & ... & -0.0166595 & -0.02797620 & -0.01208170 &  0.00343874 & -0.01595520 &  0.01301230 & -0.028101800 & 0 & V02    & M\\\\\n",
       "\t 131001173 & 131001173 & A08339 & A08339\\_V05 & 229539 &  699670 & J214024 & V02 & F & 84 & ... &  0.0164780 & -0.00476401 &  0.02233330 & -0.01398200 &  0.00148152 & -0.02277820 & -0.000707081 & 1 & V02    & F\\\\\n",
       "\t 131001245 & 131001245 & A07099 & A07099\\_V05 & 224645 &  699409 & J291749 & V02 & F & 69 & ... &  0.0132752 & -0.01870220 & -0.02439060 & -0.00570036 &  0.01178660 &  0.00763499 &  0.007431030 & 0 & V02    & F\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 38\n",
       "\n",
       "| FID &lt;int&gt; | IID &lt;int&gt; | GWAS_ID &lt;chr&gt; | GWASID_Visit &lt;chr&gt; | dbGaP_Subject_ID &lt;int&gt; | dbGaP_Sample_ID &lt;int&gt; | ARIC_ID &lt;chr&gt; | Visit.y &lt;chr&gt; | Gender &lt;chr&gt; | Age &lt;int&gt; | ... ... | PC4 &lt;dbl&gt; | PC5 &lt;dbl&gt; | PC6 &lt;dbl&gt; | PC7 &lt;dbl&gt; | PC8 &lt;dbl&gt; | PC9 &lt;dbl&gt; | PC10 &lt;dbl&gt; | ever_smoke &lt;int&gt; | v2_vs_other &lt;chr&gt; | Sex &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 131000053 | 131000053 | A05162 | A05162_V05 | 230296 |  699030 | J187560 | V02 | M | 72 | ... | -0.0160964 | -0.02318640 | -0.00253648 | -0.00182851 | -0.03706380 | -0.00436290 |  0.020383400 | 1 | V02    | M |\n",
       "| 131000094 | 131000094 | A14961 | A14961_V05 | 229859 |  701005 | J124491 | V02 | F | 69 | ... | -0.0203355 |  0.00267189 | -0.01572940 | -0.00156514 |  0.01031670 |  0.04677960 | -0.003931650 | 0 | V02    | F |\n",
       "| 131000655 | 131000655 | A12667 | A12667_V05 | 224585 | 1792295 | J307167 | V01 | F | 67 | ... | -0.0132270 |  0.00583279 | -0.00779501 | -0.01494340 | -0.00611278 |  0.01740740 |  0.002618600 | 0 | Others | F |\n",
       "| 131001064 | 131001064 | A11283 | A11283_V05 | 228563 | 1389627 | J304595 | V02 | M | 70 | ... | -0.0166595 | -0.02797620 | -0.01208170 |  0.00343874 | -0.01595520 |  0.01301230 | -0.028101800 | 0 | V02    | M |\n",
       "| 131001173 | 131001173 | A08339 | A08339_V05 | 229539 |  699670 | J214024 | V02 | F | 84 | ... |  0.0164780 | -0.00476401 |  0.02233330 | -0.01398200 |  0.00148152 | -0.02277820 | -0.000707081 | 1 | V02    | F |\n",
       "| 131001245 | 131001245 | A07099 | A07099_V05 | 224645 |  699409 | J291749 | V02 | F | 69 | ... |  0.0132752 | -0.01870220 | -0.02439060 | -0.00570036 |  0.01178660 |  0.00763499 |  0.007431030 | 0 | V02    | F |\n",
       "\n"
      ],
      "text/plain": [
       "  FID       IID       GWAS_ID GWASID_Visit dbGaP_Subject_ID dbGaP_Sample_ID\n",
       "1 131000053 131000053 A05162  A05162_V05   230296            699030        \n",
       "2 131000094 131000094 A14961  A14961_V05   229859            701005        \n",
       "3 131000655 131000655 A12667  A12667_V05   224585           1792295        \n",
       "4 131001064 131001064 A11283  A11283_V05   228563           1389627        \n",
       "5 131001173 131001173 A08339  A08339_V05   229539            699670        \n",
       "6 131001245 131001245 A07099  A07099_V05   224645            699409        \n",
       "  ARIC_ID Visit.y Gender Age ... PC4        PC5         PC6         PC7        \n",
       "1 J187560 V02     M      72  ... -0.0160964 -0.02318640 -0.00253648 -0.00182851\n",
       "2 J124491 V02     F      69  ... -0.0203355  0.00267189 -0.01572940 -0.00156514\n",
       "3 J307167 V01     F      67  ... -0.0132270  0.00583279 -0.00779501 -0.01494340\n",
       "4 J304595 V02     M      70  ... -0.0166595 -0.02797620 -0.01208170  0.00343874\n",
       "5 J214024 V02     F      84  ...  0.0164780 -0.00476401  0.02233330 -0.01398200\n",
       "6 J291749 V02     F      69  ...  0.0132752 -0.01870220 -0.02439060 -0.00570036\n",
       "  PC8         PC9         PC10         ever_smoke v2_vs_other Sex\n",
       "1 -0.03706380 -0.00436290  0.020383400 1          V02         M  \n",
       "2  0.01031670  0.04677960 -0.003931650 0          V02         F  \n",
       "3 -0.00611278  0.01740740  0.002618600 0          Others      F  \n",
       "4 -0.01595520  0.01301230 -0.028101800 0          V02         M  \n",
       "5  0.00148152 -0.02277820 -0.000707081 1          V02         F  \n",
       "6  0.01178660  0.00763499  0.007431030 0          V02         F  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 38</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>FID</th><th scope=col>IID</th><th scope=col>GWAS_ID</th><th scope=col>GWASID_Visit</th><th scope=col>dbGaP_Subject_ID</th><th scope=col>dbGaP_Sample_ID</th><th scope=col>ARIC_ID</th><th scope=col>Visit.y</th><th scope=col>Gender</th><th scope=col>Age</th><th scope=col>...</th><th scope=col>PC4</th><th scope=col>PC5</th><th scope=col>PC6</th><th scope=col>PC7</th><th scope=col>PC8</th><th scope=col>PC9</th><th scope=col>PC10</th><th scope=col>ever_smoke</th><th scope=col>v2_vs_other</th><th scope=col>Sex</th></tr>\n",
       "\t<tr><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>...</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>131000113</td><td>131000113</td><td>A15057</td><td>A15057_V05</td><td>228789</td><td>1336383</td><td>W293092</td><td>V02</td><td>M</td><td>74</td><td>...</td><td> 0.01125710</td><td>-0.007130800</td><td>-0.006006790</td><td>-2.40823e-05</td><td> 0.007327390</td><td>0.00636429</td><td>-0.00978289</td><td>0</td><td>V02</td><td>M</td></tr>\n",
       "\t<tr><td>131000147</td><td>131000147</td><td>A04826</td><td>A04826_V05</td><td>232712</td><td>1788922</td><td>M127886</td><td>V02</td><td>M</td><td>80</td><td>...</td><td> 0.00337632</td><td>-0.000636593</td><td> 0.005956820</td><td> 3.17560e-03</td><td> 0.005646950</td><td>0.00582919</td><td>-0.00585952</td><td>1</td><td>V02</td><td>M</td></tr>\n",
       "\t<tr><td>131000179</td><td>131000179</td><td>A02787</td><td>A02787_V05</td><td>230453</td><td>1788029</td><td>W235739</td><td>V02</td><td>F</td><td>74</td><td>...</td><td>-0.00218562</td><td> 0.000839117</td><td> 0.000278637</td><td> 4.24331e-03</td><td>-0.005499530</td><td>0.01091180</td><td> 0.01042130</td><td>1</td><td>V02</td><td>F</td></tr>\n",
       "\t<tr><td>131000253</td><td>131000253</td><td>A07971</td><td>A07971_V05</td><td>231733</td><td>1389147</td><td>M288311</td><td>V02</td><td>M</td><td>67</td><td>...</td><td> 0.03580850</td><td>-0.006886160</td><td>-0.003110440</td><td>-2.39232e-03</td><td>-0.017421700</td><td>0.00687928</td><td>-0.00980581</td><td>1</td><td>V02</td><td>M</td></tr>\n",
       "\t<tr><td>131000354</td><td>131000354</td><td>A02264</td><td>A02264_V05</td><td>224837</td><td>1787805</td><td>F300112</td><td>V02</td><td>M</td><td>81</td><td>...</td><td>-0.00323656</td><td>-0.000967097</td><td> 0.003248860</td><td> 1.43675e-03</td><td>-0.000950486</td><td>0.00335151</td><td> 0.00647004</td><td>1</td><td>V02</td><td>M</td></tr>\n",
       "\t<tr><td>131000401</td><td>131000401</td><td>A03618</td><td>A03618_V05</td><td>229698</td><td>1388504</td><td>M229316</td><td>V02</td><td>F</td><td>77</td><td>...</td><td>-0.00509369</td><td>-0.024892200</td><td> 0.001788970</td><td> 6.51412e-03</td><td>-0.000148752</td><td>0.00869312</td><td>-0.00103349</td><td>0</td><td>V02</td><td>F</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 38\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " FID & IID & GWAS\\_ID & GWASID\\_Visit & dbGaP\\_Subject\\_ID & dbGaP\\_Sample\\_ID & ARIC\\_ID & Visit.y & Gender & Age & ... & PC4 & PC5 & PC6 & PC7 & PC8 & PC9 & PC10 & ever\\_smoke & v2\\_vs\\_other & Sex\\\\\n",
       " <int> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <chr> & <int> & ... & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <dbl> & <int> & <chr> & <chr>\\\\\n",
       "\\hline\n",
       "\t 131000113 & 131000113 & A15057 & A15057\\_V05 & 228789 & 1336383 & W293092 & V02 & M & 74 & ... &  0.01125710 & -0.007130800 & -0.006006790 & -2.40823e-05 &  0.007327390 & 0.00636429 & -0.00978289 & 0 & V02 & M\\\\\n",
       "\t 131000147 & 131000147 & A04826 & A04826\\_V05 & 232712 & 1788922 & M127886 & V02 & M & 80 & ... &  0.00337632 & -0.000636593 &  0.005956820 &  3.17560e-03 &  0.005646950 & 0.00582919 & -0.00585952 & 1 & V02 & M\\\\\n",
       "\t 131000179 & 131000179 & A02787 & A02787\\_V05 & 230453 & 1788029 & W235739 & V02 & F & 74 & ... & -0.00218562 &  0.000839117 &  0.000278637 &  4.24331e-03 & -0.005499530 & 0.01091180 &  0.01042130 & 1 & V02 & F\\\\\n",
       "\t 131000253 & 131000253 & A07971 & A07971\\_V05 & 231733 & 1389147 & M288311 & V02 & M & 67 & ... &  0.03580850 & -0.006886160 & -0.003110440 & -2.39232e-03 & -0.017421700 & 0.00687928 & -0.00980581 & 1 & V02 & M\\\\\n",
       "\t 131000354 & 131000354 & A02264 & A02264\\_V05 & 224837 & 1787805 & F300112 & V02 & M & 81 & ... & -0.00323656 & -0.000967097 &  0.003248860 &  1.43675e-03 & -0.000950486 & 0.00335151 &  0.00647004 & 1 & V02 & M\\\\\n",
       "\t 131000401 & 131000401 & A03618 & A03618\\_V05 & 229698 & 1388504 & M229316 & V02 & F & 77 & ... & -0.00509369 & -0.024892200 &  0.001788970 &  6.51412e-03 & -0.000148752 & 0.00869312 & -0.00103349 & 0 & V02 & F\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 38\n",
       "\n",
       "| FID &lt;int&gt; | IID &lt;int&gt; | GWAS_ID &lt;chr&gt; | GWASID_Visit &lt;chr&gt; | dbGaP_Subject_ID &lt;int&gt; | dbGaP_Sample_ID &lt;int&gt; | ARIC_ID &lt;chr&gt; | Visit.y &lt;chr&gt; | Gender &lt;chr&gt; | Age &lt;int&gt; | ... ... | PC4 &lt;dbl&gt; | PC5 &lt;dbl&gt; | PC6 &lt;dbl&gt; | PC7 &lt;dbl&gt; | PC8 &lt;dbl&gt; | PC9 &lt;dbl&gt; | PC10 &lt;dbl&gt; | ever_smoke &lt;int&gt; | v2_vs_other &lt;chr&gt; | Sex &lt;chr&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| 131000113 | 131000113 | A15057 | A15057_V05 | 228789 | 1336383 | W293092 | V02 | M | 74 | ... |  0.01125710 | -0.007130800 | -0.006006790 | -2.40823e-05 |  0.007327390 | 0.00636429 | -0.00978289 | 0 | V02 | M |\n",
       "| 131000147 | 131000147 | A04826 | A04826_V05 | 232712 | 1788922 | M127886 | V02 | M | 80 | ... |  0.00337632 | -0.000636593 |  0.005956820 |  3.17560e-03 |  0.005646950 | 0.00582919 | -0.00585952 | 1 | V02 | M |\n",
       "| 131000179 | 131000179 | A02787 | A02787_V05 | 230453 | 1788029 | W235739 | V02 | F | 74 | ... | -0.00218562 |  0.000839117 |  0.000278637 |  4.24331e-03 | -0.005499530 | 0.01091180 |  0.01042130 | 1 | V02 | F |\n",
       "| 131000253 | 131000253 | A07971 | A07971_V05 | 231733 | 1389147 | M288311 | V02 | M | 67 | ... |  0.03580850 | -0.006886160 | -0.003110440 | -2.39232e-03 | -0.017421700 | 0.00687928 | -0.00980581 | 1 | V02 | M |\n",
       "| 131000354 | 131000354 | A02264 | A02264_V05 | 224837 | 1787805 | F300112 | V02 | M | 81 | ... | -0.00323656 | -0.000967097 |  0.003248860 |  1.43675e-03 | -0.000950486 | 0.00335151 |  0.00647004 | 1 | V02 | M |\n",
       "| 131000401 | 131000401 | A03618 | A03618_V05 | 229698 | 1388504 | M229316 | V02 | F | 77 | ... | -0.00509369 | -0.024892200 |  0.001788970 |  6.51412e-03 | -0.000148752 | 0.00869312 | -0.00103349 | 0 | V02 | F |\n",
       "\n"
      ],
      "text/plain": [
       "  FID       IID       GWAS_ID GWASID_Visit dbGaP_Subject_ID dbGaP_Sample_ID\n",
       "1 131000113 131000113 A15057  A15057_V05   228789           1336383        \n",
       "2 131000147 131000147 A04826  A04826_V05   232712           1788922        \n",
       "3 131000179 131000179 A02787  A02787_V05   230453           1788029        \n",
       "4 131000253 131000253 A07971  A07971_V05   231733           1389147        \n",
       "5 131000354 131000354 A02264  A02264_V05   224837           1787805        \n",
       "6 131000401 131000401 A03618  A03618_V05   229698           1388504        \n",
       "  ARIC_ID Visit.y Gender Age ... PC4         PC5          PC6         \n",
       "1 W293092 V02     M      74  ...  0.01125710 -0.007130800 -0.006006790\n",
       "2 M127886 V02     M      80  ...  0.00337632 -0.000636593  0.005956820\n",
       "3 W235739 V02     F      74  ... -0.00218562  0.000839117  0.000278637\n",
       "4 M288311 V02     M      67  ...  0.03580850 -0.006886160 -0.003110440\n",
       "5 F300112 V02     M      81  ... -0.00323656 -0.000967097  0.003248860\n",
       "6 M229316 V02     F      77  ... -0.00509369 -0.024892200  0.001788970\n",
       "  PC7          PC8          PC9        PC10        ever_smoke v2_vs_other Sex\n",
       "1 -2.40823e-05  0.007327390 0.00636429 -0.00978289 0          V02         M  \n",
       "2  3.17560e-03  0.005646950 0.00582919 -0.00585952 1          V02         M  \n",
       "3  4.24331e-03 -0.005499530 0.01091180  0.01042130 1          V02         F  \n",
       "4 -2.39232e-03 -0.017421700 0.00687928 -0.00980581 1          V02         M  \n",
       "5  1.43675e-03 -0.000950486 0.00335151  0.00647004 1          V02         M  \n",
       "6  6.51412e-03 -0.000148752 0.00869312 -0.00103349 0          V02         F  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "### Prepare data for gwas for growthrate\n",
    "aric_baseline_n_v05_aa_gwas <- fread(\"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/aric_baseline_n_v05_aa_PCA_GWAS.2023Apr3.v2.tsv\", \n",
    "                                     header=T)\n",
    "head(aric_baseline_n_v05_aa_gwas)\n",
    "aric_baseline_n_v05_ea_gwas <- fread(\"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/aric_baseline_n_v05_ea_PCA_GWAS.2023Apr3.v2.tsv\", \n",
    "                                     header=T)\n",
    "head(aric_baseline_n_v05_ea_gwas)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "###\n",
    "cln_grt.vaf2.DP20_base.corrected_ordered <- fread(\"/medpop/esp2/mesbah/projects/ch_progression/aric/pheno/cln_grt.vaf2.DP20_base.corrected_ordered.01Dec2023.csv\", \n",
    "                                                  header = T)\n",
    "\n",
    "## \n",
    "cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered <- fread(\"/medpop/esp2/mesbah/projects/ch_progression/aric/pheno/cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered.01Dec2023.csv\", \n",
    "                                                              header=T)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CHROM_POS_GWAS_VISIT_base'</li><li>'ARIC_ID'</li><li>'varID_GWASID'</li><li>'GWAS_ID.x'</li><li>'CHR'</li><li>'POS'</li><li>'REF'</li><li>'ALT'</li><li>'Gene'</li><li>'AD.v5'</li><li>'VAF.v5'</li><li>'DP.v5'</li><li>'AD.Alt.v5'</li><li>'FR.Alt.v5'</li><li>'RR.Alt.v5'</li><li>'Accession'</li><li>'transcriptOI'</li><li>'NonsynOI'</li><li>'Func.refGene'</li><li>'ExonicFunc.refGene'</li><li>'AAChange.refGene'</li><li>'AD.v2'</li><li>'VAF.v2'</li><li>'DP.v2'</li><li>'AD.Alt.v2'</li><li>'FR.Alt.v2'</li><li>'RR.Alt.v2'</li><li>'Batch'</li><li>'dAge'</li><li>'dVAF'</li><li>'pctVAF'</li><li>'growth_rate'</li><li>'log_growth_rate'</li><li>'Gene_Group'</li><li>'Gene_Group_2'</li><li>'Clone_status'</li><li>'Trajectory'</li><li>'GWAS_ID.y'</li><li>'GWASID_Visit'</li><li>'cram_id'</li><li>'Visit.x'</li><li>'Gender'</li><li>'Age'</li><li>'CH_baseline'</li><li>'CH_v05'</li><li>'incident_CH'</li><li>'incident_CHvaf05'</li><li>'incident_CHvaf10'</li><li>'incident_DNMT3A'</li><li>'incident_TET2'</li><li>'incident_ASXL1'</li><li>'incident_DTA'</li><li>'incident_SF'</li><li>'incident_DDR'</li><li>'death.x'</li><li>'death_date.x'</li><li>'hg38_CRAM_ID'</li><li>'aricid'</li><li>'Visit.y'</li><li>'sex'</li><li>'race'</li><li>'center'</li><li>'dob'</li><li>'date_base'</li><li>'age_base'</li><li>'fast_8_base'</li><li>'bmi_base'</li><li>'dm_126_base'</li><li>'gluc_base'</li><li>'htn_5_base'</li><li>'chol_med_base'</li><li>'statin_base'</li><li>'cig_base'</li><li>'chd_date'</li><li>'is_date'</li><li>'hs_date'</li><li>'hf_date'</li><li>'death.y'</li><li>'death_date.y'</li><li>'lfu_date'</li><li>'chd_base'</li><li>'is_base'</li><li>'hs_base'</li><li>'hf_base'</li><li>'ever_smoke'</li><li>'v2_vs_other'</li><li>'Sex'</li><li>'race_BW'</li><li>'BMI_cat'</li><li>'nonHDL_std'</li><li>'hdl_base_std'</li><li>'ldl_base_std'</li><li>'chol_base_std'</li><li>'tg_base_std'</li><li>'gluc_base_std'</li><li>'bmi_base_std'</li><li>'CHvaf10_baseline'</li><li>'CH_baseline_v05'</li><li>'CHvaf10_baseline_v05'</li><li>'CH_DTA'</li><li>'CH_SF'</li><li>'CH_DDR'</li><li>'CH_DNMT3A'</li><li>'CH_TET2'</li><li>'CH_ASXL1'</li><li>'nonHDL_base_std'</li><li>'age_base_sqr'</li><li>'Center'</li><li>'sbp_base'</li><li>'dbp_base'</li><li>'hem_can_base'</li><li>'hem_can_inc_base'</li><li>'hem_can_int_base'</li><li>'hem_can_inc_v5'</li><li>'hem_can_int_v5'</li><li>'chol_base'</li><li>'hdl_base'</li><li>'ldl_base'</li><li>'tg_base'</li><li>'chd_is_base'</li><li>'chol_base_INT'</li><li>'ldl_base_INT'</li><li>'hdl_base_INT'</li><li>'tg_base_INT'</li><li>'nonHDL_base'</li><li>'nonHDL_base_INT'</li><li>'bmi_base_INT'</li><li>'tg_to_hdl_base'</li><li>'tg_to_hdl_base_INT'</li><li>'ldl_base_nomal_vs_high'</li><li>'hdl_base_low'</li><li>'Dyslipidemia'</li><li>'sex_v5'</li><li>'race_v5'</li><li>'center_v5'</li><li>'dob_v5'</li><li>'date_v5'</li><li>'age_v5'</li><li>'fast_8_v5'</li><li>'bmi_v5'</li><li>'dm_126_v5'</li><li>'gluc_v5'</li><li>'htn_5_v5'</li><li>'chol_v5'</li><li>'hdl_v5'</li><li>'ldl_v5'</li><li>'tg_v5'</li><li>'chol_med_v5'</li><li>'statin_v5'</li><li>'cig_v5'</li><li>'chd_v5'</li><li>'is_v5'</li><li>'hs_v5'</li><li>'hf_v5'</li><li>'CHROM_POS'</li><li>'CHROM_POS_GWAS_VISIT_v5'</li><li>'HiSeq_CRAM_ID'</li><li>'HiSeq_Batch'</li><li>'HiSeq_GWASID_Visit'</li><li>'HiSeq_GWAS_ID'</li><li>'HiSeq_Visit'</li><li>'HiSeq_CHROM'</li><li>'HiSeq_POS'</li><li>'HiSeq_varID'</li><li>'HiSeq_REF'</li><li>'HiSeq_ALT'</li><li>'HiSeq_INFO'</li><li>'HiSeq_ADF'</li><li>'HiSeq_ADR'</li><li>'HiSeq_DP'</li><li>'HiSeq_Sample'</li><li>'HiSeq_ADF_Ref'</li><li>'HiSeq_ADF_Alt'</li><li>'HiSeq_ADR_Ref'</li><li>'HiSeq_ADR_Alt'</li><li>'HiSeq_AD.Alt'</li><li>'HiSeq_VAF'</li><li>'HiSeq_CHROM_POS'</li><li>'HiSeq_Pileup'</li><li>'varID_Match'</li><li>'Imputed_VAF_v2'</li><li>'CH_method'</li><li>'dVAF_by_dT'</li><li>'logdVAF_by_dT'</li><li>'Variant_type'</li><li>'Gene_Group_R882'</li><li>'Gene_R882'</li><li>'is_notHiSeq'</li><li>'is_notMUTECT'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CHROM\\_POS\\_GWAS\\_VISIT\\_base'\n",
       "\\item 'ARIC\\_ID'\n",
       "\\item 'varID\\_GWASID'\n",
       "\\item 'GWAS\\_ID.x'\n",
       "\\item 'CHR'\n",
       "\\item 'POS'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'Gene'\n",
       "\\item 'AD.v5'\n",
       "\\item 'VAF.v5'\n",
       "\\item 'DP.v5'\n",
       "\\item 'AD.Alt.v5'\n",
       "\\item 'FR.Alt.v5'\n",
       "\\item 'RR.Alt.v5'\n",
       "\\item 'Accession'\n",
       "\\item 'transcriptOI'\n",
       "\\item 'NonsynOI'\n",
       "\\item 'Func.refGene'\n",
       "\\item 'ExonicFunc.refGene'\n",
       "\\item 'AAChange.refGene'\n",
       "\\item 'AD.v2'\n",
       "\\item 'VAF.v2'\n",
       "\\item 'DP.v2'\n",
       "\\item 'AD.Alt.v2'\n",
       "\\item 'FR.Alt.v2'\n",
       "\\item 'RR.Alt.v2'\n",
       "\\item 'Batch'\n",
       "\\item 'dAge'\n",
       "\\item 'dVAF'\n",
       "\\item 'pctVAF'\n",
       "\\item 'growth\\_rate'\n",
       "\\item 'log\\_growth\\_rate'\n",
       "\\item 'Gene\\_Group'\n",
       "\\item 'Gene\\_Group\\_2'\n",
       "\\item 'Clone\\_status'\n",
       "\\item 'Trajectory'\n",
       "\\item 'GWAS\\_ID.y'\n",
       "\\item 'GWASID\\_Visit'\n",
       "\\item 'cram\\_id'\n",
       "\\item 'Visit.x'\n",
       "\\item 'Gender'\n",
       "\\item 'Age'\n",
       "\\item 'CH\\_baseline'\n",
       "\\item 'CH\\_v05'\n",
       "\\item 'incident\\_CH'\n",
       "\\item 'incident\\_CHvaf05'\n",
       "\\item 'incident\\_CHvaf10'\n",
       "\\item 'incident\\_DNMT3A'\n",
       "\\item 'incident\\_TET2'\n",
       "\\item 'incident\\_ASXL1'\n",
       "\\item 'incident\\_DTA'\n",
       "\\item 'incident\\_SF'\n",
       "\\item 'incident\\_DDR'\n",
       "\\item 'death.x'\n",
       "\\item 'death\\_date.x'\n",
       "\\item 'hg38\\_CRAM\\_ID'\n",
       "\\item 'aricid'\n",
       "\\item 'Visit.y'\n",
       "\\item 'sex'\n",
       "\\item 'race'\n",
       "\\item 'center'\n",
       "\\item 'dob'\n",
       "\\item 'date\\_base'\n",
       "\\item 'age\\_base'\n",
       "\\item 'fast\\_8\\_base'\n",
       "\\item 'bmi\\_base'\n",
       "\\item 'dm\\_126\\_base'\n",
       "\\item 'gluc\\_base'\n",
       "\\item 'htn\\_5\\_base'\n",
       "\\item 'chol\\_med\\_base'\n",
       "\\item 'statin\\_base'\n",
       "\\item 'cig\\_base'\n",
       "\\item 'chd\\_date'\n",
       "\\item 'is\\_date'\n",
       "\\item 'hs\\_date'\n",
       "\\item 'hf\\_date'\n",
       "\\item 'death.y'\n",
       "\\item 'death\\_date.y'\n",
       "\\item 'lfu\\_date'\n",
       "\\item 'chd\\_base'\n",
       "\\item 'is\\_base'\n",
       "\\item 'hs\\_base'\n",
       "\\item 'hf\\_base'\n",
       "\\item 'ever\\_smoke'\n",
       "\\item 'v2\\_vs\\_other'\n",
       "\\item 'Sex'\n",
       "\\item 'race\\_BW'\n",
       "\\item 'BMI\\_cat'\n",
       "\\item 'nonHDL\\_std'\n",
       "\\item 'hdl\\_base\\_std'\n",
       "\\item 'ldl\\_base\\_std'\n",
       "\\item 'chol\\_base\\_std'\n",
       "\\item 'tg\\_base\\_std'\n",
       "\\item 'gluc\\_base\\_std'\n",
       "\\item 'bmi\\_base\\_std'\n",
       "\\item 'CHvaf10\\_baseline'\n",
       "\\item 'CH\\_baseline\\_v05'\n",
       "\\item 'CHvaf10\\_baseline\\_v05'\n",
       "\\item 'CH\\_DTA'\n",
       "\\item 'CH\\_SF'\n",
       "\\item 'CH\\_DDR'\n",
       "\\item 'CH\\_DNMT3A'\n",
       "\\item 'CH\\_TET2'\n",
       "\\item 'CH\\_ASXL1'\n",
       "\\item 'nonHDL\\_base\\_std'\n",
       "\\item 'age\\_base\\_sqr'\n",
       "\\item 'Center'\n",
       "\\item 'sbp\\_base'\n",
       "\\item 'dbp\\_base'\n",
       "\\item 'hem\\_can\\_base'\n",
       "\\item 'hem\\_can\\_inc\\_base'\n",
       "\\item 'hem\\_can\\_int\\_base'\n",
       "\\item 'hem\\_can\\_inc\\_v5'\n",
       "\\item 'hem\\_can\\_int\\_v5'\n",
       "\\item 'chol\\_base'\n",
       "\\item 'hdl\\_base'\n",
       "\\item 'ldl\\_base'\n",
       "\\item 'tg\\_base'\n",
       "\\item 'chd\\_is\\_base'\n",
       "\\item 'chol\\_base\\_INT'\n",
       "\\item 'ldl\\_base\\_INT'\n",
       "\\item 'hdl\\_base\\_INT'\n",
       "\\item 'tg\\_base\\_INT'\n",
       "\\item 'nonHDL\\_base'\n",
       "\\item 'nonHDL\\_base\\_INT'\n",
       "\\item 'bmi\\_base\\_INT'\n",
       "\\item 'tg\\_to\\_hdl\\_base'\n",
       "\\item 'tg\\_to\\_hdl\\_base\\_INT'\n",
       "\\item 'ldl\\_base\\_nomal\\_vs\\_high'\n",
       "\\item 'hdl\\_base\\_low'\n",
       "\\item 'Dyslipidemia'\n",
       "\\item 'sex\\_v5'\n",
       "\\item 'race\\_v5'\n",
       "\\item 'center\\_v5'\n",
       "\\item 'dob\\_v5'\n",
       "\\item 'date\\_v5'\n",
       "\\item 'age\\_v5'\n",
       "\\item 'fast\\_8\\_v5'\n",
       "\\item 'bmi\\_v5'\n",
       "\\item 'dm\\_126\\_v5'\n",
       "\\item 'gluc\\_v5'\n",
       "\\item 'htn\\_5\\_v5'\n",
       "\\item 'chol\\_v5'\n",
       "\\item 'hdl\\_v5'\n",
       "\\item 'ldl\\_v5'\n",
       "\\item 'tg\\_v5'\n",
       "\\item 'chol\\_med\\_v5'\n",
       "\\item 'statin\\_v5'\n",
       "\\item 'cig\\_v5'\n",
       "\\item 'chd\\_v5'\n",
       "\\item 'is\\_v5'\n",
       "\\item 'hs\\_v5'\n",
       "\\item 'hf\\_v5'\n",
       "\\item 'CHROM\\_POS'\n",
       "\\item 'CHROM\\_POS\\_GWAS\\_VISIT\\_v5'\n",
       "\\item 'HiSeq\\_CRAM\\_ID'\n",
       "\\item 'HiSeq\\_Batch'\n",
       "\\item 'HiSeq\\_GWASID\\_Visit'\n",
       "\\item 'HiSeq\\_GWAS\\_ID'\n",
       "\\item 'HiSeq\\_Visit'\n",
       "\\item 'HiSeq\\_CHROM'\n",
       "\\item 'HiSeq\\_POS'\n",
       "\\item 'HiSeq\\_varID'\n",
       "\\item 'HiSeq\\_REF'\n",
       "\\item 'HiSeq\\_ALT'\n",
       "\\item 'HiSeq\\_INFO'\n",
       "\\item 'HiSeq\\_ADF'\n",
       "\\item 'HiSeq\\_ADR'\n",
       "\\item 'HiSeq\\_DP'\n",
       "\\item 'HiSeq\\_Sample'\n",
       "\\item 'HiSeq\\_ADF\\_Ref'\n",
       "\\item 'HiSeq\\_ADF\\_Alt'\n",
       "\\item 'HiSeq\\_ADR\\_Ref'\n",
       "\\item 'HiSeq\\_ADR\\_Alt'\n",
       "\\item 'HiSeq\\_AD.Alt'\n",
       "\\item 'HiSeq\\_VAF'\n",
       "\\item 'HiSeq\\_CHROM\\_POS'\n",
       "\\item 'HiSeq\\_Pileup'\n",
       "\\item 'varID\\_Match'\n",
       "\\item 'Imputed\\_VAF\\_v2'\n",
       "\\item 'CH\\_method'\n",
       "\\item 'dVAF\\_by\\_dT'\n",
       "\\item 'logdVAF\\_by\\_dT'\n",
       "\\item 'Variant\\_type'\n",
       "\\item 'Gene\\_Group\\_R882'\n",
       "\\item 'Gene\\_R882'\n",
       "\\item 'is\\_notHiSeq'\n",
       "\\item 'is\\_notMUTECT'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CHROM_POS_GWAS_VISIT_base'\n",
       "2. 'ARIC_ID'\n",
       "3. 'varID_GWASID'\n",
       "4. 'GWAS_ID.x'\n",
       "5. 'CHR'\n",
       "6. 'POS'\n",
       "7. 'REF'\n",
       "8. 'ALT'\n",
       "9. 'Gene'\n",
       "10. 'AD.v5'\n",
       "11. 'VAF.v5'\n",
       "12. 'DP.v5'\n",
       "13. 'AD.Alt.v5'\n",
       "14. 'FR.Alt.v5'\n",
       "15. 'RR.Alt.v5'\n",
       "16. 'Accession'\n",
       "17. 'transcriptOI'\n",
       "18. 'NonsynOI'\n",
       "19. 'Func.refGene'\n",
       "20. 'ExonicFunc.refGene'\n",
       "21. 'AAChange.refGene'\n",
       "22. 'AD.v2'\n",
       "23. 'VAF.v2'\n",
       "24. 'DP.v2'\n",
       "25. 'AD.Alt.v2'\n",
       "26. 'FR.Alt.v2'\n",
       "27. 'RR.Alt.v2'\n",
       "28. 'Batch'\n",
       "29. 'dAge'\n",
       "30. 'dVAF'\n",
       "31. 'pctVAF'\n",
       "32. 'growth_rate'\n",
       "33. 'log_growth_rate'\n",
       "34. 'Gene_Group'\n",
       "35. 'Gene_Group_2'\n",
       "36. 'Clone_status'\n",
       "37. 'Trajectory'\n",
       "38. 'GWAS_ID.y'\n",
       "39. 'GWASID_Visit'\n",
       "40. 'cram_id'\n",
       "41. 'Visit.x'\n",
       "42. 'Gender'\n",
       "43. 'Age'\n",
       "44. 'CH_baseline'\n",
       "45. 'CH_v05'\n",
       "46. 'incident_CH'\n",
       "47. 'incident_CHvaf05'\n",
       "48. 'incident_CHvaf10'\n",
       "49. 'incident_DNMT3A'\n",
       "50. 'incident_TET2'\n",
       "51. 'incident_ASXL1'\n",
       "52. 'incident_DTA'\n",
       "53. 'incident_SF'\n",
       "54. 'incident_DDR'\n",
       "55. 'death.x'\n",
       "56. 'death_date.x'\n",
       "57. 'hg38_CRAM_ID'\n",
       "58. 'aricid'\n",
       "59. 'Visit.y'\n",
       "60. 'sex'\n",
       "61. 'race'\n",
       "62. 'center'\n",
       "63. 'dob'\n",
       "64. 'date_base'\n",
       "65. 'age_base'\n",
       "66. 'fast_8_base'\n",
       "67. 'bmi_base'\n",
       "68. 'dm_126_base'\n",
       "69. 'gluc_base'\n",
       "70. 'htn_5_base'\n",
       "71. 'chol_med_base'\n",
       "72. 'statin_base'\n",
       "73. 'cig_base'\n",
       "74. 'chd_date'\n",
       "75. 'is_date'\n",
       "76. 'hs_date'\n",
       "77. 'hf_date'\n",
       "78. 'death.y'\n",
       "79. 'death_date.y'\n",
       "80. 'lfu_date'\n",
       "81. 'chd_base'\n",
       "82. 'is_base'\n",
       "83. 'hs_base'\n",
       "84. 'hf_base'\n",
       "85. 'ever_smoke'\n",
       "86. 'v2_vs_other'\n",
       "87. 'Sex'\n",
       "88. 'race_BW'\n",
       "89. 'BMI_cat'\n",
       "90. 'nonHDL_std'\n",
       "91. 'hdl_base_std'\n",
       "92. 'ldl_base_std'\n",
       "93. 'chol_base_std'\n",
       "94. 'tg_base_std'\n",
       "95. 'gluc_base_std'\n",
       "96. 'bmi_base_std'\n",
       "97. 'CHvaf10_baseline'\n",
       "98. 'CH_baseline_v05'\n",
       "99. 'CHvaf10_baseline_v05'\n",
       "100. 'CH_DTA'\n",
       "101. 'CH_SF'\n",
       "102. 'CH_DDR'\n",
       "103. 'CH_DNMT3A'\n",
       "104. 'CH_TET2'\n",
       "105. 'CH_ASXL1'\n",
       "106. 'nonHDL_base_std'\n",
       "107. 'age_base_sqr'\n",
       "108. 'Center'\n",
       "109. 'sbp_base'\n",
       "110. 'dbp_base'\n",
       "111. 'hem_can_base'\n",
       "112. 'hem_can_inc_base'\n",
       "113. 'hem_can_int_base'\n",
       "114. 'hem_can_inc_v5'\n",
       "115. 'hem_can_int_v5'\n",
       "116. 'chol_base'\n",
       "117. 'hdl_base'\n",
       "118. 'ldl_base'\n",
       "119. 'tg_base'\n",
       "120. 'chd_is_base'\n",
       "121. 'chol_base_INT'\n",
       "122. 'ldl_base_INT'\n",
       "123. 'hdl_base_INT'\n",
       "124. 'tg_base_INT'\n",
       "125. 'nonHDL_base'\n",
       "126. 'nonHDL_base_INT'\n",
       "127. 'bmi_base_INT'\n",
       "128. 'tg_to_hdl_base'\n",
       "129. 'tg_to_hdl_base_INT'\n",
       "130. 'ldl_base_nomal_vs_high'\n",
       "131. 'hdl_base_low'\n",
       "132. 'Dyslipidemia'\n",
       "133. 'sex_v5'\n",
       "134. 'race_v5'\n",
       "135. 'center_v5'\n",
       "136. 'dob_v5'\n",
       "137. 'date_v5'\n",
       "138. 'age_v5'\n",
       "139. 'fast_8_v5'\n",
       "140. 'bmi_v5'\n",
       "141. 'dm_126_v5'\n",
       "142. 'gluc_v5'\n",
       "143. 'htn_5_v5'\n",
       "144. 'chol_v5'\n",
       "145. 'hdl_v5'\n",
       "146. 'ldl_v5'\n",
       "147. 'tg_v5'\n",
       "148. 'chol_med_v5'\n",
       "149. 'statin_v5'\n",
       "150. 'cig_v5'\n",
       "151. 'chd_v5'\n",
       "152. 'is_v5'\n",
       "153. 'hs_v5'\n",
       "154. 'hf_v5'\n",
       "155. 'CHROM_POS'\n",
       "156. 'CHROM_POS_GWAS_VISIT_v5'\n",
       "157. 'HiSeq_CRAM_ID'\n",
       "158. 'HiSeq_Batch'\n",
       "159. 'HiSeq_GWASID_Visit'\n",
       "160. 'HiSeq_GWAS_ID'\n",
       "161. 'HiSeq_Visit'\n",
       "162. 'HiSeq_CHROM'\n",
       "163. 'HiSeq_POS'\n",
       "164. 'HiSeq_varID'\n",
       "165. 'HiSeq_REF'\n",
       "166. 'HiSeq_ALT'\n",
       "167. 'HiSeq_INFO'\n",
       "168. 'HiSeq_ADF'\n",
       "169. 'HiSeq_ADR'\n",
       "170. 'HiSeq_DP'\n",
       "171. 'HiSeq_Sample'\n",
       "172. 'HiSeq_ADF_Ref'\n",
       "173. 'HiSeq_ADF_Alt'\n",
       "174. 'HiSeq_ADR_Ref'\n",
       "175. 'HiSeq_ADR_Alt'\n",
       "176. 'HiSeq_AD.Alt'\n",
       "177. 'HiSeq_VAF'\n",
       "178. 'HiSeq_CHROM_POS'\n",
       "179. 'HiSeq_Pileup'\n",
       "180. 'varID_Match'\n",
       "181. 'Imputed_VAF_v2'\n",
       "182. 'CH_method'\n",
       "183. 'dVAF_by_dT'\n",
       "184. 'logdVAF_by_dT'\n",
       "185. 'Variant_type'\n",
       "186. 'Gene_Group_R882'\n",
       "187. 'Gene_R882'\n",
       "188. 'is_notHiSeq'\n",
       "189. 'is_notMUTECT'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"CHROM_POS_GWAS_VISIT_base\" \"ARIC_ID\"                  \n",
       "  [3] \"varID_GWASID\"              \"GWAS_ID.x\"                \n",
       "  [5] \"CHR\"                       \"POS\"                      \n",
       "  [7] \"REF\"                       \"ALT\"                      \n",
       "  [9] \"Gene\"                      \"AD.v5\"                    \n",
       " [11] \"VAF.v5\"                    \"DP.v5\"                    \n",
       " [13] \"AD.Alt.v5\"                 \"FR.Alt.v5\"                \n",
       " [15] \"RR.Alt.v5\"                 \"Accession\"                \n",
       " [17] \"transcriptOI\"              \"NonsynOI\"                 \n",
       " [19] \"Func.refGene\"              \"ExonicFunc.refGene\"       \n",
       " [21] \"AAChange.refGene\"          \"AD.v2\"                    \n",
       " [23] \"VAF.v2\"                    \"DP.v2\"                    \n",
       " [25] \"AD.Alt.v2\"                 \"FR.Alt.v2\"                \n",
       " [27] \"RR.Alt.v2\"                 \"Batch\"                    \n",
       " [29] \"dAge\"                      \"dVAF\"                     \n",
       " [31] \"pctVAF\"                    \"growth_rate\"              \n",
       " [33] \"log_growth_rate\"           \"Gene_Group\"               \n",
       " [35] \"Gene_Group_2\"              \"Clone_status\"             \n",
       " [37] \"Trajectory\"                \"GWAS_ID.y\"                \n",
       " [39] \"GWASID_Visit\"              \"cram_id\"                  \n",
       " [41] \"Visit.x\"                   \"Gender\"                   \n",
       " [43] \"Age\"                       \"CH_baseline\"              \n",
       " [45] \"CH_v05\"                    \"incident_CH\"              \n",
       " [47] \"incident_CHvaf05\"          \"incident_CHvaf10\"         \n",
       " [49] \"incident_DNMT3A\"           \"incident_TET2\"            \n",
       " [51] \"incident_ASXL1\"            \"incident_DTA\"             \n",
       " [53] \"incident_SF\"               \"incident_DDR\"             \n",
       " [55] \"death.x\"                   \"death_date.x\"             \n",
       " [57] \"hg38_CRAM_ID\"              \"aricid\"                   \n",
       " [59] \"Visit.y\"                   \"sex\"                      \n",
       " [61] \"race\"                      \"center\"                   \n",
       " [63] \"dob\"                       \"date_base\"                \n",
       " [65] \"age_base\"                  \"fast_8_base\"              \n",
       " [67] \"bmi_base\"                  \"dm_126_base\"              \n",
       " [69] \"gluc_base\"                 \"htn_5_base\"               \n",
       " [71] \"chol_med_base\"             \"statin_base\"              \n",
       " [73] \"cig_base\"                  \"chd_date\"                 \n",
       " [75] \"is_date\"                   \"hs_date\"                  \n",
       " [77] \"hf_date\"                   \"death.y\"                  \n",
       " [79] \"death_date.y\"              \"lfu_date\"                 \n",
       " [81] \"chd_base\"                  \"is_base\"                  \n",
       " [83] \"hs_base\"                   \"hf_base\"                  \n",
       " [85] \"ever_smoke\"                \"v2_vs_other\"              \n",
       " [87] \"Sex\"                       \"race_BW\"                  \n",
       " [89] \"BMI_cat\"                   \"nonHDL_std\"               \n",
       " [91] \"hdl_base_std\"              \"ldl_base_std\"             \n",
       " [93] \"chol_base_std\"             \"tg_base_std\"              \n",
       " [95] \"gluc_base_std\"             \"bmi_base_std\"             \n",
       " [97] \"CHvaf10_baseline\"          \"CH_baseline_v05\"          \n",
       " [99] \"CHvaf10_baseline_v05\"      \"CH_DTA\"                   \n",
       "[101] \"CH_SF\"                     \"CH_DDR\"                   \n",
       "[103] \"CH_DNMT3A\"                 \"CH_TET2\"                  \n",
       "[105] \"CH_ASXL1\"                  \"nonHDL_base_std\"          \n",
       "[107] \"age_base_sqr\"              \"Center\"                   \n",
       "[109] \"sbp_base\"                  \"dbp_base\"                 \n",
       "[111] \"hem_can_base\"              \"hem_can_inc_base\"         \n",
       "[113] \"hem_can_int_base\"          \"hem_can_inc_v5\"           \n",
       "[115] \"hem_can_int_v5\"            \"chol_base\"                \n",
       "[117] \"hdl_base\"                  \"ldl_base\"                 \n",
       "[119] \"tg_base\"                   \"chd_is_base\"              \n",
       "[121] \"chol_base_INT\"             \"ldl_base_INT\"             \n",
       "[123] \"hdl_base_INT\"              \"tg_base_INT\"              \n",
       "[125] \"nonHDL_base\"               \"nonHDL_base_INT\"          \n",
       "[127] \"bmi_base_INT\"              \"tg_to_hdl_base\"           \n",
       "[129] \"tg_to_hdl_base_INT\"        \"ldl_base_nomal_vs_high\"   \n",
       "[131] \"hdl_base_low\"              \"Dyslipidemia\"             \n",
       "[133] \"sex_v5\"                    \"race_v5\"                  \n",
       "[135] \"center_v5\"                 \"dob_v5\"                   \n",
       "[137] \"date_v5\"                   \"age_v5\"                   \n",
       "[139] \"fast_8_v5\"                 \"bmi_v5\"                   \n",
       "[141] \"dm_126_v5\"                 \"gluc_v5\"                  \n",
       "[143] \"htn_5_v5\"                  \"chol_v5\"                  \n",
       "[145] \"hdl_v5\"                    \"ldl_v5\"                   \n",
       "[147] \"tg_v5\"                     \"chol_med_v5\"              \n",
       "[149] \"statin_v5\"                 \"cig_v5\"                   \n",
       "[151] \"chd_v5\"                    \"is_v5\"                    \n",
       "[153] \"hs_v5\"                     \"hf_v5\"                    \n",
       "[155] \"CHROM_POS\"                 \"CHROM_POS_GWAS_VISIT_v5\"  \n",
       "[157] \"HiSeq_CRAM_ID\"             \"HiSeq_Batch\"              \n",
       "[159] \"HiSeq_GWASID_Visit\"        \"HiSeq_GWAS_ID\"            \n",
       "[161] \"HiSeq_Visit\"               \"HiSeq_CHROM\"              \n",
       "[163] \"HiSeq_POS\"                 \"HiSeq_varID\"              \n",
       "[165] \"HiSeq_REF\"                 \"HiSeq_ALT\"                \n",
       "[167] \"HiSeq_INFO\"                \"HiSeq_ADF\"                \n",
       "[169] \"HiSeq_ADR\"                 \"HiSeq_DP\"                 \n",
       "[171] \"HiSeq_Sample\"              \"HiSeq_ADF_Ref\"            \n",
       "[173] \"HiSeq_ADF_Alt\"             \"HiSeq_ADR_Ref\"            \n",
       "[175] \"HiSeq_ADR_Alt\"             \"HiSeq_AD.Alt\"             \n",
       "[177] \"HiSeq_VAF\"                 \"HiSeq_CHROM_POS\"          \n",
       "[179] \"HiSeq_Pileup\"              \"varID_Match\"              \n",
       "[181] \"Imputed_VAF_v2\"            \"CH_method\"                \n",
       "[183] \"dVAF_by_dT\"                \"logdVAF_by_dT\"            \n",
       "[185] \"Variant_type\"              \"Gene_Group_R882\"          \n",
       "[187] \"Gene_R882\"                 \"is_notHiSeq\"              \n",
       "[189] \"is_notMUTECT\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'CHROM_POS_GWAS_VISIT_base'</li><li>'ARIC_ID'</li><li>'varID_GWASID'</li><li>'GWAS_ID.x'</li><li>'CHR'</li><li>'POS'</li><li>'REF'</li><li>'ALT'</li><li>'Gene'</li><li>'AD.v5'</li><li>'VAF.v5'</li><li>'DP.v5'</li><li>'AD.Alt.v5'</li><li>'FR.Alt.v5'</li><li>'RR.Alt.v5'</li><li>'Accession'</li><li>'transcriptOI'</li><li>'NonsynOI'</li><li>'Func.refGene'</li><li>'ExonicFunc.refGene'</li><li>'AAChange.refGene'</li><li>'AD.v2'</li><li>'VAF.v2'</li><li>'DP.v2'</li><li>'AD.Alt.v2'</li><li>'FR.Alt.v2'</li><li>'RR.Alt.v2'</li><li>'Batch'</li><li>'dAge'</li><li>'dVAF'</li><li>'pctVAF'</li><li>'growth_rate'</li><li>'log_growth_rate'</li><li>'Gene_Group'</li><li>'Gene_Group_2'</li><li>'Clone_status'</li><li>'Trajectory'</li><li>'GWAS_ID.y'</li><li>'GWASID_Visit'</li><li>'cram_id'</li><li>'Visit.x'</li><li>'Gender'</li><li>'Age'</li><li>'CH_baseline'</li><li>'CH_v05'</li><li>'incident_CH'</li><li>'incident_CHvaf05'</li><li>'incident_CHvaf10'</li><li>'incident_DNMT3A'</li><li>'incident_TET2'</li><li>'incident_ASXL1'</li><li>'incident_DTA'</li><li>'incident_SF'</li><li>'incident_DDR'</li><li>'death.x'</li><li>'death_date.x'</li><li>'hg38_CRAM_ID'</li><li>'aricid'</li><li>'Visit.y'</li><li>'sex'</li><li>'race'</li><li>'center'</li><li>'dob'</li><li>'date_base'</li><li>'age_base'</li><li>'fast_8_base'</li><li>'bmi_base'</li><li>'dm_126_base'</li><li>'gluc_base'</li><li>'htn_5_base'</li><li>'chol_med_base'</li><li>'statin_base'</li><li>'cig_base'</li><li>'chd_date'</li><li>'is_date'</li><li>'hs_date'</li><li>'hf_date'</li><li>'death.y'</li><li>'death_date.y'</li><li>'lfu_date'</li><li>'chd_base'</li><li>'is_base'</li><li>'hs_base'</li><li>'hf_base'</li><li>'ever_smoke'</li><li>'v2_vs_other'</li><li>'Sex'</li><li>'race_BW'</li><li>'BMI_cat'</li><li>'nonHDL_std'</li><li>'hdl_base_std'</li><li>'ldl_base_std'</li><li>'chol_base_std'</li><li>'tg_base_std'</li><li>'gluc_base_std'</li><li>'bmi_base_std'</li><li>'CHvaf10_baseline'</li><li>'CH_baseline_v05'</li><li>'CHvaf10_baseline_v05'</li><li>'CH_DTA'</li><li>'CH_SF'</li><li>'CH_DDR'</li><li>'CH_DNMT3A'</li><li>'CH_TET2'</li><li>'CH_ASXL1'</li><li>'nonHDL_base_std'</li><li>'age_base_sqr'</li><li>'Center'</li><li>'sbp_base'</li><li>'dbp_base'</li><li>'hem_can_base'</li><li>'hem_can_inc_base'</li><li>'hem_can_int_base'</li><li>'hem_can_inc_v5'</li><li>'hem_can_int_v5'</li><li>'chol_base'</li><li>'hdl_base'</li><li>'ldl_base'</li><li>'tg_base'</li><li>'chd_is_base'</li><li>'chol_base_INT'</li><li>'ldl_base_INT'</li><li>'hdl_base_INT'</li><li>'tg_base_INT'</li><li>'nonHDL_base'</li><li>'nonHDL_base_INT'</li><li>'bmi_base_INT'</li><li>'tg_to_hdl_base'</li><li>'tg_to_hdl_base_INT'</li><li>'ldl_base_nomal_vs_high'</li><li>'hdl_base_low'</li><li>'Dyslipidemia'</li><li>'sex_v5'</li><li>'race_v5'</li><li>'center_v5'</li><li>'dob_v5'</li><li>'date_v5'</li><li>'age_v5'</li><li>'fast_8_v5'</li><li>'bmi_v5'</li><li>'dm_126_v5'</li><li>'gluc_v5'</li><li>'htn_5_v5'</li><li>'chol_v5'</li><li>'hdl_v5'</li><li>'ldl_v5'</li><li>'tg_v5'</li><li>'chol_med_v5'</li><li>'statin_v5'</li><li>'cig_v5'</li><li>'chd_v5'</li><li>'is_v5'</li><li>'hs_v5'</li><li>'hf_v5'</li><li>'CHROM_POS'</li><li>'CHROM_POS_GWAS_VISIT_v5'</li><li>'HiSeq_CRAM_ID'</li><li>'HiSeq_Batch'</li><li>'HiSeq_GWASID_Visit'</li><li>'HiSeq_GWAS_ID'</li><li>'HiSeq_Visit'</li><li>'HiSeq_CHROM'</li><li>'HiSeq_POS'</li><li>'HiSeq_varID'</li><li>'HiSeq_REF'</li><li>'HiSeq_ALT'</li><li>'HiSeq_INFO'</li><li>'HiSeq_ADF'</li><li>'HiSeq_ADR'</li><li>'HiSeq_DP'</li><li>'HiSeq_Sample'</li><li>'HiSeq_ADF_Ref'</li><li>'HiSeq_ADF_Alt'</li><li>'HiSeq_ADR_Ref'</li><li>'HiSeq_ADR_Alt'</li><li>'HiSeq_AD.Alt'</li><li>'HiSeq_VAF'</li><li>'HiSeq_CHROM_POS'</li><li>'HiSeq_Pileup'</li><li>'varID_Match'</li><li>'Imputed_VAF_v2'</li><li>'CH_method'</li><li>'dVAF_by_dT'</li><li>'logdVAF_by_dT'</li><li>'Variant_type'</li><li>'Gene_Group_R882'</li><li>'Gene_R882'</li><li>'is_notHiSeq'</li><li>'is_notMUTECT'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'CHROM\\_POS\\_GWAS\\_VISIT\\_base'\n",
       "\\item 'ARIC\\_ID'\n",
       "\\item 'varID\\_GWASID'\n",
       "\\item 'GWAS\\_ID.x'\n",
       "\\item 'CHR'\n",
       "\\item 'POS'\n",
       "\\item 'REF'\n",
       "\\item 'ALT'\n",
       "\\item 'Gene'\n",
       "\\item 'AD.v5'\n",
       "\\item 'VAF.v5'\n",
       "\\item 'DP.v5'\n",
       "\\item 'AD.Alt.v5'\n",
       "\\item 'FR.Alt.v5'\n",
       "\\item 'RR.Alt.v5'\n",
       "\\item 'Accession'\n",
       "\\item 'transcriptOI'\n",
       "\\item 'NonsynOI'\n",
       "\\item 'Func.refGene'\n",
       "\\item 'ExonicFunc.refGene'\n",
       "\\item 'AAChange.refGene'\n",
       "\\item 'AD.v2'\n",
       "\\item 'VAF.v2'\n",
       "\\item 'DP.v2'\n",
       "\\item 'AD.Alt.v2'\n",
       "\\item 'FR.Alt.v2'\n",
       "\\item 'RR.Alt.v2'\n",
       "\\item 'Batch'\n",
       "\\item 'dAge'\n",
       "\\item 'dVAF'\n",
       "\\item 'pctVAF'\n",
       "\\item 'growth\\_rate'\n",
       "\\item 'log\\_growth\\_rate'\n",
       "\\item 'Gene\\_Group'\n",
       "\\item 'Gene\\_Group\\_2'\n",
       "\\item 'Clone\\_status'\n",
       "\\item 'Trajectory'\n",
       "\\item 'GWAS\\_ID.y'\n",
       "\\item 'GWASID\\_Visit'\n",
       "\\item 'cram\\_id'\n",
       "\\item 'Visit.x'\n",
       "\\item 'Gender'\n",
       "\\item 'Age'\n",
       "\\item 'CH\\_baseline'\n",
       "\\item 'CH\\_v05'\n",
       "\\item 'incident\\_CH'\n",
       "\\item 'incident\\_CHvaf05'\n",
       "\\item 'incident\\_CHvaf10'\n",
       "\\item 'incident\\_DNMT3A'\n",
       "\\item 'incident\\_TET2'\n",
       "\\item 'incident\\_ASXL1'\n",
       "\\item 'incident\\_DTA'\n",
       "\\item 'incident\\_SF'\n",
       "\\item 'incident\\_DDR'\n",
       "\\item 'death.x'\n",
       "\\item 'death\\_date.x'\n",
       "\\item 'hg38\\_CRAM\\_ID'\n",
       "\\item 'aricid'\n",
       "\\item 'Visit.y'\n",
       "\\item 'sex'\n",
       "\\item 'race'\n",
       "\\item 'center'\n",
       "\\item 'dob'\n",
       "\\item 'date\\_base'\n",
       "\\item 'age\\_base'\n",
       "\\item 'fast\\_8\\_base'\n",
       "\\item 'bmi\\_base'\n",
       "\\item 'dm\\_126\\_base'\n",
       "\\item 'gluc\\_base'\n",
       "\\item 'htn\\_5\\_base'\n",
       "\\item 'chol\\_med\\_base'\n",
       "\\item 'statin\\_base'\n",
       "\\item 'cig\\_base'\n",
       "\\item 'chd\\_date'\n",
       "\\item 'is\\_date'\n",
       "\\item 'hs\\_date'\n",
       "\\item 'hf\\_date'\n",
       "\\item 'death.y'\n",
       "\\item 'death\\_date.y'\n",
       "\\item 'lfu\\_date'\n",
       "\\item 'chd\\_base'\n",
       "\\item 'is\\_base'\n",
       "\\item 'hs\\_base'\n",
       "\\item 'hf\\_base'\n",
       "\\item 'ever\\_smoke'\n",
       "\\item 'v2\\_vs\\_other'\n",
       "\\item 'Sex'\n",
       "\\item 'race\\_BW'\n",
       "\\item 'BMI\\_cat'\n",
       "\\item 'nonHDL\\_std'\n",
       "\\item 'hdl\\_base\\_std'\n",
       "\\item 'ldl\\_base\\_std'\n",
       "\\item 'chol\\_base\\_std'\n",
       "\\item 'tg\\_base\\_std'\n",
       "\\item 'gluc\\_base\\_std'\n",
       "\\item 'bmi\\_base\\_std'\n",
       "\\item 'CHvaf10\\_baseline'\n",
       "\\item 'CH\\_baseline\\_v05'\n",
       "\\item 'CHvaf10\\_baseline\\_v05'\n",
       "\\item 'CH\\_DTA'\n",
       "\\item 'CH\\_SF'\n",
       "\\item 'CH\\_DDR'\n",
       "\\item 'CH\\_DNMT3A'\n",
       "\\item 'CH\\_TET2'\n",
       "\\item 'CH\\_ASXL1'\n",
       "\\item 'nonHDL\\_base\\_std'\n",
       "\\item 'age\\_base\\_sqr'\n",
       "\\item 'Center'\n",
       "\\item 'sbp\\_base'\n",
       "\\item 'dbp\\_base'\n",
       "\\item 'hem\\_can\\_base'\n",
       "\\item 'hem\\_can\\_inc\\_base'\n",
       "\\item 'hem\\_can\\_int\\_base'\n",
       "\\item 'hem\\_can\\_inc\\_v5'\n",
       "\\item 'hem\\_can\\_int\\_v5'\n",
       "\\item 'chol\\_base'\n",
       "\\item 'hdl\\_base'\n",
       "\\item 'ldl\\_base'\n",
       "\\item 'tg\\_base'\n",
       "\\item 'chd\\_is\\_base'\n",
       "\\item 'chol\\_base\\_INT'\n",
       "\\item 'ldl\\_base\\_INT'\n",
       "\\item 'hdl\\_base\\_INT'\n",
       "\\item 'tg\\_base\\_INT'\n",
       "\\item 'nonHDL\\_base'\n",
       "\\item 'nonHDL\\_base\\_INT'\n",
       "\\item 'bmi\\_base\\_INT'\n",
       "\\item 'tg\\_to\\_hdl\\_base'\n",
       "\\item 'tg\\_to\\_hdl\\_base\\_INT'\n",
       "\\item 'ldl\\_base\\_nomal\\_vs\\_high'\n",
       "\\item 'hdl\\_base\\_low'\n",
       "\\item 'Dyslipidemia'\n",
       "\\item 'sex\\_v5'\n",
       "\\item 'race\\_v5'\n",
       "\\item 'center\\_v5'\n",
       "\\item 'dob\\_v5'\n",
       "\\item 'date\\_v5'\n",
       "\\item 'age\\_v5'\n",
       "\\item 'fast\\_8\\_v5'\n",
       "\\item 'bmi\\_v5'\n",
       "\\item 'dm\\_126\\_v5'\n",
       "\\item 'gluc\\_v5'\n",
       "\\item 'htn\\_5\\_v5'\n",
       "\\item 'chol\\_v5'\n",
       "\\item 'hdl\\_v5'\n",
       "\\item 'ldl\\_v5'\n",
       "\\item 'tg\\_v5'\n",
       "\\item 'chol\\_med\\_v5'\n",
       "\\item 'statin\\_v5'\n",
       "\\item 'cig\\_v5'\n",
       "\\item 'chd\\_v5'\n",
       "\\item 'is\\_v5'\n",
       "\\item 'hs\\_v5'\n",
       "\\item 'hf\\_v5'\n",
       "\\item 'CHROM\\_POS'\n",
       "\\item 'CHROM\\_POS\\_GWAS\\_VISIT\\_v5'\n",
       "\\item 'HiSeq\\_CRAM\\_ID'\n",
       "\\item 'HiSeq\\_Batch'\n",
       "\\item 'HiSeq\\_GWASID\\_Visit'\n",
       "\\item 'HiSeq\\_GWAS\\_ID'\n",
       "\\item 'HiSeq\\_Visit'\n",
       "\\item 'HiSeq\\_CHROM'\n",
       "\\item 'HiSeq\\_POS'\n",
       "\\item 'HiSeq\\_varID'\n",
       "\\item 'HiSeq\\_REF'\n",
       "\\item 'HiSeq\\_ALT'\n",
       "\\item 'HiSeq\\_INFO'\n",
       "\\item 'HiSeq\\_ADF'\n",
       "\\item 'HiSeq\\_ADR'\n",
       "\\item 'HiSeq\\_DP'\n",
       "\\item 'HiSeq\\_Sample'\n",
       "\\item 'HiSeq\\_ADF\\_Ref'\n",
       "\\item 'HiSeq\\_ADF\\_Alt'\n",
       "\\item 'HiSeq\\_ADR\\_Ref'\n",
       "\\item 'HiSeq\\_ADR\\_Alt'\n",
       "\\item 'HiSeq\\_AD.Alt'\n",
       "\\item 'HiSeq\\_VAF'\n",
       "\\item 'HiSeq\\_CHROM\\_POS'\n",
       "\\item 'HiSeq\\_Pileup'\n",
       "\\item 'varID\\_Match'\n",
       "\\item 'Imputed\\_VAF\\_v2'\n",
       "\\item 'CH\\_method'\n",
       "\\item 'dVAF\\_by\\_dT'\n",
       "\\item 'logdVAF\\_by\\_dT'\n",
       "\\item 'Variant\\_type'\n",
       "\\item 'Gene\\_Group\\_R882'\n",
       "\\item 'Gene\\_R882'\n",
       "\\item 'is\\_notHiSeq'\n",
       "\\item 'is\\_notMUTECT'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'CHROM_POS_GWAS_VISIT_base'\n",
       "2. 'ARIC_ID'\n",
       "3. 'varID_GWASID'\n",
       "4. 'GWAS_ID.x'\n",
       "5. 'CHR'\n",
       "6. 'POS'\n",
       "7. 'REF'\n",
       "8. 'ALT'\n",
       "9. 'Gene'\n",
       "10. 'AD.v5'\n",
       "11. 'VAF.v5'\n",
       "12. 'DP.v5'\n",
       "13. 'AD.Alt.v5'\n",
       "14. 'FR.Alt.v5'\n",
       "15. 'RR.Alt.v5'\n",
       "16. 'Accession'\n",
       "17. 'transcriptOI'\n",
       "18. 'NonsynOI'\n",
       "19. 'Func.refGene'\n",
       "20. 'ExonicFunc.refGene'\n",
       "21. 'AAChange.refGene'\n",
       "22. 'AD.v2'\n",
       "23. 'VAF.v2'\n",
       "24. 'DP.v2'\n",
       "25. 'AD.Alt.v2'\n",
       "26. 'FR.Alt.v2'\n",
       "27. 'RR.Alt.v2'\n",
       "28. 'Batch'\n",
       "29. 'dAge'\n",
       "30. 'dVAF'\n",
       "31. 'pctVAF'\n",
       "32. 'growth_rate'\n",
       "33. 'log_growth_rate'\n",
       "34. 'Gene_Group'\n",
       "35. 'Gene_Group_2'\n",
       "36. 'Clone_status'\n",
       "37. 'Trajectory'\n",
       "38. 'GWAS_ID.y'\n",
       "39. 'GWASID_Visit'\n",
       "40. 'cram_id'\n",
       "41. 'Visit.x'\n",
       "42. 'Gender'\n",
       "43. 'Age'\n",
       "44. 'CH_baseline'\n",
       "45. 'CH_v05'\n",
       "46. 'incident_CH'\n",
       "47. 'incident_CHvaf05'\n",
       "48. 'incident_CHvaf10'\n",
       "49. 'incident_DNMT3A'\n",
       "50. 'incident_TET2'\n",
       "51. 'incident_ASXL1'\n",
       "52. 'incident_DTA'\n",
       "53. 'incident_SF'\n",
       "54. 'incident_DDR'\n",
       "55. 'death.x'\n",
       "56. 'death_date.x'\n",
       "57. 'hg38_CRAM_ID'\n",
       "58. 'aricid'\n",
       "59. 'Visit.y'\n",
       "60. 'sex'\n",
       "61. 'race'\n",
       "62. 'center'\n",
       "63. 'dob'\n",
       "64. 'date_base'\n",
       "65. 'age_base'\n",
       "66. 'fast_8_base'\n",
       "67. 'bmi_base'\n",
       "68. 'dm_126_base'\n",
       "69. 'gluc_base'\n",
       "70. 'htn_5_base'\n",
       "71. 'chol_med_base'\n",
       "72. 'statin_base'\n",
       "73. 'cig_base'\n",
       "74. 'chd_date'\n",
       "75. 'is_date'\n",
       "76. 'hs_date'\n",
       "77. 'hf_date'\n",
       "78. 'death.y'\n",
       "79. 'death_date.y'\n",
       "80. 'lfu_date'\n",
       "81. 'chd_base'\n",
       "82. 'is_base'\n",
       "83. 'hs_base'\n",
       "84. 'hf_base'\n",
       "85. 'ever_smoke'\n",
       "86. 'v2_vs_other'\n",
       "87. 'Sex'\n",
       "88. 'race_BW'\n",
       "89. 'BMI_cat'\n",
       "90. 'nonHDL_std'\n",
       "91. 'hdl_base_std'\n",
       "92. 'ldl_base_std'\n",
       "93. 'chol_base_std'\n",
       "94. 'tg_base_std'\n",
       "95. 'gluc_base_std'\n",
       "96. 'bmi_base_std'\n",
       "97. 'CHvaf10_baseline'\n",
       "98. 'CH_baseline_v05'\n",
       "99. 'CHvaf10_baseline_v05'\n",
       "100. 'CH_DTA'\n",
       "101. 'CH_SF'\n",
       "102. 'CH_DDR'\n",
       "103. 'CH_DNMT3A'\n",
       "104. 'CH_TET2'\n",
       "105. 'CH_ASXL1'\n",
       "106. 'nonHDL_base_std'\n",
       "107. 'age_base_sqr'\n",
       "108. 'Center'\n",
       "109. 'sbp_base'\n",
       "110. 'dbp_base'\n",
       "111. 'hem_can_base'\n",
       "112. 'hem_can_inc_base'\n",
       "113. 'hem_can_int_base'\n",
       "114. 'hem_can_inc_v5'\n",
       "115. 'hem_can_int_v5'\n",
       "116. 'chol_base'\n",
       "117. 'hdl_base'\n",
       "118. 'ldl_base'\n",
       "119. 'tg_base'\n",
       "120. 'chd_is_base'\n",
       "121. 'chol_base_INT'\n",
       "122. 'ldl_base_INT'\n",
       "123. 'hdl_base_INT'\n",
       "124. 'tg_base_INT'\n",
       "125. 'nonHDL_base'\n",
       "126. 'nonHDL_base_INT'\n",
       "127. 'bmi_base_INT'\n",
       "128. 'tg_to_hdl_base'\n",
       "129. 'tg_to_hdl_base_INT'\n",
       "130. 'ldl_base_nomal_vs_high'\n",
       "131. 'hdl_base_low'\n",
       "132. 'Dyslipidemia'\n",
       "133. 'sex_v5'\n",
       "134. 'race_v5'\n",
       "135. 'center_v5'\n",
       "136. 'dob_v5'\n",
       "137. 'date_v5'\n",
       "138. 'age_v5'\n",
       "139. 'fast_8_v5'\n",
       "140. 'bmi_v5'\n",
       "141. 'dm_126_v5'\n",
       "142. 'gluc_v5'\n",
       "143. 'htn_5_v5'\n",
       "144. 'chol_v5'\n",
       "145. 'hdl_v5'\n",
       "146. 'ldl_v5'\n",
       "147. 'tg_v5'\n",
       "148. 'chol_med_v5'\n",
       "149. 'statin_v5'\n",
       "150. 'cig_v5'\n",
       "151. 'chd_v5'\n",
       "152. 'is_v5'\n",
       "153. 'hs_v5'\n",
       "154. 'hf_v5'\n",
       "155. 'CHROM_POS'\n",
       "156. 'CHROM_POS_GWAS_VISIT_v5'\n",
       "157. 'HiSeq_CRAM_ID'\n",
       "158. 'HiSeq_Batch'\n",
       "159. 'HiSeq_GWASID_Visit'\n",
       "160. 'HiSeq_GWAS_ID'\n",
       "161. 'HiSeq_Visit'\n",
       "162. 'HiSeq_CHROM'\n",
       "163. 'HiSeq_POS'\n",
       "164. 'HiSeq_varID'\n",
       "165. 'HiSeq_REF'\n",
       "166. 'HiSeq_ALT'\n",
       "167. 'HiSeq_INFO'\n",
       "168. 'HiSeq_ADF'\n",
       "169. 'HiSeq_ADR'\n",
       "170. 'HiSeq_DP'\n",
       "171. 'HiSeq_Sample'\n",
       "172. 'HiSeq_ADF_Ref'\n",
       "173. 'HiSeq_ADF_Alt'\n",
       "174. 'HiSeq_ADR_Ref'\n",
       "175. 'HiSeq_ADR_Alt'\n",
       "176. 'HiSeq_AD.Alt'\n",
       "177. 'HiSeq_VAF'\n",
       "178. 'HiSeq_CHROM_POS'\n",
       "179. 'HiSeq_Pileup'\n",
       "180. 'varID_Match'\n",
       "181. 'Imputed_VAF_v2'\n",
       "182. 'CH_method'\n",
       "183. 'dVAF_by_dT'\n",
       "184. 'logdVAF_by_dT'\n",
       "185. 'Variant_type'\n",
       "186. 'Gene_Group_R882'\n",
       "187. 'Gene_R882'\n",
       "188. 'is_notHiSeq'\n",
       "189. 'is_notMUTECT'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "  [1] \"CHROM_POS_GWAS_VISIT_base\" \"ARIC_ID\"                  \n",
       "  [3] \"varID_GWASID\"              \"GWAS_ID.x\"                \n",
       "  [5] \"CHR\"                       \"POS\"                      \n",
       "  [7] \"REF\"                       \"ALT\"                      \n",
       "  [9] \"Gene\"                      \"AD.v5\"                    \n",
       " [11] \"VAF.v5\"                    \"DP.v5\"                    \n",
       " [13] \"AD.Alt.v5\"                 \"FR.Alt.v5\"                \n",
       " [15] \"RR.Alt.v5\"                 \"Accession\"                \n",
       " [17] \"transcriptOI\"              \"NonsynOI\"                 \n",
       " [19] \"Func.refGene\"              \"ExonicFunc.refGene\"       \n",
       " [21] \"AAChange.refGene\"          \"AD.v2\"                    \n",
       " [23] \"VAF.v2\"                    \"DP.v2\"                    \n",
       " [25] \"AD.Alt.v2\"                 \"FR.Alt.v2\"                \n",
       " [27] \"RR.Alt.v2\"                 \"Batch\"                    \n",
       " [29] \"dAge\"                      \"dVAF\"                     \n",
       " [31] \"pctVAF\"                    \"growth_rate\"              \n",
       " [33] \"log_growth_rate\"           \"Gene_Group\"               \n",
       " [35] \"Gene_Group_2\"              \"Clone_status\"             \n",
       " [37] \"Trajectory\"                \"GWAS_ID.y\"                \n",
       " [39] \"GWASID_Visit\"              \"cram_id\"                  \n",
       " [41] \"Visit.x\"                   \"Gender\"                   \n",
       " [43] \"Age\"                       \"CH_baseline\"              \n",
       " [45] \"CH_v05\"                    \"incident_CH\"              \n",
       " [47] \"incident_CHvaf05\"          \"incident_CHvaf10\"         \n",
       " [49] \"incident_DNMT3A\"           \"incident_TET2\"            \n",
       " [51] \"incident_ASXL1\"            \"incident_DTA\"             \n",
       " [53] \"incident_SF\"               \"incident_DDR\"             \n",
       " [55] \"death.x\"                   \"death_date.x\"             \n",
       " [57] \"hg38_CRAM_ID\"              \"aricid\"                   \n",
       " [59] \"Visit.y\"                   \"sex\"                      \n",
       " [61] \"race\"                      \"center\"                   \n",
       " [63] \"dob\"                       \"date_base\"                \n",
       " [65] \"age_base\"                  \"fast_8_base\"              \n",
       " [67] \"bmi_base\"                  \"dm_126_base\"              \n",
       " [69] \"gluc_base\"                 \"htn_5_base\"               \n",
       " [71] \"chol_med_base\"             \"statin_base\"              \n",
       " [73] \"cig_base\"                  \"chd_date\"                 \n",
       " [75] \"is_date\"                   \"hs_date\"                  \n",
       " [77] \"hf_date\"                   \"death.y\"                  \n",
       " [79] \"death_date.y\"              \"lfu_date\"                 \n",
       " [81] \"chd_base\"                  \"is_base\"                  \n",
       " [83] \"hs_base\"                   \"hf_base\"                  \n",
       " [85] \"ever_smoke\"                \"v2_vs_other\"              \n",
       " [87] \"Sex\"                       \"race_BW\"                  \n",
       " [89] \"BMI_cat\"                   \"nonHDL_std\"               \n",
       " [91] \"hdl_base_std\"              \"ldl_base_std\"             \n",
       " [93] \"chol_base_std\"             \"tg_base_std\"              \n",
       " [95] \"gluc_base_std\"             \"bmi_base_std\"             \n",
       " [97] \"CHvaf10_baseline\"          \"CH_baseline_v05\"          \n",
       " [99] \"CHvaf10_baseline_v05\"      \"CH_DTA\"                   \n",
       "[101] \"CH_SF\"                     \"CH_DDR\"                   \n",
       "[103] \"CH_DNMT3A\"                 \"CH_TET2\"                  \n",
       "[105] \"CH_ASXL1\"                  \"nonHDL_base_std\"          \n",
       "[107] \"age_base_sqr\"              \"Center\"                   \n",
       "[109] \"sbp_base\"                  \"dbp_base\"                 \n",
       "[111] \"hem_can_base\"              \"hem_can_inc_base\"         \n",
       "[113] \"hem_can_int_base\"          \"hem_can_inc_v5\"           \n",
       "[115] \"hem_can_int_v5\"            \"chol_base\"                \n",
       "[117] \"hdl_base\"                  \"ldl_base\"                 \n",
       "[119] \"tg_base\"                   \"chd_is_base\"              \n",
       "[121] \"chol_base_INT\"             \"ldl_base_INT\"             \n",
       "[123] \"hdl_base_INT\"              \"tg_base_INT\"              \n",
       "[125] \"nonHDL_base\"               \"nonHDL_base_INT\"          \n",
       "[127] \"bmi_base_INT\"              \"tg_to_hdl_base\"           \n",
       "[129] \"tg_to_hdl_base_INT\"        \"ldl_base_nomal_vs_high\"   \n",
       "[131] \"hdl_base_low\"              \"Dyslipidemia\"             \n",
       "[133] \"sex_v5\"                    \"race_v5\"                  \n",
       "[135] \"center_v5\"                 \"dob_v5\"                   \n",
       "[137] \"date_v5\"                   \"age_v5\"                   \n",
       "[139] \"fast_8_v5\"                 \"bmi_v5\"                   \n",
       "[141] \"dm_126_v5\"                 \"gluc_v5\"                  \n",
       "[143] \"htn_5_v5\"                  \"chol_v5\"                  \n",
       "[145] \"hdl_v5\"                    \"ldl_v5\"                   \n",
       "[147] \"tg_v5\"                     \"chol_med_v5\"              \n",
       "[149] \"statin_v5\"                 \"cig_v5\"                   \n",
       "[151] \"chd_v5\"                    \"is_v5\"                    \n",
       "[153] \"hs_v5\"                     \"hf_v5\"                    \n",
       "[155] \"CHROM_POS\"                 \"CHROM_POS_GWAS_VISIT_v5\"  \n",
       "[157] \"HiSeq_CRAM_ID\"             \"HiSeq_Batch\"              \n",
       "[159] \"HiSeq_GWASID_Visit\"        \"HiSeq_GWAS_ID\"            \n",
       "[161] \"HiSeq_Visit\"               \"HiSeq_CHROM\"              \n",
       "[163] \"HiSeq_POS\"                 \"HiSeq_varID\"              \n",
       "[165] \"HiSeq_REF\"                 \"HiSeq_ALT\"                \n",
       "[167] \"HiSeq_INFO\"                \"HiSeq_ADF\"                \n",
       "[169] \"HiSeq_ADR\"                 \"HiSeq_DP\"                 \n",
       "[171] \"HiSeq_Sample\"              \"HiSeq_ADF_Ref\"            \n",
       "[173] \"HiSeq_ADF_Alt\"             \"HiSeq_ADR_Ref\"            \n",
       "[175] \"HiSeq_ADR_Alt\"             \"HiSeq_AD.Alt\"             \n",
       "[177] \"HiSeq_VAF\"                 \"HiSeq_CHROM_POS\"          \n",
       "[179] \"HiSeq_Pileup\"              \"varID_Match\"              \n",
       "[181] \"Imputed_VAF_v2\"            \"CH_method\"                \n",
       "[183] \"dVAF_by_dT\"                \"logdVAF_by_dT\"            \n",
       "[185] \"Variant_type\"              \"Gene_Group_R882\"          \n",
       "[187] \"Gene_R882\"                 \"is_notHiSeq\"              \n",
       "[189] \"is_notMUTECT\"             "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(cln_grt.vaf2.DP20_base.corrected_ordered)\n",
    "\n",
    "names(cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE \n",
       "  797   167 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE \n",
       "  354   610 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE \n",
       "  540   118 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/plain": [
       "\n",
       "FALSE  TRUE \n",
       "  241   417 "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "table(cln_grt.vaf2.DP20_base.corrected_ordered$ARIC_ID %in% aric_baseline_n_v05_aa_gwas$ARIC_ID)\n",
    "table(cln_grt.vaf2.DP20_base.corrected_ordered$ARIC_ID %in% aric_baseline_n_v05_ea_gwas$ARIC_ID)\n",
    "\n",
    "table(cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered$ARIC_ID %in% aric_baseline_n_v05_aa_gwas$ARIC_ID)\n",
    "table(cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered$ARIC_ID %in% aric_baseline_n_v05_ea_gwas$ARIC_ID)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 4 x 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>CHROM_POS_GWAS_VISIT_base</th><th scope=col>ARIC_ID</th><th scope=col>varID_GWASID</th><th scope=col>dVAF_by_dT</th><th scope=col>logdVAF_by_dT</th><th scope=col>Variant_type</th><th scope=col>Gene_Group_R882</th><th scope=col>Gene_R882</th><th scope=col>is_notHiSeq</th><th scope=col>is_notMUTECT</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr2_25744364_A00651_V02 </td><td>W321690</td><td>chr2_25744364_G_GC_A00651 </td><td>0.004363158</td><td>0.35375925</td><td>INDEL   </td><td>Other</td><td>ASXL2</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>chr16_30729462_A07730_V02</td><td>W320923</td><td>chr16_30729462_TC_T_A07730</td><td>0.002152632</td><td>0.31663985</td><td>INDEL   </td><td>Other</td><td>SRCAP</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>chr4_105275518_A11085_V02</td><td>W320503</td><td>chr4_105275518_CT_C_A11085</td><td>0.004661905</td><td>0.32797869</td><td>INDEL   </td><td>TET2 </td><td>TET2 </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>chr9_5073770_A08738_V02  </td><td>W320457</td><td>chr9_5073770_G_T_A08738   </td><td>0.009952381</td><td>0.08071571</td><td>Missense</td><td>Other</td><td>JAK2 </td><td>1</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 4 x 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " CHROM\\_POS\\_GWAS\\_VISIT\\_base & ARIC\\_ID & varID\\_GWASID & dVAF\\_by\\_dT & logdVAF\\_by\\_dT & Variant\\_type & Gene\\_Group\\_R882 & Gene\\_R882 & is\\_notHiSeq & is\\_notMUTECT\\\\\n",
       " <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr2\\_25744364\\_A00651\\_V02  & W321690 & chr2\\_25744364\\_G\\_GC\\_A00651  & 0.004363158 & 0.35375925 & INDEL    & Other & ASXL2 & 0 & 1\\\\\n",
       "\t chr16\\_30729462\\_A07730\\_V02 & W320923 & chr16\\_30729462\\_TC\\_T\\_A07730 & 0.002152632 & 0.31663985 & INDEL    & Other & SRCAP & 0 & 1\\\\\n",
       "\t chr4\\_105275518\\_A11085\\_V02 & W320503 & chr4\\_105275518\\_CT\\_C\\_A11085 & 0.004661905 & 0.32797869 & INDEL    & TET2  & TET2  & 0 & 1\\\\\n",
       "\t chr9\\_5073770\\_A08738\\_V02   & W320457 & chr9\\_5073770\\_G\\_T\\_A08738    & 0.009952381 & 0.08071571 & Missense & Other & JAK2  & 1 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 4 x 10\n",
       "\n",
       "| CHROM_POS_GWAS_VISIT_base &lt;chr&gt; | ARIC_ID &lt;chr&gt; | varID_GWASID &lt;chr&gt; | dVAF_by_dT &lt;dbl&gt; | logdVAF_by_dT &lt;dbl&gt; | Variant_type &lt;chr&gt; | Gene_Group_R882 &lt;chr&gt; | Gene_R882 &lt;chr&gt; | is_notHiSeq &lt;int&gt; | is_notMUTECT &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr2_25744364_A00651_V02  | W321690 | chr2_25744364_G_GC_A00651  | 0.004363158 | 0.35375925 | INDEL    | Other | ASXL2 | 0 | 1 |\n",
       "| chr16_30729462_A07730_V02 | W320923 | chr16_30729462_TC_T_A07730 | 0.002152632 | 0.31663985 | INDEL    | Other | SRCAP | 0 | 1 |\n",
       "| chr4_105275518_A11085_V02 | W320503 | chr4_105275518_CT_C_A11085 | 0.004661905 | 0.32797869 | INDEL    | TET2  | TET2  | 0 | 1 |\n",
       "| chr9_5073770_A08738_V02   | W320457 | chr9_5073770_G_T_A08738    | 0.009952381 | 0.08071571 | Missense | Other | JAK2  | 1 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHROM_POS_GWAS_VISIT_base ARIC_ID varID_GWASID               dVAF_by_dT \n",
       "1 chr2_25744364_A00651_V02  W321690 chr2_25744364_G_GC_A00651  0.004363158\n",
       "2 chr16_30729462_A07730_V02 W320923 chr16_30729462_TC_T_A07730 0.002152632\n",
       "3 chr4_105275518_A11085_V02 W320503 chr4_105275518_CT_C_A11085 0.004661905\n",
       "4 chr9_5073770_A08738_V02   W320457 chr9_5073770_G_T_A08738    0.009952381\n",
       "  logdVAF_by_dT Variant_type Gene_Group_R882 Gene_R882 is_notHiSeq is_notMUTECT\n",
       "1 0.35375925    INDEL        Other           ASXL2     0           1           \n",
       "2 0.31663985    INDEL        Other           SRCAP     0           1           \n",
       "3 0.32797869    INDEL        TET2            TET2      0           1           \n",
       "4 0.08071571    Missense     Other           JAK2      1           0           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 4 x 10</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>CHROM_POS_GWAS_VISIT_base</th><th scope=col>ARIC_ID</th><th scope=col>varID_GWASID</th><th scope=col>dVAF_by_dT</th><th scope=col>logdVAF_by_dT</th><th scope=col>Variant_type</th><th scope=col>Gene_Group_R882</th><th scope=col>Gene_R882</th><th scope=col>is_notHiSeq</th><th scope=col>is_notMUTECT</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr2_25744364_A00651_V02 </td><td>W321690</td><td>chr2_25744364_G_GC_A00651 </td><td>0.004363158</td><td>0.35375925</td><td>INDEL   </td><td>Other         </td><td>ASXL2         </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>chr4_105275518_A11085_V02</td><td>W320503</td><td>chr4_105275518_CT_C_A11085</td><td>0.004661905</td><td>0.32797869</td><td>INDEL   </td><td>TET2          </td><td>TET2          </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>chr9_5073770_A08738_V02  </td><td>W320457</td><td>chr9_5073770_G_T_A08738   </td><td>0.009952381</td><td>0.08071571</td><td>Missense</td><td>Other         </td><td>JAK2          </td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>chr2_25234311_A16062_V02 </td><td>W319785</td><td>chr2_25234311_CG_C_A16062 </td><td>0.004105263</td><td>0.02967200</td><td>INDEL   </td><td>DNMT3A_nonR882</td><td>DNMT3A_nonR882</td><td>0</td><td>0</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 4 x 10\n",
       "\\begin{tabular}{llllllllll}\n",
       " CHROM\\_POS\\_GWAS\\_VISIT\\_base & ARIC\\_ID & varID\\_GWASID & dVAF\\_by\\_dT & logdVAF\\_by\\_dT & Variant\\_type & Gene\\_Group\\_R882 & Gene\\_R882 & is\\_notHiSeq & is\\_notMUTECT\\\\\n",
       " <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t chr2\\_25744364\\_A00651\\_V02  & W321690 & chr2\\_25744364\\_G\\_GC\\_A00651  & 0.004363158 & 0.35375925 & INDEL    & Other          & ASXL2          & 0 & 1\\\\\n",
       "\t chr4\\_105275518\\_A11085\\_V02 & W320503 & chr4\\_105275518\\_CT\\_C\\_A11085 & 0.004661905 & 0.32797869 & INDEL    & TET2           & TET2           & 0 & 1\\\\\n",
       "\t chr9\\_5073770\\_A08738\\_V02   & W320457 & chr9\\_5073770\\_G\\_T\\_A08738    & 0.009952381 & 0.08071571 & Missense & Other          & JAK2           & 1 & 0\\\\\n",
       "\t chr2\\_25234311\\_A16062\\_V02  & W319785 & chr2\\_25234311\\_CG\\_C\\_A16062  & 0.004105263 & 0.02967200 & INDEL    & DNMT3A\\_nonR882 & DNMT3A\\_nonR882 & 0 & 0\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 4 x 10\n",
       "\n",
       "| CHROM_POS_GWAS_VISIT_base &lt;chr&gt; | ARIC_ID &lt;chr&gt; | varID_GWASID &lt;chr&gt; | dVAF_by_dT &lt;dbl&gt; | logdVAF_by_dT &lt;dbl&gt; | Variant_type &lt;chr&gt; | Gene_Group_R882 &lt;chr&gt; | Gene_R882 &lt;chr&gt; | is_notHiSeq &lt;int&gt; | is_notMUTECT &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|\n",
       "| chr2_25744364_A00651_V02  | W321690 | chr2_25744364_G_GC_A00651  | 0.004363158 | 0.35375925 | INDEL    | Other          | ASXL2          | 0 | 1 |\n",
       "| chr4_105275518_A11085_V02 | W320503 | chr4_105275518_CT_C_A11085 | 0.004661905 | 0.32797869 | INDEL    | TET2           | TET2           | 0 | 1 |\n",
       "| chr9_5073770_A08738_V02   | W320457 | chr9_5073770_G_T_A08738    | 0.009952381 | 0.08071571 | Missense | Other          | JAK2           | 1 | 0 |\n",
       "| chr2_25234311_A16062_V02  | W319785 | chr2_25234311_CG_C_A16062  | 0.004105263 | 0.02967200 | INDEL    | DNMT3A_nonR882 | DNMT3A_nonR882 | 0 | 0 |\n",
       "\n"
      ],
      "text/plain": [
       "  CHROM_POS_GWAS_VISIT_base ARIC_ID varID_GWASID               dVAF_by_dT \n",
       "1 chr2_25744364_A00651_V02  W321690 chr2_25744364_G_GC_A00651  0.004363158\n",
       "2 chr4_105275518_A11085_V02 W320503 chr4_105275518_CT_C_A11085 0.004661905\n",
       "3 chr9_5073770_A08738_V02   W320457 chr9_5073770_G_T_A08738    0.009952381\n",
       "4 chr2_25234311_A16062_V02  W319785 chr2_25234311_CG_C_A16062  0.004105263\n",
       "  logdVAF_by_dT Variant_type Gene_Group_R882 Gene_R882      is_notHiSeq\n",
       "1 0.35375925    INDEL        Other           ASXL2          0          \n",
       "2 0.32797869    INDEL        TET2            TET2           0          \n",
       "3 0.08071571    Missense     Other           JAK2           1          \n",
       "4 0.02967200    INDEL        DNMT3A_nonR882  DNMT3A_nonR882 0          \n",
       "  is_notMUTECT\n",
       "1 1           \n",
       "2 1           \n",
       "3 0           \n",
       "4 0           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(cln_grt.vaf2.DP20_base.corrected_ordered[,c(1:3, 183:189)],4)\n",
    "head(cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered[,c(1:3, 183:189)],4)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "grt_aa.dp20 <- merge(aric_baseline_n_v05_aa_gwas, \n",
    "                     cln_grt.vaf2.DP20_base.corrected_ordered[,c(1:3, 183:189)], \n",
    "                     by=\"ARIC_ID\")\n",
    "\n",
    "grt_ea.dp20 <- merge(aric_baseline_n_v05_ea_gwas, \n",
    "                     cln_grt.vaf2.DP20_base.corrected_ordered[,c(1:3, 183:189)], \n",
    "                     by=\"ARIC_ID\")\n",
    "\n",
    "grt_aa.dp20allAD5FRRR2 <- merge(aric_baseline_n_v05_aa_gwas, \n",
    "                     cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered[,c(1:3, 183:189)], \n",
    "                     by=\"ARIC_ID\")\n",
    "\n",
    "grt_ea.dp20allAD5FRRR2 <- merge(aric_baseline_n_v05_ea_gwas, \n",
    "                     cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered[,c(1:3, 183:189)], \n",
    "                     by=\"ARIC_ID\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 47</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ARIC_ID</th><th scope=col>FID</th><th scope=col>IID</th><th scope=col>GWAS_ID</th><th scope=col>GWASID_Visit</th><th scope=col>dbGaP_Subject_ID</th><th scope=col>dbGaP_Sample_ID</th><th scope=col>Visit.y</th><th scope=col>Gender</th><th scope=col>Age</th><th scope=col>...</th><th scope=col>Sex</th><th scope=col>CHROM_POS_GWAS_VISIT_base</th><th scope=col>varID_GWASID</th><th scope=col>dVAF_by_dT</th><th scope=col>logdVAF_by_dT</th><th scope=col>Variant_type</th><th scope=col>Gene_Group_R882</th><th scope=col>Gene_R882</th><th scope=col>is_notHiSeq</th><th scope=col>is_notMUTECT</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>...</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>F108396</td><td>131164765</td><td>131164765</td><td>A01355</td><td>A01355_V05</td><td>228830</td><td>698247</td><td>V02</td><td>M</td><td>71</td><td>...</td><td>M</td><td>chrX_45083470_A01355_V02 </td><td>chrX_45083470_C_T_A01355                  </td><td>0.002450000</td><td>0.28598042</td><td>Stopgain</td><td>Other         </td><td>KDM6A         </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F120829</td><td>131047508</td><td>131047508</td><td>A11026</td><td>A11026_V05</td><td>228749</td><td>700233</td><td>V03</td><td>M</td><td>77</td><td>...</td><td>M</td><td>chr13_32758127_A11026_V03</td><td>chr13_32758127_G_T_A11026                 </td><td>0.001889474</td><td>0.30979495</td><td>Stopgain</td><td>Other         </td><td>PDS5B         </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F123365</td><td>131123957</td><td>131123957</td><td>A05140</td><td>A05140_V05</td><td>223229</td><td>699024</td><td>V03</td><td>M</td><td>69</td><td>...</td><td>M</td><td>chr4_105272637_A05140_V03</td><td>chr4_105272637_CT_C_A05140                </td><td>0.005935294</td><td>0.40692386</td><td>INDEL   </td><td>TET2          </td><td>TET2          </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F143923</td><td>131026458</td><td>131026458</td><td>A16098</td><td>A16098_V05</td><td>227512</td><td>701242</td><td>V02</td><td>M</td><td>75</td><td>...</td><td>M</td><td>chr2_25246159_A16098_V02 </td><td>chr2_25246159_C_T_A16098                  </td><td>0.007809524</td><td>0.09801848</td><td>Splicing</td><td>DNMT3A_nonR882</td><td>DNMT3A_nonR882</td><td>0</td><td>0</td></tr>\n",
       "\t<tr><td>F156749</td><td>131003516</td><td>131003516</td><td>A09965</td><td>A09965_V05</td><td>232484</td><td>700011</td><td>V03</td><td>F</td><td>85</td><td>...</td><td>F</td><td>chr17_60663118_A09965_V03</td><td>chr17_60663118_C_T_A09965                 </td><td>0.012347059</td><td>0.44998192</td><td>Stopgain</td><td>DDR           </td><td>PPM1D         </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F205527</td><td>131149782</td><td>131149782</td><td>A05890</td><td>A05890_V05</td><td>229747</td><td>699176</td><td>V02</td><td>M</td><td>80</td><td>...</td><td>M</td><td>chr4_105236124_A05890_V02</td><td>chr4_105236124_GCACAAACACAACCATCC_G_A05890</td><td>0.002450000</td><td>0.28598042</td><td>INDEL   </td><td>TET2          </td><td>TET2          </td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 47\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ARIC\\_ID & FID & IID & GWAS\\_ID & GWASID\\_Visit & dbGaP\\_Subject\\_ID & dbGaP\\_Sample\\_ID & Visit.y & Gender & Age & ... & Sex & CHROM\\_POS\\_GWAS\\_VISIT\\_base & varID\\_GWASID & dVAF\\_by\\_dT & logdVAF\\_by\\_dT & Variant\\_type & Gene\\_Group\\_R882 & Gene\\_R882 & is\\_notHiSeq & is\\_notMUTECT\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <int> & ... & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t F108396 & 131164765 & 131164765 & A01355 & A01355\\_V05 & 228830 & 698247 & V02 & M & 71 & ... & M & chrX\\_45083470\\_A01355\\_V02  & chrX\\_45083470\\_C\\_T\\_A01355                   & 0.002450000 & 0.28598042 & Stopgain & Other          & KDM6A          & 0 & 1\\\\\n",
       "\t F120829 & 131047508 & 131047508 & A11026 & A11026\\_V05 & 228749 & 700233 & V03 & M & 77 & ... & M & chr13\\_32758127\\_A11026\\_V03 & chr13\\_32758127\\_G\\_T\\_A11026                  & 0.001889474 & 0.30979495 & Stopgain & Other          & PDS5B          & 0 & 1\\\\\n",
       "\t F123365 & 131123957 & 131123957 & A05140 & A05140\\_V05 & 223229 & 699024 & V03 & M & 69 & ... & M & chr4\\_105272637\\_A05140\\_V03 & chr4\\_105272637\\_CT\\_C\\_A05140                 & 0.005935294 & 0.40692386 & INDEL    & TET2           & TET2           & 0 & 1\\\\\n",
       "\t F143923 & 131026458 & 131026458 & A16098 & A16098\\_V05 & 227512 & 701242 & V02 & M & 75 & ... & M & chr2\\_25246159\\_A16098\\_V02  & chr2\\_25246159\\_C\\_T\\_A16098                   & 0.007809524 & 0.09801848 & Splicing & DNMT3A\\_nonR882 & DNMT3A\\_nonR882 & 0 & 0\\\\\n",
       "\t F156749 & 131003516 & 131003516 & A09965 & A09965\\_V05 & 232484 & 700011 & V03 & F & 85 & ... & F & chr17\\_60663118\\_A09965\\_V03 & chr17\\_60663118\\_C\\_T\\_A09965                  & 0.012347059 & 0.44998192 & Stopgain & DDR            & PPM1D          & 0 & 1\\\\\n",
       "\t F205527 & 131149782 & 131149782 & A05890 & A05890\\_V05 & 229747 & 699176 & V02 & M & 80 & ... & M & chr4\\_105236124\\_A05890\\_V02 & chr4\\_105236124\\_GCACAAACACAACCATCC\\_G\\_A05890 & 0.002450000 & 0.28598042 & INDEL    & TET2           & TET2           & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 47\n",
       "\n",
       "| ARIC_ID &lt;chr&gt; | FID &lt;int&gt; | IID &lt;int&gt; | GWAS_ID &lt;chr&gt; | GWASID_Visit &lt;chr&gt; | dbGaP_Subject_ID &lt;int&gt; | dbGaP_Sample_ID &lt;int&gt; | Visit.y &lt;chr&gt; | Gender &lt;chr&gt; | Age &lt;int&gt; | ... ... | Sex &lt;chr&gt; | CHROM_POS_GWAS_VISIT_base &lt;chr&gt; | varID_GWASID &lt;chr&gt; | dVAF_by_dT &lt;dbl&gt; | logdVAF_by_dT &lt;dbl&gt; | Variant_type &lt;chr&gt; | Gene_Group_R882 &lt;chr&gt; | Gene_R882 &lt;chr&gt; | is_notHiSeq &lt;int&gt; | is_notMUTECT &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| F108396 | 131164765 | 131164765 | A01355 | A01355_V05 | 228830 | 698247 | V02 | M | 71 | ... | M | chrX_45083470_A01355_V02  | chrX_45083470_C_T_A01355                   | 0.002450000 | 0.28598042 | Stopgain | Other          | KDM6A          | 0 | 1 |\n",
       "| F120829 | 131047508 | 131047508 | A11026 | A11026_V05 | 228749 | 700233 | V03 | M | 77 | ... | M | chr13_32758127_A11026_V03 | chr13_32758127_G_T_A11026                  | 0.001889474 | 0.30979495 | Stopgain | Other          | PDS5B          | 0 | 1 |\n",
       "| F123365 | 131123957 | 131123957 | A05140 | A05140_V05 | 223229 | 699024 | V03 | M | 69 | ... | M | chr4_105272637_A05140_V03 | chr4_105272637_CT_C_A05140                 | 0.005935294 | 0.40692386 | INDEL    | TET2           | TET2           | 0 | 1 |\n",
       "| F143923 | 131026458 | 131026458 | A16098 | A16098_V05 | 227512 | 701242 | V02 | M | 75 | ... | M | chr2_25246159_A16098_V02  | chr2_25246159_C_T_A16098                   | 0.007809524 | 0.09801848 | Splicing | DNMT3A_nonR882 | DNMT3A_nonR882 | 0 | 0 |\n",
       "| F156749 | 131003516 | 131003516 | A09965 | A09965_V05 | 232484 | 700011 | V03 | F | 85 | ... | F | chr17_60663118_A09965_V03 | chr17_60663118_C_T_A09965                  | 0.012347059 | 0.44998192 | Stopgain | DDR            | PPM1D          | 0 | 1 |\n",
       "| F205527 | 131149782 | 131149782 | A05890 | A05890_V05 | 229747 | 699176 | V02 | M | 80 | ... | M | chr4_105236124_A05890_V02 | chr4_105236124_GCACAAACACAACCATCC_G_A05890 | 0.002450000 | 0.28598042 | INDEL    | TET2           | TET2           | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  ARIC_ID FID       IID       GWAS_ID GWASID_Visit dbGaP_Subject_ID\n",
       "1 F108396 131164765 131164765 A01355  A01355_V05   228830          \n",
       "2 F120829 131047508 131047508 A11026  A11026_V05   228749          \n",
       "3 F123365 131123957 131123957 A05140  A05140_V05   223229          \n",
       "4 F143923 131026458 131026458 A16098  A16098_V05   227512          \n",
       "5 F156749 131003516 131003516 A09965  A09965_V05   232484          \n",
       "6 F205527 131149782 131149782 A05890  A05890_V05   229747          \n",
       "  dbGaP_Sample_ID Visit.y Gender Age ... Sex CHROM_POS_GWAS_VISIT_base\n",
       "1 698247          V02     M      71  ... M   chrX_45083470_A01355_V02 \n",
       "2 700233          V03     M      77  ... M   chr13_32758127_A11026_V03\n",
       "3 699024          V03     M      69  ... M   chr4_105272637_A05140_V03\n",
       "4 701242          V02     M      75  ... M   chr2_25246159_A16098_V02 \n",
       "5 700011          V03     F      85  ... F   chr17_60663118_A09965_V03\n",
       "6 699176          V02     M      80  ... M   chr4_105236124_A05890_V02\n",
       "  varID_GWASID                               dVAF_by_dT  logdVAF_by_dT\n",
       "1 chrX_45083470_C_T_A01355                   0.002450000 0.28598042   \n",
       "2 chr13_32758127_G_T_A11026                  0.001889474 0.30979495   \n",
       "3 chr4_105272637_CT_C_A05140                 0.005935294 0.40692386   \n",
       "4 chr2_25246159_C_T_A16098                   0.007809524 0.09801848   \n",
       "5 chr17_60663118_C_T_A09965                  0.012347059 0.44998192   \n",
       "6 chr4_105236124_GCACAAACACAACCATCC_G_A05890 0.002450000 0.28598042   \n",
       "  Variant_type Gene_Group_R882 Gene_R882      is_notHiSeq is_notMUTECT\n",
       "1 Stopgain     Other           KDM6A          0           1           \n",
       "2 Stopgain     Other           PDS5B          0           1           \n",
       "3 INDEL        TET2            TET2           0           1           \n",
       "4 Splicing     DNMT3A_nonR882  DNMT3A_nonR882 0           0           \n",
       "5 Stopgain     DDR             PPM1D          0           1           \n",
       "6 INDEL        TET2            TET2           0           1           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.table: 6 x 47</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>ARIC_ID</th><th scope=col>FID</th><th scope=col>IID</th><th scope=col>GWAS_ID</th><th scope=col>GWASID_Visit</th><th scope=col>dbGaP_Subject_ID</th><th scope=col>dbGaP_Sample_ID</th><th scope=col>Visit.y</th><th scope=col>Gender</th><th scope=col>Age</th><th scope=col>...</th><th scope=col>Sex</th><th scope=col>CHROM_POS_GWAS_VISIT_base</th><th scope=col>varID_GWASID</th><th scope=col>dVAF_by_dT</th><th scope=col>logdVAF_by_dT</th><th scope=col>Variant_type</th><th scope=col>Gene_Group_R882</th><th scope=col>Gene_R882</th><th scope=col>is_notHiSeq</th><th scope=col>is_notMUTECT</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>...</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>F100601</td><td>131179968</td><td>131179968</td><td>A09377</td><td>A09377_V05</td><td>229137</td><td>1790886</td><td>V03</td><td>F</td><td>83</td><td>...</td><td>F</td><td>chr2_25248155_A09377_V03 </td><td>chr2_25248155_GCAGGAGGGCTGGCCTCCTCCACCTT_G_A09377</td><td>0.009573684</td><td>0.39508378</td><td>INDEL   </td><td>DNMT3A_nonR882</td><td>DNMT3A_nonR882</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F100920</td><td>131156438</td><td>131156438</td><td>A14695</td><td>A14695_V05</td><td>235154</td><td>1793159</td><td>V03</td><td>F</td><td>83</td><td>...</td><td>F</td><td>chr2_25240714_A14695_V03 </td><td>chr2_25240714_G_A_A14695                         </td><td>0.006845000</td><td>0.36112830</td><td>Missense</td><td>DNMT3A_nonR882</td><td>DNMT3A_nonR882</td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F101063</td><td>131024996</td><td>131024996</td><td>A14432</td><td>A14432_V05</td><td>227292</td><td>1336307</td><td>MRI</td><td>M</td><td>81</td><td>...</td><td>M</td><td>chrX_155120076_A14432_MRI</td><td>chrX_155120076_C_T_A14432                        </td><td>0.107128571</td><td>1.27466547</td><td>Stopgain</td><td>Other         </td><td>BRCC3         </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F101108</td><td>131088751</td><td>131088751</td><td>A01650</td><td>A01650_V05</td><td>230347</td><td>1334735</td><td>V02</td><td>M</td><td>83</td><td>...</td><td>M</td><td>chr17_76736877_A01650_V02</td><td>chr17_76736877_G_T_A01650                        </td><td>0.002359091</td><td>0.28426495</td><td>Missense</td><td>SF            </td><td>SRSF2         </td><td>0</td><td>1</td></tr>\n",
       "\t<tr><td>F101279</td><td>131010885</td><td>131010885</td><td>A11069</td><td>A11069_V05</td><td>224927</td><td>1335914</td><td>V02</td><td>M</td><td>70</td><td>...</td><td>M</td><td>chr2_25240420_A11069_V02 </td><td>chr2_25240420_T_C_A11069                         </td><td>0.001636364</td><td>0.03215145</td><td>Missense</td><td>DNMT3A_nonR882</td><td>DNMT3A_nonR882</td><td>1</td><td>0</td></tr>\n",
       "\t<tr><td>F102435</td><td>131177079</td><td>131177079</td><td>A16447</td><td>A16447_V05</td><td>234825</td><td>1336548</td><td>V03</td><td>F</td><td>73</td><td>...</td><td>F</td><td>chr2_25244340_A16447_V03 </td><td>chr2_25244340_T_C_A16447                         </td><td>0.002195000</td><td>0.30433874</td><td>Splicing</td><td>DNMT3A_nonR882</td><td>DNMT3A_nonR882</td><td>0</td><td>1</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.table: 6 x 47\n",
       "\\begin{tabular}{lllllllllllllllllllll}\n",
       " ARIC\\_ID & FID & IID & GWAS\\_ID & GWASID\\_Visit & dbGaP\\_Subject\\_ID & dbGaP\\_Sample\\_ID & Visit.y & Gender & Age & ... & Sex & CHROM\\_POS\\_GWAS\\_VISIT\\_base & varID\\_GWASID & dVAF\\_by\\_dT & logdVAF\\_by\\_dT & Variant\\_type & Gene\\_Group\\_R882 & Gene\\_R882 & is\\_notHiSeq & is\\_notMUTECT\\\\\n",
       " <chr> & <int> & <int> & <chr> & <chr> & <int> & <int> & <chr> & <chr> & <int> & ... & <chr> & <chr> & <chr> & <dbl> & <dbl> & <chr> & <chr> & <chr> & <int> & <int>\\\\\n",
       "\\hline\n",
       "\t F100601 & 131179968 & 131179968 & A09377 & A09377\\_V05 & 229137 & 1790886 & V03 & F & 83 & ... & F & chr2\\_25248155\\_A09377\\_V03  & chr2\\_25248155\\_GCAGGAGGGCTGGCCTCCTCCACCTT\\_G\\_A09377 & 0.009573684 & 0.39508378 & INDEL    & DNMT3A\\_nonR882 & DNMT3A\\_nonR882 & 0 & 1\\\\\n",
       "\t F100920 & 131156438 & 131156438 & A14695 & A14695\\_V05 & 235154 & 1793159 & V03 & F & 83 & ... & F & chr2\\_25240714\\_A14695\\_V03  & chr2\\_25240714\\_G\\_A\\_A14695                          & 0.006845000 & 0.36112830 & Missense & DNMT3A\\_nonR882 & DNMT3A\\_nonR882 & 0 & 1\\\\\n",
       "\t F101063 & 131024996 & 131024996 & A14432 & A14432\\_V05 & 227292 & 1336307 & MRI & M & 81 & ... & M & chrX\\_155120076\\_A14432\\_MRI & chrX\\_155120076\\_C\\_T\\_A14432                         & 0.107128571 & 1.27466547 & Stopgain & Other          & BRCC3          & 0 & 1\\\\\n",
       "\t F101108 & 131088751 & 131088751 & A01650 & A01650\\_V05 & 230347 & 1334735 & V02 & M & 83 & ... & M & chr17\\_76736877\\_A01650\\_V02 & chr17\\_76736877\\_G\\_T\\_A01650                         & 0.002359091 & 0.28426495 & Missense & SF             & SRSF2          & 0 & 1\\\\\n",
       "\t F101279 & 131010885 & 131010885 & A11069 & A11069\\_V05 & 224927 & 1335914 & V02 & M & 70 & ... & M & chr2\\_25240420\\_A11069\\_V02  & chr2\\_25240420\\_T\\_C\\_A11069                          & 0.001636364 & 0.03215145 & Missense & DNMT3A\\_nonR882 & DNMT3A\\_nonR882 & 1 & 0\\\\\n",
       "\t F102435 & 131177079 & 131177079 & A16447 & A16447\\_V05 & 234825 & 1336548 & V03 & F & 73 & ... & F & chr2\\_25244340\\_A16447\\_V03  & chr2\\_25244340\\_T\\_C\\_A16447                          & 0.002195000 & 0.30433874 & Splicing & DNMT3A\\_nonR882 & DNMT3A\\_nonR882 & 0 & 1\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.table: 6 x 47\n",
       "\n",
       "| ARIC_ID &lt;chr&gt; | FID &lt;int&gt; | IID &lt;int&gt; | GWAS_ID &lt;chr&gt; | GWASID_Visit &lt;chr&gt; | dbGaP_Subject_ID &lt;int&gt; | dbGaP_Sample_ID &lt;int&gt; | Visit.y &lt;chr&gt; | Gender &lt;chr&gt; | Age &lt;int&gt; | ... ... | Sex &lt;chr&gt; | CHROM_POS_GWAS_VISIT_base &lt;chr&gt; | varID_GWASID &lt;chr&gt; | dVAF_by_dT &lt;dbl&gt; | logdVAF_by_dT &lt;dbl&gt; | Variant_type &lt;chr&gt; | Gene_Group_R882 &lt;chr&gt; | Gene_R882 &lt;chr&gt; | is_notHiSeq &lt;int&gt; | is_notMUTECT &lt;int&gt; |\n",
       "|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|---|\n",
       "| F100601 | 131179968 | 131179968 | A09377 | A09377_V05 | 229137 | 1790886 | V03 | F | 83 | ... | F | chr2_25248155_A09377_V03  | chr2_25248155_GCAGGAGGGCTGGCCTCCTCCACCTT_G_A09377 | 0.009573684 | 0.39508378 | INDEL    | DNMT3A_nonR882 | DNMT3A_nonR882 | 0 | 1 |\n",
       "| F100920 | 131156438 | 131156438 | A14695 | A14695_V05 | 235154 | 1793159 | V03 | F | 83 | ... | F | chr2_25240714_A14695_V03  | chr2_25240714_G_A_A14695                          | 0.006845000 | 0.36112830 | Missense | DNMT3A_nonR882 | DNMT3A_nonR882 | 0 | 1 |\n",
       "| F101063 | 131024996 | 131024996 | A14432 | A14432_V05 | 227292 | 1336307 | MRI | M | 81 | ... | M | chrX_155120076_A14432_MRI | chrX_155120076_C_T_A14432                         | 0.107128571 | 1.27466547 | Stopgain | Other          | BRCC3          | 0 | 1 |\n",
       "| F101108 | 131088751 | 131088751 | A01650 | A01650_V05 | 230347 | 1334735 | V02 | M | 83 | ... | M | chr17_76736877_A01650_V02 | chr17_76736877_G_T_A01650                         | 0.002359091 | 0.28426495 | Missense | SF             | SRSF2          | 0 | 1 |\n",
       "| F101279 | 131010885 | 131010885 | A11069 | A11069_V05 | 224927 | 1335914 | V02 | M | 70 | ... | M | chr2_25240420_A11069_V02  | chr2_25240420_T_C_A11069                          | 0.001636364 | 0.03215145 | Missense | DNMT3A_nonR882 | DNMT3A_nonR882 | 1 | 0 |\n",
       "| F102435 | 131177079 | 131177079 | A16447 | A16447_V05 | 234825 | 1336548 | V03 | F | 73 | ... | F | chr2_25244340_A16447_V03  | chr2_25244340_T_C_A16447                          | 0.002195000 | 0.30433874 | Splicing | DNMT3A_nonR882 | DNMT3A_nonR882 | 0 | 1 |\n",
       "\n"
      ],
      "text/plain": [
       "  ARIC_ID FID       IID       GWAS_ID GWASID_Visit dbGaP_Subject_ID\n",
       "1 F100601 131179968 131179968 A09377  A09377_V05   229137          \n",
       "2 F100920 131156438 131156438 A14695  A14695_V05   235154          \n",
       "3 F101063 131024996 131024996 A14432  A14432_V05   227292          \n",
       "4 F101108 131088751 131088751 A01650  A01650_V05   230347          \n",
       "5 F101279 131010885 131010885 A11069  A11069_V05   224927          \n",
       "6 F102435 131177079 131177079 A16447  A16447_V05   234825          \n",
       "  dbGaP_Sample_ID Visit.y Gender Age ... Sex CHROM_POS_GWAS_VISIT_base\n",
       "1 1790886         V03     F      83  ... F   chr2_25248155_A09377_V03 \n",
       "2 1793159         V03     F      83  ... F   chr2_25240714_A14695_V03 \n",
       "3 1336307         MRI     M      81  ... M   chrX_155120076_A14432_MRI\n",
       "4 1334735         V02     M      83  ... M   chr17_76736877_A01650_V02\n",
       "5 1335914         V02     M      70  ... M   chr2_25240420_A11069_V02 \n",
       "6 1336548         V03     F      73  ... F   chr2_25244340_A16447_V03 \n",
       "  varID_GWASID                                      dVAF_by_dT  logdVAF_by_dT\n",
       "1 chr2_25248155_GCAGGAGGGCTGGCCTCCTCCACCTT_G_A09377 0.009573684 0.39508378   \n",
       "2 chr2_25240714_G_A_A14695                          0.006845000 0.36112830   \n",
       "3 chrX_155120076_C_T_A14432                         0.107128571 1.27466547   \n",
       "4 chr17_76736877_G_T_A01650                         0.002359091 0.28426495   \n",
       "5 chr2_25240420_T_C_A11069                          0.001636364 0.03215145   \n",
       "6 chr2_25244340_T_C_A16447                          0.002195000 0.30433874   \n",
       "  Variant_type Gene_Group_R882 Gene_R882      is_notHiSeq is_notMUTECT\n",
       "1 INDEL        DNMT3A_nonR882  DNMT3A_nonR882 0           1           \n",
       "2 Missense     DNMT3A_nonR882  DNMT3A_nonR882 0           1           \n",
       "3 Stopgain     Other           BRCC3          0           1           \n",
       "4 Missense     SF              SRSF2          0           1           \n",
       "5 Missense     DNMT3A_nonR882  DNMT3A_nonR882 1           0           \n",
       "6 Splicing     DNMT3A_nonR882  DNMT3A_nonR882 0           1           "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "head(grt_aa.dp20)\n",
    "head(grt_ea.dp20)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'aric_baseline_n_v05_aa_gwas'</li><li>'aric_baseline_n_v05_ea_gwas'</li><li>'cln_grt.vaf2.DP20_base.corrected_ordered'</li><li>'cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered'</li><li>'grt_aa.dp20'</li><li>'grt_aa.dp20allAD5FRRR2'</li><li>'grt_ea.dp20'</li><li>'grt_ea.dp20allAD5FRRR2'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'aric\\_baseline\\_n\\_v05\\_aa\\_gwas'\n",
       "\\item 'aric\\_baseline\\_n\\_v05\\_ea\\_gwas'\n",
       "\\item 'cln\\_grt.vaf2.DP20\\_base.corrected\\_ordered'\n",
       "\\item 'cln\\_grt.vaf2.DP20\\_base\\_allAD5FRRR2.corrected\\_ordered'\n",
       "\\item 'grt\\_aa.dp20'\n",
       "\\item 'grt\\_aa.dp20allAD5FRRR2'\n",
       "\\item 'grt\\_ea.dp20'\n",
       "\\item 'grt\\_ea.dp20allAD5FRRR2'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'aric_baseline_n_v05_aa_gwas'\n",
       "2. 'aric_baseline_n_v05_ea_gwas'\n",
       "3. 'cln_grt.vaf2.DP20_base.corrected_ordered'\n",
       "4. 'cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered'\n",
       "5. 'grt_aa.dp20'\n",
       "6. 'grt_aa.dp20allAD5FRRR2'\n",
       "7. 'grt_ea.dp20'\n",
       "8. 'grt_ea.dp20allAD5FRRR2'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"aric_baseline_n_v05_aa_gwas\"                         \n",
       "[2] \"aric_baseline_n_v05_ea_gwas\"                         \n",
       "[3] \"cln_grt.vaf2.DP20_base.corrected_ordered\"            \n",
       "[4] \"cln_grt.vaf2.DP20_base_allAD5FRRR2.corrected_ordered\"\n",
       "[5] \"grt_aa.dp20\"                                         \n",
       "[6] \"grt_aa.dp20allAD5FRRR2\"                              \n",
       "[7] \"grt_ea.dp20\"                                         \n",
       "[8] \"grt_ea.dp20allAD5FRRR2\"                              "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "ls()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "## GWAS_ID==SAMPLE_ID\n",
    "# fwrite(grt_aa.dp20[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_aa.dp20.1Dec2023.csv\", \n",
    "  #      row.names = F, col.names = T, sep=\",\", na=\"NA\")\n",
    "\n",
    "# fwrite(grt_aa.dp20[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_aa.dp20.1Dec2023.tsv\", \n",
    "  #      row.names = F, col.names = T, sep=\"\\t\", na=\"NA\", quote = F)\n",
    "\n",
    "# fwrite(grt_ea.dp20[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_ea.dp20.1Dec2023.csv\", \n",
    "  #      row.names = F, col.names = T, sep=\",\", na=\"NA\")\n",
    "\n",
    "# fwrite(grt_ea.dp20[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_ea.dp20.1Dec2023.tsv\", \n",
    "  #      row.names = F, col.names = T, sep=\"\\t\", na=\"NA\", quote = F)\n",
    "\n",
    "# grt_aa.dp20allAD5FRRR2\n",
    "# fwrite(grt_aa.dp20allAD5FRRR2[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_aa.dp20allAD5FRRR2.1Dec2023.csv\", \n",
    "  #      row.names = F, col.names = T, sep=\",\", na=\"NA\")\n",
    "\n",
    "# fwrite(grt_aa.dp20allAD5FRRR2[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_aa.dp20allAD5FRRR2.1Dec2023.tsv\", \n",
    "  #      row.names = F, col.names = T, sep=\"\\t\", na=\"NA\", quote = F)\n",
    "\n",
    "# fwrite(grt_ea.dp20allAD5FRRR2[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_ea.dp20allAD5FRRR2.1Dec2023.csv\", \n",
    "  #      row.names = F, col.names = T, sep=\",\", na=\"NA\")\n",
    "\n",
    "# fwrite(grt_ea.dp20allAD5FRRR2[,c(2,3,1,4:47)], \"/medpop/esp2/mesbah/projects/ch_progression/aric/gwas/grt_ea.dp20allAD5FRRR2.1Dec2023.tsv\", \n",
    " #       row.names = F, col.names = T, sep=\"\\t\", na=\"NA\", quote = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'FID'</li><li>'IID'</li><li>'ARIC_ID'</li><li>'GWAS_ID'</li><li>'GWASID_Visit'</li><li>'dbGaP_Subject_ID'</li><li>'dbGaP_Sample_ID'</li><li>'Visit.y'</li><li>'Gender'</li><li>'Age'</li><li>'CH_baseline'</li><li>'CH_v05'</li><li>'incident_CH'</li><li>'incident_CHvaf05'</li><li>'incident_CHvaf10'</li><li>'incident_DNMT3A'</li><li>'incident_TET2'</li><li>'incident_ASXL1'</li><li>'incident_DTA'</li><li>'incident_SF'</li><li>'incident_DDR'</li><li>'incident_CH_or_growingClones'</li><li>'age_base'</li><li>'age_base_sqr'</li><li>'Center'</li><li>'PC1'</li><li>'PC2'</li><li>'PC3'</li><li>'PC4'</li><li>'PC5'</li><li>'PC6'</li><li>'PC7'</li><li>'PC8'</li><li>'PC9'</li><li>'PC10'</li><li>'ever_smoke'</li><li>'v2_vs_other'</li><li>'Sex'</li><li>'CHROM_POS_GWAS_VISIT_base'</li><li>'varID_GWASID'</li><li>'dVAF_by_dT'</li><li>'logdVAF_by_dT'</li><li>'Variant_type'</li><li>'Gene_Group_R882'</li><li>'Gene_R882'</li><li>'is_notHiSeq'</li><li>'is_notMUTECT'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item 'FID'\n",
       "\\item 'IID'\n",
       "\\item 'ARIC\\_ID'\n",
       "\\item 'GWAS\\_ID'\n",
       "\\item 'GWASID\\_Visit'\n",
       "\\item 'dbGaP\\_Subject\\_ID'\n",
       "\\item 'dbGaP\\_Sample\\_ID'\n",
       "\\item 'Visit.y'\n",
       "\\item 'Gender'\n",
       "\\item 'Age'\n",
       "\\item 'CH\\_baseline'\n",
       "\\item 'CH\\_v05'\n",
       "\\item 'incident\\_CH'\n",
       "\\item 'incident\\_CHvaf05'\n",
       "\\item 'incident\\_CHvaf10'\n",
       "\\item 'incident\\_DNMT3A'\n",
       "\\item 'incident\\_TET2'\n",
       "\\item 'incident\\_ASXL1'\n",
       "\\item 'incident\\_DTA'\n",
       "\\item 'incident\\_SF'\n",
       "\\item 'incident\\_DDR'\n",
       "\\item 'incident\\_CH\\_or\\_growingClones'\n",
       "\\item 'age\\_base'\n",
       "\\item 'age\\_base\\_sqr'\n",
       "\\item 'Center'\n",
       "\\item 'PC1'\n",
       "\\item 'PC2'\n",
       "\\item 'PC3'\n",
       "\\item 'PC4'\n",
       "\\item 'PC5'\n",
       "\\item 'PC6'\n",
       "\\item 'PC7'\n",
       "\\item 'PC8'\n",
       "\\item 'PC9'\n",
       "\\item 'PC10'\n",
       "\\item 'ever\\_smoke'\n",
       "\\item 'v2\\_vs\\_other'\n",
       "\\item 'Sex'\n",
       "\\item 'CHROM\\_POS\\_GWAS\\_VISIT\\_base'\n",
       "\\item 'varID\\_GWASID'\n",
       "\\item 'dVAF\\_by\\_dT'\n",
       "\\item 'logdVAF\\_by\\_dT'\n",
       "\\item 'Variant\\_type'\n",
       "\\item 'Gene\\_Group\\_R882'\n",
       "\\item 'Gene\\_R882'\n",
       "\\item 'is\\_notHiSeq'\n",
       "\\item 'is\\_notMUTECT'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. 'FID'\n",
       "2. 'IID'\n",
       "3. 'ARIC_ID'\n",
       "4. 'GWAS_ID'\n",
       "5. 'GWASID_Visit'\n",
       "6. 'dbGaP_Subject_ID'\n",
       "7. 'dbGaP_Sample_ID'\n",
       "8. 'Visit.y'\n",
       "9. 'Gender'\n",
       "10. 'Age'\n",
       "11. 'CH_baseline'\n",
       "12. 'CH_v05'\n",
       "13. 'incident_CH'\n",
       "14. 'incident_CHvaf05'\n",
       "15. 'incident_CHvaf10'\n",
       "16. 'incident_DNMT3A'\n",
       "17. 'incident_TET2'\n",
       "18. 'incident_ASXL1'\n",
       "19. 'incident_DTA'\n",
       "20. 'incident_SF'\n",
       "21. 'incident_DDR'\n",
       "22. 'incident_CH_or_growingClones'\n",
       "23. 'age_base'\n",
       "24. 'age_base_sqr'\n",
       "25. 'Center'\n",
       "26. 'PC1'\n",
       "27. 'PC2'\n",
       "28. 'PC3'\n",
       "29. 'PC4'\n",
       "30. 'PC5'\n",
       "31. 'PC6'\n",
       "32. 'PC7'\n",
       "33. 'PC8'\n",
       "34. 'PC9'\n",
       "35. 'PC10'\n",
       "36. 'ever_smoke'\n",
       "37. 'v2_vs_other'\n",
       "38. 'Sex'\n",
       "39. 'CHROM_POS_GWAS_VISIT_base'\n",
       "40. 'varID_GWASID'\n",
       "41. 'dVAF_by_dT'\n",
       "42. 'logdVAF_by_dT'\n",
       "43. 'Variant_type'\n",
       "44. 'Gene_Group_R882'\n",
       "45. 'Gene_R882'\n",
       "46. 'is_notHiSeq'\n",
       "47. 'is_notMUTECT'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       " [1] \"FID\"                          \"IID\"                         \n",
       " [3] \"ARIC_ID\"                      \"GWAS_ID\"                     \n",
       " [5] \"GWASID_Visit\"                 \"dbGaP_Subject_ID\"            \n",
       " [7] \"dbGaP_Sample_ID\"              \"Visit.y\"                     \n",
       " [9] \"Gender\"                       \"Age\"                         \n",
       "[11] \"CH_baseline\"                  \"CH_v05\"                      \n",
       "[13] \"incident_CH\"                  \"incident_CHvaf05\"            \n",
       "[15] \"incident_CHvaf10\"             \"incident_DNMT3A\"             \n",
       "[17] \"incident_TET2\"                \"incident_ASXL1\"              \n",
       "[19] \"incident_DTA\"                 \"incident_SF\"                 \n",
       "[21] \"incident_DDR\"                 \"incident_CH_or_growingClones\"\n",
       "[23] \"age_base\"                     \"age_base_sqr\"                \n",
       "[25] \"Center\"                       \"PC1\"                         \n",
       "[27] \"PC2\"                          \"PC3\"                         \n",
       "[29] \"PC4\"                          \"PC5\"                         \n",
       "[31] \"PC6\"                          \"PC7\"                         \n",
       "[33] \"PC8\"                          \"PC9\"                         \n",
       "[35] \"PC10\"                         \"ever_smoke\"                  \n",
       "[37] \"v2_vs_other\"                  \"Sex\"                         \n",
       "[39] \"CHROM_POS_GWAS_VISIT_base\"    \"varID_GWASID\"                \n",
       "[41] \"dVAF_by_dT\"                   \"logdVAF_by_dT\"               \n",
       "[43] \"Variant_type\"                 \"Gene_Group_R882\"             \n",
       "[45] \"Gene_R882\"                    \"is_notHiSeq\"                 \n",
       "[47] \"is_notMUTECT\"                "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "names(grt_ea.dp20[,c(2,3,1,4:47)])"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
