{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8e568eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)\n",
    "library(stringr)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(ggpubr)\n",
    "library(cowplot)\n",
    "library(ggridges)\n",
    "theme_set(theme_cowplot())\n",
    "library(viridis)\n",
    "\n",
    "## \n",
    "setwd(\"/medpop/esp2/mesbah/projects/ch_progression/aric/pheno/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c998b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# qcd CH variants with phenotypes\n",
    "ch_var_in_v05_qcd.pheno <- fread(\"ch_var_in_v05_qcd_pheno.23Mar2023.csv\", header = T)\n",
    "\n",
    "# aric_baseline_n_v05 <- fread(\"aric_baseline_n_v05_N4189.pheno_ch_status.9May2023.csv\", header=T)\n",
    "aric_baseline_n_v05_noPrevHeme <- fread(\"aric_baseline_n_v05_N4187.pheno_ch_status.noHemeCA.9May2023.csv\", header=T)\n",
    "# \n",
    "aric_baseline_n_v05_noPrevHeme$dAge <- aric_baseline_n_v05_noPrevHeme$Age - aric_baseline_n_v05_noPrevHeme$age_base\n",
    "summary(aric_baseline_n_v05_noPrevHeme$dAge)\n",
    "\n",
    "## Baseline pop\n",
    "aric_baseline_full_noPrevHeme <-  fread(\"aric_baseline_N10881.pheno_ch_status_bp_heme.noHemeCA.10May2023.csv\", \n",
    "                                        header=T)\n",
    "##     # qcd CH variants with phenotypes\n",
    "ch_var_in_baseline_qcd.pheno <- fread(\"ch_var_in_baseline_qcd_pheno.23Mar2023.csv\", header = T)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "898b8ca2",
   "metadata": {},
   "outputs": [],
   "source": [
    "nrow(ch_var_in_baseline_qcd.pheno)\n",
    "nrow(ch_var_in_v05_qcd.pheno)\n",
    "\n",
    "str(ch_var_in_baseline_qcd.pheno[,c(23,9,16:21)])\n",
    "cat(\"\\nvisit 5\\n\")\n",
    "cat(\"\\n\\n\")\n",
    "str(ch_var_in_v05_qcd.pheno[,c(22,9,16:21)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5188b806",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(base_var_table); \n",
    "\n",
    "base_var_table <- ch_var_in_baseline_qcd.pheno[,c(23,9,16:21)] %>% \n",
    "                                               group_by(varID) %>% \n",
    "                                               summarise(Number_of_Occurance = n(), \n",
    "                                                         across(everything(), \n",
    "                                                                unique))\n",
    "                                               \n",
    "                                               \n",
    "str(base_var_table)\n",
    "\n",
    "fwrite(base_var_table, \"Tables.base_var_table.csv\", sep=\",\", row.names = F, col.names = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a84d489",
   "metadata": {},
   "outputs": [],
   "source": [
    "v5_var_table <- ch_var_in_v05_qcd.pheno[,c(22,5:9,16:21)] %>% \n",
    "                                               group_by(varID) %>% \n",
    "                                               summarise(Number_of_Occurance = n(), \n",
    "                                                         across(everything(), \n",
    "                                                                unique))\n",
    "                                               \n",
    "                                               \n",
    "str(v5_var_table)\n",
    "\n",
    "fwrite(v5_var_table, \"TableS.v5_var_table.csv\", sep=\",\", row.names = F, col.names = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef1c04b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(unique(v5_var_table$varID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0942eeaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "sum(v5_var_table$Number_of_Occurance)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf212460",
   "metadata": {},
   "outputs": [],
   "source": [
    "str(ch_var_in_v05_qcd.pheno[,c(1:30)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "556db588",
   "metadata": {},
   "outputs": [],
   "source": [
    "aric_baseline_n_v05_noPrevHeme$CHvaf10_v05 <- ifelse(aric_baseline_n_v05_noPrevHeme$CH_v05==1 & \n",
    "                                                      aric_baseline_n_v05_noPrevHeme$GWAS_ID %in% \n",
    "                                                      ch_var_in_v05_qcd.pheno$GWAS_ID[ch_var_in_v05_qcd.pheno$VAF>=0.10], \n",
    "                                                      1, 0)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_v05)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CHvaf10_v05)\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_v05,aric_baseline_n_v05_noPrevHeme$CHvaf10_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7423f420",
   "metadata": {},
   "outputs": [],
   "source": [
    "nrow(aric_baseline_full_noPrevHeme)\n",
    "table(ch_var_in_baseline_qcd.pheno$GWAS_ID %in% aric_baseline_full_noPrevHeme$gwasid, exclude = NULL)\n",
    "\n",
    "table(ch_var_in_baseline_qcd.pheno$GWAS_ID %in% aric_baseline_n_v05_noPrevHeme$GWAS_ID, exclude = NULL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3221a41",
   "metadata": {},
   "outputs": [],
   "source": [
    "table(aric_baseline_full_noPrevHeme$CH_baseline, exclude = NULL)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_baseline, exclude= NULL)\n",
    "\n",
    "round(prop.table(table(aric_baseline_full_noPrevHeme$CH_baseline, exclude = NULL))*100,1)\n",
    "round(prop.table(table(aric_baseline_n_v05_noPrevHeme$CH_baseline, exclude= NULL))*100,1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fea6e21e",
   "metadata": {},
   "outputs": [],
   "source": [
    "aric_baseline_full_noPrevHeme$Follow_up <- ifelse(aric_baseline_full_noPrevHeme$gwasid %in% aric_baseline_n_v05_noPrevHeme$GWAS_ID, \"w/ follow-up\", \"w/o follow-up\")\n",
    "table(aric_baseline_full_noPrevHeme$Follow_up)\n",
    "summary(aric_baseline_full_noPrevHeme$age_base)\n",
    "nrow(aric_baseline_n_v05_noPrevHeme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1095130",
   "metadata": {},
   "outputs": [],
   "source": [
    "###### group by\n",
    "# library(dplyr)\n",
    "ch_var_base_count <- ch_var_in_baseline_qcd.pheno[,c(1,9,11)] %>% \n",
    "  group_by(GWAS_ID) %>% \n",
    "  summarise(nCHIP=n(), \n",
    "            Gene=paste(Gene,collapse = \";\"), \n",
    "            VAF=paste(round(VAF,3),collapse = \";\")\n",
    "  )\n",
    "\n",
    "ch_var_base_count %>% filter(nCHIP==2) %>% arrange(., nCHIP)\n",
    "\n",
    "ch_var_base_count %>% filter(nCHIP>2) %>% arrange(., nCHIP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "161726fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "ch_var_v05_count <- ch_var_in_v05_qcd.pheno[,c(1,9,11)] %>% \n",
    "  group_by(GWAS_ID) %>% \n",
    "  summarise(nCHIP=n(), \n",
    "            Gene=paste(Gene,collapse = \";\"), \n",
    "            VAF=paste(round(VAF,3),collapse = \";\")\n",
    "  )\n",
    "\n",
    "\n",
    "ch_var_v05_count %>% filter(nCHIP==2) %>% arrange(., nCHIP)\n",
    "\n",
    "ch_var_v05_count %>% filter(nCHIP>2) %>% arrange(., nCHIP)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "07b6ae81",
   "metadata": {},
   "outputs": [],
   "source": [
    "####### Fig 1a-h: baseline vs. follow-up visit CH\n",
    "#### Fig a ####\n",
    "## Base: a,b,c,d\n",
    "\n",
    "# Prev. at baseline\n",
    "a <- ggplot(data=aric_baseline_n_v05_noPrevHeme, aes(x=age_base), group=1) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "    geom_smooth(aes(y=CH_baseline, colour=\"VAF>=2%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "    geom_smooth(aes(y=CHvaf10_baseline, colour=\"VAF>=10%\"), \n",
    "             method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "    # geom_smooth(aes(y=CH_baseline, colour=\"VAF>=2%\"), method =\"gam\") +\n",
    "    # geom_smooth(aes(y=CHvaf10_baseline, colour=\"VAF>=10%\"), method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  +\n",
    "scale_y_continuous(breaks = c(0,0.05,0.1, 0.15, seq(.1,.5,.1)) )\n",
    "\n",
    "## Prev. at Follow-up\n",
    "e <- ggplot(data=aric_baseline_n_v05_noPrevHeme, aes(x=Age), group=1) + \n",
    "  xlab(\"Age (follow-up)\") + ylab(\"Prevalence\") +\n",
    "    geom_smooth(aes(y=CH_v05, colour=\"VAF>=2%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "    geom_smooth(aes(y=CHvaf10_v05, colour=\"VAF>=10%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "    # geom_smooth(aes(y=CH_v05, colour=\"VAF>=2%\"), method =\"gam\") +\n",
    "    # geom_smooth(aes(y=CHvaf10_v05, colour=\"VAF>=10%\"), method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"e\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) + \n",
    "scale_y_continuous(breaks = c(0,0.05,0.1, 0.15, seq(.1,.5,.1)) )\n",
    "a\n",
    "e"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "572cdb5c",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Gene distributions by age bin\n",
    "\n",
    "### baseline\n",
    "nrow(ch_var_in_baseline_qcd.pheno)\n",
    "ch_var_in_baseline_qcd.pheno.followup <- subset(ch_var_in_baseline_qcd.pheno, \n",
    "                                                ch_var_in_baseline_qcd.pheno$GWAS_ID %in% \n",
    "                                                aric_baseline_n_v05_noPrevHeme$GWAS_ID)\n",
    "nrow(ch_var_in_baseline_qcd.pheno.followup)\n",
    "\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_bin <- ifelse(ch_var_in_baseline_qcd.pheno$age_base<50,\"<50\\n(n=45)\",\n",
    "                                               ifelse(ch_var_in_baseline_qcd.pheno$age_base>=50 &  \n",
    "                                                        ch_var_in_baseline_qcd.pheno$age_base<55,\n",
    "                                                      \"50-55\\n(n=166)\", \n",
    "                                                      ifelse(ch_var_in_baseline_qcd.pheno$age_base>=55 &\n",
    "                                                               ch_var_in_baseline_qcd.pheno$age_base<60,\n",
    "                                                             \"55-60\\n(n=162)\", \n",
    "                                                             ifelse(ch_var_in_baseline_qcd.pheno$age_base>=60 & \n",
    "                                                                      ch_var_in_baseline_qcd.pheno$age_base<65,\n",
    "                                                                    \"60-65\\n(n=122)\",\n",
    "                                                                    ifelse(ch_var_in_baseline_qcd.pheno$age_base>=65 & \n",
    "                                                                             ch_var_in_baseline_qcd.pheno$age_base<70,\n",
    "                                                                           \"65-70\\n(n=67)\",\">=70\\n(n=14)\")))))\n",
    "\n",
    "table(ch_var_in_baseline_qcd.pheno.followup$Age_bin)\n",
    "    # order\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_bin <- factor(ch_var_in_baseline_qcd.pheno.followup$Age_bin,\n",
    "                                          levels = c(\"<50\\n(n=45)\", \"50-55\\n(n=166)\", \"55-60\\n(n=162)\",\n",
    "                                                     \"60-65\\n(n=122)\", \"65-70\\n(n=67)\", \">=70\\n(n=14)\"),\n",
    "                                          ordered = T)\n",
    "\n",
    "ch_var_in_baseline_qcd.pheno.followup$Gene_cat <- ifelse(ch_var_in_baseline_qcd.pheno.followup$Gene %in% \n",
    "                             c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "                               \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "                               \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "                               \"ZRSR2\"), \n",
    "                           as.character(ch_var_in_baseline_qcd.pheno.followup$Gene), \"Other\")\n",
    "\n",
    "\n",
    "# ch_base <- ch_var_in_baseline_qcd.pheno.followup[,c(9,128)] %>% group_by(Age_bin) \n",
    "ch_base <- ch_var_in_baseline_qcd.pheno.followup[,c(129,128)] %>% group_by(Age_bin) \n",
    "\n",
    "ch_base <- as.data.frame(prop.table(table(ch_base),margin = 2)*100)\n",
    "\n",
    "# ch_base$Gene_cat<- ifelse(ch_base$Gene %in% \n",
    "  #                            c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "   #                             \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "    #                            \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "     #                           \"ZRSR2\"), \n",
    "      #                      as.character(ch_base$Gene), \"Other\")\n",
    "\n",
    "## order: https://stackoverflow.com/questions/9227389/ordering-stacks-by-size-in-a-ggplot2-stacked-bar-graph\n",
    "ch_base$Gene_cat <- reorder(ch_base$Gene_cat, ch_base$Freq)\n",
    "# ch_base$Gene_cat <- factor(ch_base$Gene_cat, \n",
    "#                            levels = rev(levels(ch_base$Gene_cat)))\n",
    "\n",
    "## plot b\n",
    "b <- ch_base %>% filter(Freq>0) %>% ggplot(., \n",
    "                                     aes(fill=Gene_cat, \n",
    "                                         y=Freq, x=Age_bin)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion (%)\") + xlab(\"Age (baseline)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "scale_color_brewer(type = \"div\", \n",
    "                     palette = \"RdYlBu\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"b\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  \n",
    "b\n",
    "\n",
    "\n",
    "###\n",
    "## follow-up\n",
    "ch_var_in_v05_qcd.pheno$Age_bin <- ifelse(ch_var_in_v05_qcd.pheno$Age<70,\"<70\\n(n=91)\",\n",
    "                                          ifelse(ch_var_in_v05_qcd.pheno$Age>=70 &  \n",
    "                                                   ch_var_in_v05_qcd.pheno$Age<75,\n",
    "                                                 \"70-75\\n(n=376)\", \n",
    "                                                 ifelse(ch_var_in_v05_qcd.pheno$Age>=75 &\n",
    "                                                          ch_var_in_v05_qcd.pheno$Age<80,\n",
    "                                                        \"75-80\\n(n=368)\", \n",
    "                                                        ifelse(ch_var_in_v05_qcd.pheno$Age>=80 & \n",
    "                                                                 ch_var_in_v05_qcd.pheno$Age<85,\n",
    "                                                               \"80-85\\n(n=316)\",\"85-90\\n(n=151)\"))))\n",
    "\n",
    "ch_var_in_v05_qcd.pheno$Gene_cat <- ifelse(ch_var_in_v05_qcd.pheno$Gene %in% \n",
    "                            c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "                              \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "                              \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "                              \"ZRSR2\"), \n",
    "                          as.character(ch_var_in_v05_qcd.pheno$Gene), \"Other\")\n",
    "\n",
    "\n",
    "table(ch_var_in_v05_qcd.pheno$Age_bin)\n",
    "    # order\n",
    "ch_var_in_v05_qcd.pheno$Age_bin <- factor(ch_var_in_v05_qcd.pheno$Age_bin, \n",
    "                                          levels = c(\"<70\\n(n=91)\", \n",
    "                                                     \"70-75\\n(n=376)\", \n",
    "                                                     \"75-80\\n(n=368)\",\n",
    "                                                      \"80-85\\n(n=316)\", \n",
    "                                                     \"85-90\\n(n=151)\"), \n",
    "                                          ordered = T)\n",
    "\n",
    "# ch_v05 <- ch_var_in_v05_qcd.pheno[,c(9,127)] %>% group_by(Age_bin)\n",
    "ch_v05 <- ch_var_in_v05_qcd.pheno[,c(128,127)] %>% group_by(Age_bin) \n",
    "ch_v05 <- as.data.frame(prop.table(table(ch_v05),margin = 2)*100)\n",
    "\n",
    "# ch_v05$Gene_cat <- ifelse(ch_v05$Gene %in% \n",
    "  #                         c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "   #                          \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "    #                         \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "     #                         \"ZRSR2\"), \n",
    "      #                   as.character(ch_v05$Gene), \"Other\")\n",
    "ch_v05$Gene_cat <- reorder(ch_v05$Gene_cat, ch_v05$Freq)\n",
    "\n",
    "## plot f\n",
    "f <- ch_v05 %>% filter(Freq>0) %>% ggplot(., \n",
    "                                     aes(fill=Gene_cat, \n",
    "                                         y=Freq, x=Age_bin)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion (%)\") + xlab(\"Age (follow-up)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "scale_color_brewer(type = \"div\", \n",
    "                     palette = \"RdYlBu\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"f\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  \n",
    "f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1a21bc96",
   "metadata": {},
   "outputs": [],
   "source": [
    "aric_baseline_n_v05_noPrevHeme$Age_bin_base <- ifelse(aric_baseline_n_v05_noPrevHeme$age_base<50,\"<50\\n(n=587)\",\n",
    "                                               ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=50 &  \n",
    "                                                        aric_baseline_n_v05_noPrevHeme$age_base<55,\n",
    "                                                      \"50-55\\n(n=1478)\", \n",
    "                                                      ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=55 &\n",
    "                                                               aric_baseline_n_v05_noPrevHeme$age_base<60,\n",
    "                                                             \"55-60\\n(n=1127)\", \n",
    "                                                             ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=60 & \n",
    "                                                                      aric_baseline_n_v05_noPrevHeme$age_base<65,\n",
    "                                                                    \"60-65\\n(n=686)\",\n",
    "                                                                    ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=65 & \n",
    "                                                                             aric_baseline_n_v05_noPrevHeme$age_base<70,\n",
    "                                                                           \"65-70\\n(n=270)\",\">=70\\n(n=39)\")))))\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$Age_bin_base)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ada61710",
   "metadata": {},
   "outputs": [],
   "source": [
    "aric_baseline_n_v05_noPrevHeme$Age_bin_v05 <- ifelse(aric_baseline_n_v05_noPrevHeme$Age<70,\"<70\\n(n=416)\",\n",
    "                                          ifelse(aric_baseline_n_v05_noPrevHeme$Age>=70 &  \n",
    "                                                   aric_baseline_n_v05_noPrevHeme$Age<75,\n",
    "                                                 \"70-75\\n(n=1542)\", \n",
    "                                                 ifelse(aric_baseline_n_v05_noPrevHeme$Age>=75 &\n",
    "                                                          aric_baseline_n_v05_noPrevHeme$Age<80,\n",
    "                                                        \"75-80\\n(n=1178)\", \n",
    "                                                        ifelse(aric_baseline_n_v05_noPrevHeme$Age>=80 & \n",
    "                                                                 aric_baseline_n_v05_noPrevHeme$Age<85,\n",
    "                                                               \"80-85\\n(n=753)\",\"85-90\\n(n=298)\"))))\n",
    "\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$Age_bin_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6d1638e9",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Baseline\n",
    "## Count\n",
    "ch_base.ncount <- as.data.frame(table(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID), \n",
    "                               stringsAsFactors = F)\n",
    "names(ch_base.ncount) <- c(\"GWAS_ID\", \"nCH_base\")\n",
    "ch_v05.ncount <- as.data.frame(table(ch_var_in_v05_qcd.pheno$GWAS_ID), stringsAsFactors = F)\n",
    "names(ch_v05.ncount) <- c(\"GWAS_ID\", \"nCH_v05\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "56a47785",
   "metadata": {},
   "outputs": [],
   "source": [
    "nCH_base_v05 <- merge(aric_baseline_n_v05_noPrevHeme[,c(2,115,116)], ch_base.ncount, by=\"GWAS_ID\", all.x=T)\n",
    "nCH_base_v05 <- merge(nCH_base_v05, ch_v05.ncount, by=\"GWAS_ID\", all.x=T)\n",
    "\n",
    "head(nCH_base_v05)\n",
    "table(nCH_base_v05$nCH_base, exclude = NULL)\n",
    "table(nCH_base_v05$nCH_v05, exclude = NULL)\n",
    "\n",
    "nCH_base_v05$nCH_base[is.na(nCH_base_v05$nCH_base)] <- 0\n",
    "nCH_base_v05$nCH_v05[is.na(nCH_base_v05$nCH_v05)] <- 0\n",
    "table(nCH_base_v05$nCH_base, exclude = NULL)\n",
    "table(nCH_base_v05$nCH_v05, exclude = NULL)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c8a1ca92",
   "metadata": {},
   "outputs": [],
   "source": [
    "nCH_base_v05$nCH_base[nCH_base_v05$nCH_base>=5] <- \"5+\"\n",
    "table(nCH_base_v05$nCH_base)\n",
    "\n",
    "nCH_base_v05$nCH_v05[nCH_base_v05$nCH_v05>=5] <- \"5+\"\n",
    "table(nCH_base_v05$nCH_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9444a8c9",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(nCH_base_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57e6aaab",
   "metadata": {},
   "outputs": [],
   "source": [
    "ch_base.ncount <- as.data.frame(prop.table(table(nCH_base_v05[,c(2,4)]),margin = 1)*100)\n",
    "ch_base.ncount\n",
    "ch_v05.ncount <- as.data.frame(prop.table(table(nCH_base_v05[,c(3,5)]),margin = 1)*100)\n",
    "ch_v05.ncount"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6a639103",
   "metadata": {},
   "outputs": [],
   "source": [
    "table(nCH_base_v05$Age_bin_base)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f9bec582",
   "metadata": {},
   "outputs": [],
   "source": [
    "    # Base\n",
    "ch_base.ncount$Age_bin_base <- factor(ch_base.ncount$Age_bin_base,\n",
    "                                          levels = c(\"<50\\n(n=587)\", \"50-55\\n(n=1478)\", \"55-60\\n(n=1127)\", \n",
    "                                                     \"60-65\\n(n=686)\", \"65-70\\n(n=270)\", \">=70\\n(n=39)\"),\n",
    "                                          ordered = T)\n",
    "# ch_base.ncount$nCH_base <- reorder(ch_base.ncount$nCH_base, ch_base.ncount$Freq)\n",
    "\n",
    "ch_base.ncount$nCH_base <- factor(ch_base.ncount$nCH_base, levels = c(\"0\", \"1\", \"2\", \"3\", \"4\", \"5+\"))\n",
    "\n",
    "ch_base.ncount$nCH_base <- factor(ch_base.ncount$nCH_base, levels = rev(levels(ch_base.ncount$nCH_base)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57a0b2f6",
   "metadata": {},
   "outputs": [],
   "source": [
    "table(nCH_base_v05$Age_bin_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba5e3c72",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    # V05\n",
    "ch_v05.ncount$Age_bin_v05 <- factor(ch_v05.ncount$Age_bin_v05,\n",
    "                                          levels = c(\"<70\\n(n=416)\", \n",
    "                                                     \"70-75\\n(n=1542)\", \n",
    "                                                     \"75-80\\n(n=1178)\",\n",
    "                                                      \"80-85\\n(n=753)\", \n",
    "                                                     \"85-90\\n(n=298)\"), \n",
    "                                          ordered = T)\n",
    "# ch_base.ncount$nCH_base <- reorder(ch_base.ncount$nCH_base, ch_base.ncount$Freq)\n",
    "\n",
    "ch_v05.ncount$nCH_v05 <- factor(ch_v05.ncount$nCH_v05, levels = c(\"0\", \"1\", \"2\", \"3\", \"4\", \"5+\"))\n",
    "\n",
    "ch_v05.ncount$nCH_v05 <- factor(ch_v05.ncount$nCH_v05, levels = rev(levels(ch_v05.ncount$nCH_v05)))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01853016",
   "metadata": {},
   "outputs": [],
   "source": [
    "## c\n",
    "c <- ch_base.ncount %>% filter(Freq>0) %>% ggplot(., \n",
    "                                              aes(fill=nCH_base, \n",
    "                                                  y=Freq, \n",
    "                                                  x=Age_bin_base)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion\") + xlab(\"Age (baseline)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "scale_color_brewer(type = \"div\", \n",
    "                     palette = \"RdBu\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"c\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) \n",
    "\n",
    "## g\n",
    "g <- ch_v05.ncount %>% filter(Freq>0) %>% ggplot(., \n",
    "                                     aes(fill=nCH_v05, \n",
    "                                         y=Freq, \n",
    "                                         x=Age_bin_v05)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion\") + xlab(\"Age (follow-up)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "scale_color_brewer(type = \"div\", \n",
    "                     palette = \"RdBu\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"g\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) \n",
    "\n",
    "c\n",
    "g\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "680d9005",
   "metadata": {},
   "outputs": [],
   "source": [
    "### # of CH per sample\n",
    "# length(unique(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID))\n",
    "\n",
    "# length(unique(ch_var_in_v05_qcd.pheno$GWAS_ID))\n",
    "\n",
    "\n",
    "## Baseline\n",
    "## Count\n",
    "# ch_base.ncount <- as.data.frame(table(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID), \n",
    "#                               stringsAsFactors = F)\n",
    "# ch_base.ncount <- merge(ch_base.ncount, \n",
    "  #                     ch_var_in_baseline_qcd.pheno.followup[!duplicated(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID),\n",
    "  #                                                           c(1,79)], \n",
    "  #                     by.x = \"Var1\", \n",
    "  #                     by.y = \"GWAS_ID\")\n",
    "\n",
    "# ch_base.ncount$Age_bin <- ifelse(ch_base.ncount$age_base<50,\"<50\\n(n=42)\",\n",
    "#                                               ifelse(ch_base.ncount$age_base>=50 &  \n",
    "#                                                        ch_base.ncount$age_base<55,\n",
    "#                                                      \"50-55\\n(n=131)\", \n",
    "#                                                      ifelse(ch_base.ncount$age_base>=55 &\n",
    "#                                                               ch_base.ncount$age_base<60,\n",
    "#                                                             \"55-60\\n(n=127)\", \n",
    "#                                                             ifelse(ch_base.ncount$age_base>=60 & \n",
    "#                                                                      ch_base.ncount$age_base<65,\n",
    "#                                                                    \"60-65\\n(n=98)\",\n",
    "#                                                                    ifelse(ch_base.ncount$age_base>=65 & \n",
    "#                                                                             ch_base.ncount$age_base<70,\n",
    "#                                                                           \"65-70\\n(n=48)\",\">=70\\n(n=11)\")))))\n",
    "# table(ch_base.ncount$Age_bin)\n",
    "    # order\n",
    "# ch_base.ncount$Age_bin <- factor(ch_base.ncount$Age_bin,\n",
    "#                                          levels = c(\"<50\\n(n=42)\", \"50-55\\n(n=131)\", \"55-60\\n(n=127)\", \n",
    "#                                                     \"60-65\\n(n=98)\", \"65-70\\n(n=48)\", \">=70\\n(n=11)\"),\n",
    "#                                          ordered = T)\n",
    "\n",
    "\n",
    "\n",
    "# ch_base.ncount <- as.data.frame(prop.table(table(ch_base.ncount[,c(2,4)]),margin = 2)*100)\n",
    "\n",
    "## c\n",
    "# c <- ch_base.ncount %>% filter(Freq.1>0) %>% ggplot(., \n",
    "#                                              aes(fill=Freq, \n",
    "#                                                  y=Freq.1, \n",
    "#                                                  x=Age_bin)) + \n",
    "#  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "#  ylab(\"Proportion\") + xlab(\"Age (baseline)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "#  theme(legend.title = element_blank(), \n",
    "#        legend.position = \"\") +\n",
    "#  ggtitle(\"c\") + \n",
    "#  theme(legend.position = \"right\", \n",
    "#        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "#        axis.text=element_text(size=12, face=\"bold\"),\n",
    "#        axis.title=element_text(size=14,face=\"bold\")) \n",
    "\n",
    "\n",
    "## Folow-up\n",
    "# ch_v05.ncount <- as.data.frame(table(ch_var_in_v05_qcd.pheno$GWAS_ID), stringsAsFactors = F)\n",
    "\n",
    "# ch_v05.ncount <- merge(ch_v05.ncount, \n",
    "#                        ch_var_in_v05_qcd.pheno[!duplicated(ch_var_in_v05_qcd.pheno$GWAS_ID),c(1,30)], \n",
    " #                      by.x = \"Var1\", \n",
    " #                      by.y = \"GWAS_ID\")\n",
    "\n",
    "# ch_v05.ncount$Age_bin <- ifelse(ch_v05.ncount$Age<70,\"<70\\n(n=83)\",\n",
    "#                                          ifelse(ch_v05.ncount$Age>=70 &  \n",
    "#                                                   ch_v05.ncount$Age<75,\n",
    "#                                                 \"70-75\\n(n=305)\", \n",
    "#                                                 ifelse(ch_v05.ncount$Age>=75 &\n",
    "#                                                          ch_v05.ncount$Age<80,\n",
    "#                                                        \"75-80\\n(n=304)\", \n",
    "#                                                        ifelse(ch_v05.ncount$Age>=80 & \n",
    "#                                                                 ch_v05.ncount$Age<85,\n",
    "#                                                               \"80-85\\n(n=242)\",\"85-90\\n(n=113)\"))))\n",
    "\n",
    "# table(ch_v05.ncount$Age_bin)\n",
    "   # order\n",
    "# ch_v05.ncount$Age_bin <- factor(ch_v05.ncount$Age_bin, \n",
    "#                                          levels = c(\"<70\\n(n=83)\", \n",
    "#                                                     \"70-75\\n(n=305)\", \n",
    "#                                                     \"75-80\\n(n=304)\",\n",
    "#                                                      \"80-85\\n(n=242)\", \n",
    "#                                                     \"85-90\\n(n=113)\"), \n",
    "#                                          ordered = T)\n",
    "\n",
    "# ch_v05.ncount <- as.data.frame(prop.table(table(ch_v05.ncount[,c(2,4)]),margin = 2)*100)\n",
    "\n",
    "## g\n",
    "# g <- ch_v05.ncount %>% filter(Freq.1>0) %>% ggplot(., \n",
    "#                                     aes(fill=Freq, \n",
    "#                                         y=Freq.1, \n",
    "#                                         x=Age_bin)) + \n",
    "#  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "#  ylab(\"Proportion\") + xlab(\"Age (follow-up)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "#  theme(legend.title = element_blank(), \n",
    "#        legend.position = \"\") +\n",
    "#  ggtitle(\"g\") + \n",
    "#  theme(legend.position = \"right\", \n",
    "#        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "#        axis.text=element_text(size=12, face=\"bold\"),\n",
    "#        axis.title=element_text(size=14,face=\"bold\")) \n",
    "#\n",
    "# c\n",
    "# g\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "28053d09",
   "metadata": {},
   "outputs": [],
   "source": [
    "## remove every thing after \"\\n\"\n",
    "## gsub(\"stringto_find..*\", \"relaplcewith\", my_input_string)\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_Bin <- gsub(pattern = \"\\n..*\",\n",
    "                                                      replacement = \"\", \n",
    "                                                      x = ch_var_in_baseline_qcd.pheno.followup$Age_bin)\n",
    "\n",
    "ch_var_in_v05_qcd.pheno$Age_Bin <- gsub(pattern = \"\\n..*\",\n",
    "                                                      replacement = \"\", \n",
    "                                                      x = ch_var_in_v05_qcd.pheno$Age_bin)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e7b7bb9a",
   "metadata": {},
   "outputs": [],
   "source": [
    "table(ch_var_in_baseline_qcd.pheno.followup$Age_Bin)\n",
    "table(ch_var_in_v05_qcd.pheno$Age_Bin)\n",
    "table(ch_var_in_baseline_qcd.pheno.followup$Age_bin)\n",
    "table(ch_var_in_v05_qcd.pheno$Age_bin)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e118a733",
   "metadata": {},
   "outputs": [],
   "source": [
    "ch_var_in_baseline_qcd.pheno.followup$Age_bin <- factor(ch_var_in_baseline_qcd.pheno.followup$Age_bin,\n",
    "                                          levels = c(\"<50\\n(n=45)\", \n",
    "                                                     \"50-55\\n(n=166)\", \n",
    "                                                     \"55-60\\n(n=162)\",\n",
    "                                                      \"60-65\\n(n=122)\", \n",
    "                                                     \"65-70\\n(n=67)\", \n",
    "                                                     \">=70\\n(n=14)\"), \n",
    "                                          ordered = T)\n",
    "\n",
    "ch_var_in_v05_qcd.pheno$Age_bin <- factor(ch_var_in_v05_qcd.pheno$Age_bin,\n",
    "                                          levels = c(\"<70\\n(n=91)\", \n",
    "                                                     \"70-75\\n(n=376)\", \n",
    "                                                     \"75-80\\n(n=368)\",\n",
    "                                                      \"80-85\\n(n=316)\", \n",
    "                                                     \"85-90\\n(n=151)\"), \n",
    "                                          ordered = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e6d2aa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## VAF distribution by Age bin\n",
    "\n",
    "## d\n",
    "d <- ggplot(data=ch_var_in_baseline_qcd.pheno.followup, aes(x=Age_bin, y=VAF, fill=Age_bin)) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Variant Allele Fraction\") + \n",
    "  geom_boxplot() + scale_y_log10(breaks = c(0.02,0.1,.2,.3, 0.5,1)) +\n",
    "  theme(# axis.text.x = element_text(angle = 40, vjust = 1,  hjust=1),\n",
    "        legend.title = element_blank(), legend.position = \"\",\n",
    "        plot.title = element_text(size = 20, face = \"bold\"),\n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) + \n",
    "scale_color_brewer(type = \"seq\", \n",
    "                     palette = \"Blues\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "  ggtitle(\"d\") \n",
    "\n",
    "\n",
    "## h\n",
    "h <- ggplot(data=ch_var_in_v05_qcd.pheno, aes(x=Age_bin, y=VAF, fill=Age_bin)) + \n",
    "  xlab(\"Age (Follow-up)\") + ylab(\"Variant Allele Fraction\") + \n",
    "  geom_boxplot() + scale_y_log10(breaks = c(0.02,0.1,.2,.3, 0.5,1)) +\n",
    "  theme( # axis.text.x = element_text(angle = 40, vjust = 1,  hjust=1),\n",
    "        legend.title = element_blank(), legend.position = \"\",\n",
    "        plot.title = element_text(size = 20, face = \"bold\"),\n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) + \n",
    "scale_color_brewer(type = \"seq\", \n",
    "                     palette = \"Blues\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "  ggtitle(\"h\") \n",
    "\n",
    "d\n",
    "h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01519c0a",
   "metadata": {},
   "outputs": [],
   "source": [
    "## multi-panel plot\n",
    "\n",
    "#png(\"~/Documents/Project/Baylor_ARIC_Exomes/Paper1_ARIC/Display/fig1.dist_incident_CH.noHemeCa_28May23.png\",\n",
    " #  width=24, height=12, units= \"in\", res=300, pointsize = 4)\n",
    "\n",
    "ggarrange(a, b, c, d, \n",
    "          e, f, g, h,\n",
    "          ncol = 4, \n",
    "          nrow = 2 )\n",
    "\n",
    "# dev.off()\n",
    "####\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a31c2cc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "##########################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8d5edc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig a ####\n",
    "a <- ggplot(data=aric_baseline_full_noPrevHeme, aes(x=age_base, y=CH_baseline, group=Follow_up) ) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "  # geom_smooth(aes(colour=Follow_up), \n",
    "    #          method =\"glm\", method.args= list(family = \"binomial\")) +\n",
    "  geom_smooth(aes(colour=Follow_up), \n",
    "              method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  +\n",
    "scale_y_continuous(breaks = c(seq(from = 0, to = 1, by = 0.1)) ) \n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b99e5a70",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig a ####\n",
    "ggplot(data=aric_baseline_full_noPrevHeme, aes(x=age_base, y=CH_baseline, group=Follow_up) ) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "   geom_smooth(aes(colour=Follow_up), \n",
    "              method =\"glm\", method.args= list(family = \"binomial\")) +\n",
    "  #geom_smooth(aes(colour=Follow_up), \n",
    "   #           method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  +\n",
    "scale_y_continuous(breaks = c(seq(from = 0, to = 0.5, by = 0.05)) ) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b954f2ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "a = ggplot(data=aric_baseline_n_v05_noPrevHeme, aes(x=age_base), group=1) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "    geom_smooth(aes(y=CH_baseline, colour=\"VAF>=2%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "    geom_smooth(aes(y=CHvaf10_baseline, colour=\"VAF>=10%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60ac5471",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d34227ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(aric_baseline_n_v05_noPrevHeme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa165ddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(maftools)\n",
    "maf.aric_v05 <- annovarToMaf(annovar = \"maftools.ch_var_in_v05_qcd_pheno.23Mar2023.txt\", \n",
    "                             refBuild = 'hg38', ens2hugo = TRUE, \n",
    "                             MAFobj = TRUE, tsbCol=\"GWAS_ID\")\n",
    "\n",
    "\n",
    "plotmafSummary(maf = maf.aric_v05, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)\n",
    "\n",
    "oncoplot(maf = maf.aric_v05, top = 15)\n",
    "\n",
    "plotVaf(maf = maf.aric_v05, vafCol = 't_vaf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c093036d",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Somatic Interactions\n",
    "# Mutually exclusive or co-occurring set of genes can be detected using somaticInteractions function, which performs pair-wise Fisherâ€™s Exact test to detect such significant pair of genes.\n",
    "# exclusive/co-occurance event analysis on top 10 mutated genes. \n",
    "somaticInteractions(maf = maf.aric_v05, \n",
    "                    top = 10, \n",
    "                    pvalue = c(0.01, 0.05))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
