{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d8e568eb",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(data.table)\n",
    "library(stringr)\n",
    "library(dplyr)\n",
    "library(tidyr)\n",
    "library(ggplot2)\n",
    "library(ggpubr)\n",
    "library(cowplot)\n",
    "library(ggridges)\n",
    "theme_set(theme_cowplot())\n",
    "library(viridis)\n",
    "# to create breaks\n",
    "# install.packages('ggbreak')\n",
    "# library(ggbreak) \n",
    "library(patchwork)\n",
    "## \n",
    "\n",
    "## \n",
    "setwd(\"/medpop/esp2/mesbah/projects/ch_progression/aric/pheno/\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1c998b5d",
   "metadata": {},
   "outputs": [],
   "source": [
    "# qcd CH variants with phenotypes\n",
    "ch_var_in_v05_qcd.pheno <- fread(\"ch_var_in_v05_qcd_pheno.23Mar2023.csv\", header = T)\n",
    "\n",
    "# aric_baseline_n_v05 <- fread(\"aric_baseline_n_v05_N4189.pheno_ch_status.9May2023.csv\", header=T)\n",
    "aric_baseline_n_v05_noPrevHeme <- fread(\"aric_baseline_n_v05_N4187.pheno_ch_status.noHemeCA.9May2023.csv\", header=T)\n",
    "# \n",
    "aric_baseline_n_v05_noPrevHeme$dAge <- aric_baseline_n_v05_noPrevHeme$Age - aric_baseline_n_v05_noPrevHeme$age_base\n",
    "summary(aric_baseline_n_v05_noPrevHeme$dAge)\n",
    "\n",
    "## Baseline pop\n",
    "aric_baseline_full_noPrevHeme <-  fread(\"aric_baseline_N10881.pheno_ch_status_bp_heme.noHemeCA.10May2023.csv\", \n",
    "                                        header=T)\n",
    "##     # qcd CH variants with phenotypes\n",
    "ch_var_in_baseline_qcd.pheno <- fread(\"ch_var_in_baseline_qcd_pheno.23Mar2023.csv\", header = T)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "46af63f7",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "028954fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "aric_baseline_n_v05_noPrevHeme$CHvaf10_v05 <- ifelse(aric_baseline_n_v05_noPrevHeme$CH_v05==1 & \n",
    "                                                      aric_baseline_n_v05_noPrevHeme$GWAS_ID %in% \n",
    "                                                      ch_var_in_v05_qcd.pheno$GWAS_ID[ch_var_in_v05_qcd.pheno$VAF>=0.10], \n",
    "                                                      1, 0)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_v05)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CHvaf10_v05)\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_v05,aric_baseline_n_v05_noPrevHeme$CHvaf10_v05)\n",
    "\n",
    "nrow(aric_baseline_full_noPrevHeme)\n",
    "table(ch_var_in_baseline_qcd.pheno$GWAS_ID %in% aric_baseline_full_noPrevHeme$gwasid, exclude = NULL)\n",
    "\n",
    "table(ch_var_in_baseline_qcd.pheno$GWAS_ID %in% aric_baseline_n_v05_noPrevHeme$GWAS_ID, exclude = NULL)\n",
    "\n",
    "table(aric_baseline_full_noPrevHeme$CH_baseline, exclude = NULL)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_baseline, exclude= NULL)\n",
    "\n",
    "round(prop.table(table(aric_baseline_full_noPrevHeme$CH_baseline, exclude = NULL))*100,1)\n",
    "round(prop.table(table(aric_baseline_n_v05_noPrevHeme$CH_baseline, exclude= NULL))*100,1)\n",
    "\n",
    "aric_baseline_full_noPrevHeme$Follow_up <- ifelse(aric_baseline_full_noPrevHeme$gwasid %in% aric_baseline_n_v05_noPrevHeme$GWAS_ID, \"w/ follow-up\", \"w/o follow-up\")\n",
    "table(aric_baseline_full_noPrevHeme$Follow_up)\n",
    "summary(aric_baseline_full_noPrevHeme$age_base)\n",
    "nrow(aric_baseline_n_v05_noPrevHeme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2346ccea",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "898b8ca2",
   "metadata": {},
   "outputs": [],
   "source": [
    "nrow(ch_var_in_baseline_qcd.pheno)\n",
    "nrow(ch_var_in_v05_qcd.pheno)\n",
    "\n",
    "str(ch_var_in_baseline_qcd.pheno[,c(23,9,16:21)])\n",
    "cat(\"\\nvisit 5\\n\")\n",
    "cat(\"\\n\\n\")\n",
    "str(ch_var_in_v05_qcd.pheno[,c(22,9,16:21)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "5188b806",
   "metadata": {},
   "outputs": [],
   "source": [
    "rm(base_var_table); \n",
    "\n",
    "base_var_table <- ch_var_in_baseline_qcd.pheno[,c(23,9,16:21)] %>% \n",
    "                                               group_by(varID) %>% \n",
    "                                               summarise(Number_of_Occurance = n(), \n",
    "                                                         across(everything(), \n",
    "                                                                unique))\n",
    "                                               \n",
    "                                               \n",
    "str(base_var_table)\n",
    "\n",
    "fwrite(base_var_table, \"Tables.base_var_table.csv\", sep=\",\", row.names = F, col.names = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0a84d489",
   "metadata": {},
   "outputs": [],
   "source": [
    "v5_var_table <- ch_var_in_v05_qcd.pheno[,c(22,5:9,16:21)] %>% \n",
    "                                               group_by(varID) %>% \n",
    "                                               summarise(Number_of_Occurance = n(), \n",
    "                                                         across(everything(), \n",
    "                                                                unique))\n",
    "                                               \n",
    "                                               \n",
    "str(v5_var_table)\n",
    "\n",
    "fwrite(v5_var_table, \"TableS.v5_var_table.csv\", sep=\",\", row.names = F, col.names = T)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ef1c04b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "length(unique(v5_var_table$varID))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0942eeaa",
   "metadata": {},
   "outputs": [],
   "source": [
    "sum(v5_var_table$Number_of_Occurance)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bf212460",
   "metadata": {},
   "outputs": [],
   "source": [
    "str(ch_var_in_v05_qcd.pheno[,c(1:30)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "556db588",
   "metadata": {},
   "outputs": [],
   "source": [
    "aric_baseline_n_v05_noPrevHeme$CHvaf10_v05 <- ifelse(aric_baseline_n_v05_noPrevHeme$CH_v05==1 & \n",
    "                                                      aric_baseline_n_v05_noPrevHeme$GWAS_ID %in% \n",
    "                                                      ch_var_in_v05_qcd.pheno$GWAS_ID[ch_var_in_v05_qcd.pheno$VAF>=0.10], \n",
    "                                                      1, 0)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_v05)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CHvaf10_v05)\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_v05,aric_baseline_n_v05_noPrevHeme$CHvaf10_v05)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7423f420",
   "metadata": {},
   "outputs": [],
   "source": [
    "nrow(aric_baseline_full_noPrevHeme)\n",
    "table(ch_var_in_baseline_qcd.pheno$GWAS_ID %in% aric_baseline_full_noPrevHeme$gwasid, exclude = NULL)\n",
    "\n",
    "table(ch_var_in_baseline_qcd.pheno$GWAS_ID %in% aric_baseline_n_v05_noPrevHeme$GWAS_ID, exclude = NULL)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c3221a41",
   "metadata": {},
   "outputs": [],
   "source": [
    "table(aric_baseline_full_noPrevHeme$CH_baseline, exclude = NULL)\n",
    "table(aric_baseline_n_v05_noPrevHeme$CH_baseline, exclude= NULL)\n",
    "\n",
    "round(prop.table(table(aric_baseline_full_noPrevHeme$CH_baseline, exclude = NULL))*100,1)\n",
    "round(prop.table(table(aric_baseline_n_v05_noPrevHeme$CH_baseline, exclude= NULL))*100,1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fea6e21e",
   "metadata": {},
   "outputs": [],
   "source": [
    "aric_baseline_full_noPrevHeme$Follow_up <- ifelse(aric_baseline_full_noPrevHeme$gwasid %in% aric_baseline_n_v05_noPrevHeme$GWAS_ID, \"w/ follow-up\", \"w/o follow-up\")\n",
    "table(aric_baseline_full_noPrevHeme$Follow_up)\n",
    "summary(aric_baseline_full_noPrevHeme$age_base)\n",
    "nrow(aric_baseline_n_v05_noPrevHeme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1095130",
   "metadata": {},
   "outputs": [],
   "source": [
    "###### group by\n",
    "# library(dplyr)\n",
    "ch_var_base_count <- ch_var_in_baseline_qcd.pheno[,c(1,9,11)] %>% \n",
    "  group_by(GWAS_ID) %>% \n",
    "  summarise(nCHIP=n(), \n",
    "            Gene=paste(Gene,collapse = \";\"), \n",
    "            VAF=paste(round(VAF,3),collapse = \";\")\n",
    "  )\n",
    "\n",
    "ch_var_base_count %>% filter(nCHIP==2) %>% arrange(., nCHIP)\n",
    "\n",
    "ch_var_base_count %>% filter(nCHIP>2) %>% arrange(., nCHIP)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "161726fb",
   "metadata": {},
   "outputs": [],
   "source": [
    "ch_var_v05_count <- ch_var_in_v05_qcd.pheno[,c(1,9,11)] %>% \n",
    "  group_by(GWAS_ID) %>% \n",
    "  summarise(nCHIP=n(), \n",
    "            Gene=paste(Gene,collapse = \";\"), \n",
    "            VAF=paste(round(VAF,3),collapse = \";\")\n",
    "  )\n",
    "\n",
    "\n",
    "ch_var_v05_count %>% filter(nCHIP==2) %>% arrange(., nCHIP)\n",
    "\n",
    "ch_var_v05_count %>% filter(nCHIP>2) %>% arrange(., nCHIP)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "235761b5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "63d198f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "####### Fig 1a-h: baseline vs. follow-up visit CH\n",
    "#### Fig a ####\n",
    "## Base: a,b,c,d\n",
    "\n",
    "# Prev. at baseline\n",
    "a <- ggplot(data=aric_baseline_n_v05_noPrevHeme, \n",
    "            aes(x=age_base), group=1) + \n",
    "  xlab(\"Age at baseline (years)\") + ylab(\"Prevalence\") +\n",
    "    geom_smooth(aes(y=CH_baseline, colour=\"VAF>=2%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\"), se=T, level = 0.95) +\n",
    "    geom_smooth(aes(y=CHvaf10_baseline, colour=\"VAF>=10%\"), \n",
    "             method =\"glm\", method.args = list(family = \"binomial\"), se=T, level = 0.95) +\n",
    "    # geom_smooth(aes(y=CH_baseline, colour=\"VAF>=2%\"), method =\"gam\") +\n",
    "    # geom_smooth(aes(y=CHvaf10_baseline, colour=\"VAF>=10%\"), method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) +\n",
    "scale_y_continuous(breaks = c(0,0.05,0.1, 0.15,\n",
    "                                seq(0.1,0.4,0.1), \n",
    "                                seq(0.4,1,.2)) , \n",
    "                     limits = c(0,1) ) +\n",
    "scale_color_brewer(type = \"qual\", \n",
    "                     palette = \"Dark2\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"colour\" ) \n",
    "# https://colorbrewer2.org/#type=qualitative&scheme=Dark2&n=3\n",
    "a\n",
    "# a <- a + scale_y_break(breaks=c(0.60, 0.9), \n",
    "#                 scales=c(0.1, 10), \n",
    "#                 #ticklabels = c(0,60,70,80,90,100),\n",
    "#                 expand = F,space = 0.5) + \n",
    "#   theme(plot.title = element_text(hjust = 0.0, vjust = 2)) \n",
    "# \n",
    "# a + scale_y_continuous(breaks = c(0,0.05,0.1, 0.15, 0.2, \n",
    "#                                   0.3, 0.4, 0.5,0.6))\n",
    "# + \n",
    "#   theme(axis.title.y.right = element_blank(),\n",
    "#         axis.text.y.right = element_blank(), \n",
    "#         axis.line.y.right = element_blank(), \n",
    "#         axis.ticks.y.right = element_blank())\n",
    "# a\n",
    "## Prev. at Follow-up\n",
    "e <- ggplot(data=aric_baseline_n_v05_noPrevHeme, \n",
    "            aes(x=Age), group=1) + \n",
    "  xlab(\"Age at follow-up (years)\") + ylab(\"Prevalence\") +\n",
    "    geom_smooth(aes(y=CH_v05, colour=\"VAF>=2%\"),na.rm = T, \n",
    "              method =\"glm\", method.args = list(family = \"binomial\"), se=T, level = 0.95) +\n",
    "    geom_smooth(aes(y=CHvaf10_v05, colour=\"VAF>=10%\"), na.rm = T,\n",
    "              method =\"glm\", method.args = list(family = \"binomial\"), se=T, level = 0.95) +\n",
    "    # geom_smooth(aes(y=CH_v05, colour=\"VAF>=2%\"), method =\"gam\") +\n",
    "    # geom_smooth(aes(y=CHvaf10_v05, colour=\"VAF>=10%\"), method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"e\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  +\n",
    "  scale_y_continuous(breaks = c(0,0.05,0.1, 0.15,\n",
    "                                seq(0.1,0.4,0.1), \n",
    "                                seq(0.4,1,.2)) , \n",
    "                     limits = c(0,1)) +\n",
    "  scale_color_brewer(type = \"qual\", \n",
    "                     palette = \"Dark2\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"colour\" ) # https://colorbrewer2.org/#type=qualitative&scheme=Dark2&n=3\n",
    "e\n",
    "# e <- e + scale_y_break(breaks = c(0.02,0.05))\n",
    "# \n",
    "# e + scale_y_continuous(breaks = c(-.05,0, 0.05, 0.1, 0.15, \n",
    "#                                   0.2, 0.3, 0.4, 0.5), \n",
    "#                        ) # + \n",
    "#   theme(axis.title.y.right = element_blank(),\n",
    "#         axis.text.y.right = element_blank(), \n",
    "#         axis.line.y.right = element_blank(), \n",
    "#         axis.ticks.y.right = element_blank())\n",
    "\n",
    "a\n",
    "e\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "10ab0bf5",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d016df86",
   "metadata": {},
   "outputs": [],
   "source": [
    "###### Fig 1b,f\n",
    "\n",
    "aric_baseline_n_v05_noPrevHeme$Age_bin_base <- ifelse(aric_baseline_n_v05_noPrevHeme$age_base<50,\"<50\\n(n=587)\",\n",
    "                                               ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=50 &  \n",
    "                                                        aric_baseline_n_v05_noPrevHeme$age_base<55,\n",
    "                                                      \"50-55\\n(n=1478)\", \n",
    "                                                      ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=55 &\n",
    "                                                               aric_baseline_n_v05_noPrevHeme$age_base<60,\n",
    "                                                             \"55-60\\n(n=1127)\", \n",
    "                                                             ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=60 & \n",
    "                                                                      aric_baseline_n_v05_noPrevHeme$age_base<65,\n",
    "                                                                    \"60-65\\n(n=686)\",\n",
    "                                                                    ifelse(aric_baseline_n_v05_noPrevHeme$age_base>=65 & \n",
    "                                                                             aric_baseline_n_v05_noPrevHeme$age_base<70,\n",
    "                                                                           \"65-70\\n(n=270)\",\">=70\\n(n=39)\")))))\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$Age_bin_base)\n",
    "\n",
    "\n",
    "aric_baseline_n_v05_noPrevHeme$Age_bin_v05 <- ifelse(aric_baseline_n_v05_noPrevHeme$Age<70,\"<70\\n(n=416)\",\n",
    "                                          ifelse(aric_baseline_n_v05_noPrevHeme$Age>=70 &  \n",
    "                                                   aric_baseline_n_v05_noPrevHeme$Age<75,\n",
    "                                                 \"70-75\\n(n=1542)\", \n",
    "                                                 ifelse(aric_baseline_n_v05_noPrevHeme$Age>=75 &\n",
    "                                                          aric_baseline_n_v05_noPrevHeme$Age<80,\n",
    "                                                        \"75-80\\n(n=1178)\", \n",
    "                                                        ifelse(aric_baseline_n_v05_noPrevHeme$Age>=80 & \n",
    "                                                                 aric_baseline_n_v05_noPrevHeme$Age<85,\n",
    "                                                               \"80-85\\n(n=753)\",\"85-90\\n(n=298)\"))))\n",
    "\n",
    "\n",
    "table(aric_baseline_n_v05_noPrevHeme$Age_bin_v05)\n",
    "\n",
    "## Baseline\n",
    "# ## Count\n",
    "# ch_base.ncount <- as.data.frame(table(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID), \n",
    "#                                stringsAsFactors = F)\n",
    "# names(ch_base.ncount) <- c(\"GWAS_ID\", \"nCH_base\")\n",
    "# \n",
    "# ch_v05.ncount <- as.data.frame(table(ch_var_in_v05_qcd.pheno$GWAS_ID), stringsAsFactors = F)\n",
    "# names(ch_v05.ncount) <- c(\"GWAS_ID\", \"nCH_v05\")\n",
    "# \n",
    "# nCH_base_v05 <- merge(aric_baseline_n_v05_noPrevHeme[,c(2,115,116)], \n",
    "#                       ch_base.ncount, by=\"GWAS_ID\", all.x=T)\n",
    "# \n",
    "# nCH_base_v05 <- merge(nCH_base_v05, ch_v05.ncount, \n",
    "#                       by=\"GWAS_ID\", all.x=T)\n",
    "# \n",
    "# head(nCH_base_v05)\n",
    "# table(nCH_base_v05$nCH_base, exclude = NULL)\n",
    "# table(nCH_base_v05$nCH_v05, exclude = NULL)\n",
    "# \n",
    "# nCH_base_v05$nCH_base[is.na(nCH_base_v05$nCH_base)] <- 0\n",
    "# nCH_base_v05$nCH_v05[is.na(nCH_base_v05$nCH_v05)] <- 0\n",
    "# table(nCH_base_v05$nCH_base, exclude = NULL)\n",
    "# table(nCH_base_v05$nCH_v05, exclude = NULL)\n",
    "# \n",
    "# \n",
    "# nCH_base_v05$nCH_base[nCH_base_v05$nCH_base>=4] <- \"4+\"\n",
    "# table(nCH_base_v05$nCH_base)\n",
    "# \n",
    "# nCH_base_v05$nCH_v05[nCH_base_v05$nCH_v05>=4] <- \"4+\"\n",
    "# table(nCH_base_v05$nCH_v05)\n",
    "# \n",
    "# names(nCH_base_v05)\n",
    "# \n",
    "# # ch_base.ncount <- as.data.frame(prop.table(table(nCH_base_v05[,c(2,4)]),margin = 1)*100)\n",
    "# \n",
    "# \n",
    "# ch_base.ncount\n",
    "# ch_v05.ncount <- as.data.frame(prop.table(table(nCH_base_v05[,c(3,5)]),margin = 1)*100)\n",
    "# ch_v05.ncount\n",
    "# \n",
    "# table(nCH_base_v05$Age_bin_base)\n",
    "# \n",
    "#     # Base\n",
    "# ch_base.ncount$Age_bin_base <- factor(ch_base.ncount$Age_bin_base,\n",
    "#                                           levels = c(\"<50\\n(n=587)\", \"50-55\\n(n=1478)\", \"55-60\\n(n=1127)\", \n",
    "#                                                      \"60-65\\n(n=686)\", \"65-70\\n(n=270)\", \">=70\\n(n=39)\"),\n",
    "#                                           ordered = T)\n",
    "# # ch_base.ncount$nCH_base <- reorder(ch_base.ncount$nCH_base, ch_base.ncount$Freq)\n",
    "# \n",
    "# ch_base.ncount$nCH_base <- factor(ch_base.ncount$nCH_base, levels = c(\"0\", \"1\", \"2\", \"3\", \"4+\"))\n",
    "# \n",
    "# ch_base.ncount$nCH_base <- factor(ch_base.ncount$nCH_base, levels = rev(levels(ch_base.ncount$nCH_base)))\n",
    "# \n",
    "# \n",
    "# table(nCH_base_v05$Age_bin_v05)\n",
    "# \n",
    "# \n",
    "#     # V05\n",
    "# ch_v05.ncount$Age_bin_v05 <- factor(ch_v05.ncount$Age_bin_v05,\n",
    "#                                           levels = c(\"<70\\n(n=416)\", \n",
    "#                                                      \"70-75\\n(n=1542)\", \n",
    "#                                                      \"75-80\\n(n=1178)\",\n",
    "#                                                       \"80-85\\n(n=753)\", \n",
    "#                                                      \"85-90\\n(n=298)\"), \n",
    "#                                           ordered = T)\n",
    "# # ch_base.ncount$nCH_base <- reorder(ch_base.ncount$nCH_base, ch_base.ncount$Freq)\n",
    "# \n",
    "# ch_v05.ncount$nCH_v05 <- factor(ch_v05.ncount$nCH_v05, \n",
    "#                                 levels = c(\"0\", \"1\", \"2\", \"3\", \"4+\"))\n",
    "# \n",
    "# ch_v05.ncount$nCH_v05 <- factor(ch_v05.ncount$nCH_v05, \n",
    "#                                 levels = rev(levels(ch_v05.ncount$nCH_v05)))\n",
    "# \n",
    "# \n",
    "# ## b\n",
    "# b <- ch_base.ncount %>% filter(Freq>0) %>% ggplot(., \n",
    "#                                               aes(fill=nCH_base, \n",
    "#                                                   y=Freq, \n",
    "#                                                   x=Age_bin_base)) + \n",
    "#   geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "#   ylab(\"Proportion (%)\") + xlab(\"Age at baseline (years)\") + \n",
    "# #  scale_fill_viridis(discrete = T) +\n",
    "# scale_color_brewer(type = \"seq\", \n",
    "#                      palette = \"GnBu\",\n",
    "#                      direction = 1, \n",
    "#                      aesthetics = \"fill\" ) +\n",
    "#   theme(legend.title = element_blank(), \n",
    "#         legend.position = \"\") +\n",
    "#   ggtitle(\"b\") + \n",
    "#   theme(legend.position = \"right\", \n",
    "#         plot.title = element_text(size = 20, face = \"bold\"), \n",
    "#         axis.text=element_text(size=12, face=\"bold\"),\n",
    "#         axis.title=element_text(size=14,face=\"bold\"))  \n",
    "# b <- b + scale_y_break(breaks=c(50, 60), \n",
    "#                   scales=c(10, 0.00001), \n",
    "#                   #ticklabels = c(0,60,70,80,90,100),\n",
    "#                   expand = F,space = 0.5) + \n",
    "#   theme(plot.title = element_text(hjust = 0.0, vjust = 2)) \n",
    "# \n",
    "# b <- b + scale_y_continuous(breaks = c(0,50,60,70,80,90,100)) + \n",
    "#   theme(axis.title.y.right = element_blank(),\n",
    "#         axis.text.y.right = element_blank(), \n",
    "#         axis.line.y.right = element_blank(), \n",
    "#         axis.ticks.y.right = element_blank())\n",
    "# b\n",
    "# ## f\n",
    "# f <- ch_v05.ncount %>% filter(Freq>0) %>% ggplot(., \n",
    "#                                      aes(fill=nCH_v05, \n",
    "#                                          y=Freq, \n",
    "#                                          x=Age_bin_v05)) + \n",
    "#   geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "#   ylab(\"Proportion (%)\") + xlab(\"Age at follow-up (years)\") + \n",
    "# #  scale_fill_viridis(discrete = T) +\n",
    "# # scale_color_brewer(type = \"div\", \n",
    "# #                      palette = \"RdYlBu\",\n",
    "# #                      direction = 1, \n",
    "# #                      aesthetics = \"fill\" ) +\n",
    "# scale_color_brewer(type = \"seq\", \n",
    "#                      palette = \"GnBu\",\n",
    "#                      direction = 1, \n",
    "#                      aesthetics = \"fill\" ) +\n",
    "#   theme(legend.title = element_blank(), \n",
    "#         legend.position = \"\") +\n",
    "#   ggtitle(\"f\") + \n",
    "#   theme(legend.position = \"right\", \n",
    "#         plot.title = element_text(size = 20, face = \"bold\"), \n",
    "#         axis.text=element_text(size=12, face=\"bold\"),\n",
    "#         axis.title=element_text(size=14,face=\"bold\"),\n",
    "#         # axis.title.y.left =  element_blank(),\n",
    "#         # axis.text.y.left = element_blank(), \n",
    "#         # axis.line.y.left = element_blank(), \n",
    "#         # axis.ticks.y.left = element_blank(),\n",
    "#         # axis.line.x.bottom =  element_blank(), \n",
    "#         # axis.ticks.x.bottom = element_blank()\n",
    "#         ) \n",
    "# \n",
    "# f <- f + scale_y_break(breaks=c(50, 60), \n",
    "#                 scales=c(10, 0.00001), \n",
    "#                 #ticklabels = c(0,60,70,80,90,100),\n",
    "#                 expand = F,space = 0.5) + \n",
    "#   theme(plot.title = element_text(hjust = 0.0, vjust = 2)) \n",
    "# \n",
    "# f <- f + scale_y_continuous(breaks = c(0,50,60,70,80,90,100)) + \n",
    "#   theme(axis.title.y.right = element_blank(),\n",
    "#         axis.text.y.right = element_blank(), \n",
    "#         axis.line.y.right = element_blank(), \n",
    "#         axis.ticks.y.right = element_blank())\n",
    "# \n",
    "# b\n",
    "# f\n",
    "# \n",
    "\n",
    "### # of CH per sample\n",
    "length(unique(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID))\n",
    "\n",
    "length(unique(ch_var_in_v05_qcd.pheno$GWAS_ID))\n",
    "\n",
    "\n",
    "## Baseline\n",
    "## Count\n",
    "ch_base.ncount <- as.data.frame(table(ch_var_in_baseline_qcd.pheno.followup$GWAS_ID),\n",
    "                               stringsAsFactors = F)\n",
    "\n",
    "names(ch_base.ncount) <- c(\"GWAS_ID\", \"nCH_base\")\n",
    "ch_base.ncount <- merge(ch_base.ncount,\n",
    "                        aric_baseline_n_v05_noPrevHeme[,c(2,58)],\n",
    "                        by.y = \"GWAS_ID\")\n",
    "names(ch_base.ncount)\n",
    "\n",
    "ch_base.ncount$nCH_base[ch_base.ncount$nCH_base>=4] <- \"4+\"\n",
    "table(ch_base.ncount$nCH_base)\n",
    "\n",
    "\n",
    "ch_base.ncount$Age_bin <- ifelse(ch_base.ncount$age_base<50,\"<50\\n(n=42)\",\n",
    "                                 ifelse(ch_base.ncount$age_base>=50 &\n",
    "                                          ch_base.ncount$age_base<55,\n",
    "                                        \"50-55\\n(n=131)\",\n",
    "                                        ifelse(ch_base.ncount$age_base>=55 &\n",
    "                                                 ch_base.ncount$age_base<60,\n",
    "                                               \"55-60\\n(n=127)\",\n",
    "                                               ifelse(ch_base.ncount$age_base>=60 &\n",
    "                                                        ch_base.ncount$age_base<65,\n",
    "                                                      \"60-65\\n(n=98)\",\n",
    "                                                      ifelse(ch_base.ncount$age_base>=65 &\n",
    "                                                               ch_base.ncount$age_base<70,\n",
    "                                                             \"65-70\\n(n=48)\",\">=70\\n(n=11)\")))))\n",
    "table(ch_base.ncount$Age_bin)\n",
    "\n",
    "\n",
    "ch_base.ncount <- as.data.frame(prop.table(table(ch_base.ncount[,c(2,4)]), \n",
    "                                           margin = 2)*100)\n",
    "\n",
    "# order\n",
    "ch_base.ncount$Age_bin <- factor(ch_base.ncount$Age_bin,\n",
    "                                 levels = c(\"<50\\n(n=42)\", \n",
    "                                            \"50-55\\n(n=131)\", \n",
    "                                            \"55-60\\n(n=127)\",\n",
    "                                            \"60-65\\n(n=98)\", \n",
    "                                            \"65-70\\n(n=48)\", \n",
    "                                            \">=70\\n(n=11)\"),\n",
    "                                 ordered = T)\n",
    "\n",
    "ch_base.ncount$nCH_base <- factor(ch_base.ncount$nCH_base, \n",
    "                                  levels = c(\"1\", \"2\", \"3\", \"4+\"))\n",
    "\n",
    "ch_base.ncount$nCH_base <- factor(ch_base.ncount$nCH_base, \n",
    "                                  levels = rev(levels(ch_base.ncount$nCH_base)))\n",
    "\n",
    "  # V05\n",
    "ch_v05.ncount <- as.data.frame(table(ch_var_in_v05_qcd.pheno$GWAS_ID), \n",
    "                               stringsAsFactors = F)\n",
    "names(ch_v05.ncount) <- c(\"GWAS_ID\", \"nCH_v05\")\n",
    "\n",
    "\n",
    "nCH_v05 <- merge(aric_baseline_n_v05_noPrevHeme[,c(2,7)],\n",
    "                      ch_v05.ncount, by=\"GWAS_ID\")\n",
    "\n",
    "\n",
    "\n",
    "nCH_v05$nCH_v05[nCH_v05$nCH_v05>=4] <- \"4+\"\n",
    "table(nCH_v05$nCH_v05)\n",
    "\n",
    "nCH_v05$Age_bin_v05 <- ifelse(nCH_v05$Age<70, \"<70\\n(n=83)\",\n",
    "                              ifelse(nCH_v05$Age>=70 & \n",
    "                                       nCH_v05$Age<75,\n",
    "                                     \"70-75\\n(n=305)\", \n",
    "                                     ifelse(nCH_v05$Age>=75 & \n",
    "                                              nCH_v05$Age<80,\n",
    "                                            \"75-80\\n(n=304)\",\n",
    "                                            ifelse(nCH_v05$Age>=80 & \n",
    "                                                     nCH_v05$Age<85, \n",
    "                                                   \"80-85\\n(n=242)\",\n",
    "                                                   \"85-90\\n(n=113)\"))))\n",
    "table(nCH_v05$Age_bin_v05)\n",
    "\n",
    "nCH_v05 <- as.data.frame(prop.table(table(nCH_v05[,c(3,4)]),margin = 2)*100)\n",
    "\n",
    "nCH_v05$Age_bin_v05 <- factor(nCH_v05$Age_bin_v05,\n",
    "                                 levels = c(\"<70\\n(n=83)\", \n",
    "                                            \"70-75\\n(n=305)\", \n",
    "                                            \"75-80\\n(n=304)\",\n",
    "                                            \"80-85\\n(n=242)\", \n",
    "                                            \"85-90\\n(n=113)\"),\n",
    "                                 ordered = T)\n",
    "\n",
    "nCH_v05$nCH_v05 <- factor(nCH_v05$nCH_v05, \n",
    "                          levels = c(\"1\", \"2\", \"3\", \"4+\"))\n",
    "\n",
    "nCH_v05$nCH_v05 <- factor(nCH_v05$nCH_v05, \n",
    "                           levels = rev(levels(nCH_v05$nCH_v05)))\n",
    "\n",
    "\n",
    "## c\n",
    "# c <- ch_base.ncount %>% filter(Freq.1>0) %>% ggplot(., \n",
    "#                                              aes(fill=Freq, \n",
    "#                                                  y=Freq.1, \n",
    "#                                                  x=Age_bin)) + \n",
    "#  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "#  ylab(\"Proportion\") + xlab(\"Age (baseline)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "#  theme(legend.title = element_blank(), \n",
    "#        legend.position = \"\") +\n",
    "#  ggtitle(\"c\") + \n",
    "#  theme(legend.position = \"right\", \n",
    "#        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "#        axis.text=element_text(size=12, face=\"bold\"),\n",
    "#        axis.title=element_text(size=14,face=\"bold\")) \n",
    "\n",
    "\n",
    "## Folow-up\n",
    "# ch_v05.ncount <- as.data.frame(table(ch_var_in_v05_qcd.pheno$GWAS_ID), stringsAsFactors = F)\n",
    "\n",
    "# ch_v05.ncount <- merge(ch_v05.ncount, \n",
    "#                        ch_var_in_v05_qcd.pheno[!duplicated(ch_var_in_v05_qcd.pheno$GWAS_ID),c(1,30)], \n",
    " #                      by.x = \"Var1\", \n",
    " #                      by.y = \"GWAS_ID\")\n",
    "\n",
    "# ch_v05.ncount$Age_bin <- ifelse(ch_v05.ncount$Age<70,\"<70\\n(n=83)\",\n",
    "#                                          ifelse(ch_v05.ncount$Age>=70 &  \n",
    "#                                                   ch_v05.ncount$Age<75,\n",
    "#                                                 \"70-75\\n(n=305)\", \n",
    "#                                                 ifelse(ch_v05.ncount$Age>=75 &\n",
    "#                                                          ch_v05.ncount$Age<80,\n",
    "#                                                        \"75-80\\n(n=304)\", \n",
    "#                                                        ifelse(ch_v05.ncount$Age>=80 & \n",
    "#                                                                 ch_v05.ncount$Age<85,\n",
    "#                                                               \"80-85\\n(n=242)\",\"85-90\\n(n=113)\"))))\n",
    "\n",
    "# table(ch_v05.ncount$Age_bin)\n",
    "   # order\n",
    "# ch_v05.ncount$Age_bin <- factor(ch_v05.ncount$Age_bin, \n",
    "#                                          levels = c(\"<70\\n(n=83)\", \n",
    "#                                                     \"70-75\\n(n=305)\", \n",
    "#                                                     \"75-80\\n(n=304)\",\n",
    "#                                                      \"80-85\\n(n=242)\", \n",
    "#                                                     \"85-90\\n(n=113)\"), \n",
    "#                                          ordered = T)\n",
    "\n",
    "# ch_v05.ncount <- as.data.frame(prop.table(table(ch_v05.ncount[,c(2,4)]),margin = 2)*100)\n",
    "\n",
    "## g\n",
    "# g <- ch_v05.ncount %>% filter(Freq.1>0) %>% ggplot(., \n",
    "#                                     aes(fill=Freq, \n",
    "#                                         y=Freq.1, \n",
    "#                                         x=Age_bin)) + \n",
    "#  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "#  ylab(\"Proportion\") + xlab(\"Age (follow-up)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "#  theme(legend.title = element_blank(), \n",
    "#        legend.position = \"\") +\n",
    "#  ggtitle(\"g\") + \n",
    "#  theme(legend.position = \"right\", \n",
    "#        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "#        axis.text=element_text(size=12, face=\"bold\"),\n",
    "#        axis.title=element_text(size=14,face=\"bold\")) \n",
    "#\n",
    "# c\n",
    "# g\n",
    "\n",
    "## b\n",
    "b <- ch_base.ncount %>% filter(Freq>0) %>% \n",
    "  ggplot(., \n",
    "         aes(fill=nCH_base, \n",
    "             y=Freq, \n",
    "             x=Age_bin)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion (%)\") + xlab(\"Age at baseline (years)\") + \n",
    "  #  scale_fill_viridis(discrete = T) +\n",
    "  scale_color_brewer(type = \"seq\", \n",
    "                     palette = \"GnBu\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"b\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  \n",
    "b\n",
    "\n",
    "# b <- b + scale_y_break(breaks=c(5, 70),\n",
    "  #                     scales=c(10, 0.00001),\n",
    "   #                    #ticklabels = c(0,60,70,80,90,100),\n",
    "    #                   expand = F,space = 0.5) +\n",
    " # theme(plot.title = element_text(hjust = 0.0, vjust = 2))\n",
    "\n",
    "# b <- b + scale_y_continuous(breaks = c(0,5,70,80,90,100)) +\n",
    " # theme(axis.title.y.right = element_blank(),\n",
    "  #      axis.text.y.right = element_blank(),\n",
    "   #     axis.line.y.right = element_blank(),\n",
    "    #    axis.ticks.y.right = element_blank())\n",
    "# b\n",
    "## f\n",
    "f <- nCH_v05 %>% filter(Freq>0) %>% \n",
    "  ggplot(., \n",
    "         aes(fill=nCH_v05, \n",
    "             y=Freq, x=Age_bin_v05)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion (%)\") + xlab(\"Age at follow-up (years)\") + \n",
    "  #  scale_fill_viridis(discrete = T) +\n",
    "  # scale_color_brewer(type = \"div\", \n",
    "  #                      palette = \"RdYlBu\",\n",
    "  #                      direction = 1, \n",
    "  #                      aesthetics = \"fill\" ) +\n",
    "  scale_color_brewer(type = \"seq\", \n",
    "                     palette = \"GnBu\",\n",
    "                     direction = 1, \n",
    "                     aesthetics = \"fill\" ) +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"f\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"),\n",
    "        # axis.title.y.left =  element_blank(),\n",
    "        # axis.text.y.left = element_blank(), \n",
    "        # axis.line.y.left = element_blank(), \n",
    "        # axis.ticks.y.left = element_blank(),\n",
    "        # axis.line.x.bottom =  element_blank(), \n",
    "        # axis.ticks.x.bottom = element_blank()\n",
    "  ) \n",
    "\n",
    "f\n",
    "\n",
    "# f <- f + scale_y_break(breaks=c(5, 70),\n",
    "  #                     scales=c(10, 0.00001),\n",
    "   #                    #ticklabels = c(0,60,70,80,90,100),\n",
    "    #                   expand = F,space = 0.5) +\n",
    " # theme(plot.title = element_text(hjust = 0.0, vjust = 2))\n",
    "\n",
    "# f <- f + scale_y_continuous(breaks = c(0,5,70,80,90,100)) +\n",
    " # theme(axis.title.y.right = element_blank(),\n",
    "  #      axis.text.y.right = element_blank(),\n",
    "   #     axis.line.y.right = element_blank(),\n",
    "    #    axis.ticks.y.right = element_blank())\n",
    "\n",
    "b\n",
    "f\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "39c51618",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "410a9142",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig 1c,g\n",
    "# Gene distributions by age bin\n",
    "\n",
    "### baseline\n",
    "nrow(ch_var_in_baseline_qcd.pheno)\n",
    "ch_var_in_baseline_qcd.pheno.followup <- subset(ch_var_in_baseline_qcd.pheno, \n",
    "                                                ch_var_in_baseline_qcd.pheno$GWAS_ID %in% \n",
    "                                                aric_baseline_n_v05_noPrevHeme$GWAS_ID)\n",
    "nrow(ch_var_in_baseline_qcd.pheno.followup)\n",
    "# ## each 10 years\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Age_bin_10yr <- ifelse(ch_var_in_baseline_qcd.pheno$age_base<50,\"<50\\n(n=45)\",\n",
    "#                                                         ifelse(ch_var_in_baseline_qcd.pheno$age_base>=50 &  \n",
    "#                                                                  ch_var_in_baseline_qcd.pheno$age_base<60,\n",
    "#                                                                \"50-60\\n(n=328)\", \n",
    "#                                                                ifelse(ch_var_in_baseline_qcd.pheno$age_base>=60 &\n",
    "#                                                                         ch_var_in_baseline_qcd.pheno$age_base<70,\n",
    "#                                                                       \"60-70\\n(n=189)\", \">=70\\n(n=14)\")))\n",
    "# \n",
    "# \n",
    "# table(ch_var_in_baseline_qcd.pheno.followup$Age_bin_10yr)\n",
    "# \n",
    "## each 5-years\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_bin <- ifelse(ch_var_in_baseline_qcd.pheno$age_base<50,\"<50\\n(n=45)\",\n",
    "                                               ifelse(ch_var_in_baseline_qcd.pheno$age_base>=50 &  \n",
    "                                                        ch_var_in_baseline_qcd.pheno$age_base<55,\n",
    "                                                      \"50-55\\n(n=166)\", \n",
    "                                                      ifelse(ch_var_in_baseline_qcd.pheno$age_base>=55 &\n",
    "                                                               ch_var_in_baseline_qcd.pheno$age_base<60,\n",
    "                                                             \"55-60\\n(n=162)\", \n",
    "                                                             ifelse(ch_var_in_baseline_qcd.pheno$age_base>=60 & \n",
    "                                                                      ch_var_in_baseline_qcd.pheno$age_base<65,\n",
    "                                                                    \"60-65\\n(n=122)\",\n",
    "                                                                    ifelse(ch_var_in_baseline_qcd.pheno$age_base>=65 & \n",
    "                                                                             ch_var_in_baseline_qcd.pheno$age_base<70,\n",
    "                                                                           \"65-70\\n(n=67)\",\">=70\\n(n=14)\")))))\n",
    "\n",
    "table(ch_var_in_baseline_qcd.pheno.followup$Age_bin)\n",
    "    # order\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_bin <- factor(ch_var_in_baseline_qcd.pheno.followup$Age_bin,\n",
    "                                          levels = c(\"<50\\n(n=45)\", \"50-55\\n(n=166)\", \"55-60\\n(n=162)\",\n",
    "                                                     \"60-65\\n(n=122)\", \"65-70\\n(n=67)\", \">=70\\n(n=14)\"),\n",
    "                                          ordered = T)\n",
    "# # 10y\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Age_bin_10yr <- factor(ch_var_in_baseline_qcd.pheno.followup$Age_bin_10yr,\n",
    "#                                                         levels = c(\"<50\\n(n=45)\", \"50-60\\n(n=328)\", \"60-70\\n(n=189)\",\n",
    "#                                                                     \">=70\\n(n=14)\"),\n",
    "#                                                         ordered = T)\n",
    "ch_var_in_baseline_qcd.pheno.followup$Gene_cat <- ifelse(ch_var_in_baseline_qcd.pheno.followup$Gene %in% \n",
    "                             c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "                               \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "                               \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "                               \"ZRSR2\"), \n",
    "                           as.character(ch_var_in_baseline_qcd.pheno.followup$Gene), \"Other\")\n",
    "\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"DNMT3A\"] <- \"#a6cee3\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"TET2\"] <- \"#1f78b4\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"ASXL1\"] <- \"#b2df8a\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"JAK2\"] <- \"#33a02c\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"SF3B1\"] <- \"#cab2d6\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"PPM1D\"] <- \"#fdbf6f\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"U2AF1\"] <- \"#e31a1c\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"SRSF2\"] <-  \"#ff7f00\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"TP53\"] <- \"#6a3d9a\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"ZRSR2\"] <- \"#fb9a99\"\n",
    "# ch_var_in_baseline_qcd.pheno.followup$Gene_color[ch_var_in_baseline_qcd.pheno.followup$Gene_cat==\"Other\"] <- \"#b15928\"\n",
    "\n",
    "# ch_base <- ch_var_in_baseline_qcd.pheno.followup[,c(9,128)] %>% group_by(Age_bin) \n",
    "# Each 5y\n",
    "ch_base <- ch_var_in_baseline_qcd.pheno.followup[,c(129,128)] %>% group_by(Age_bin) \n",
    "#   # each 10y\n",
    "# ch_base <- ch_var_in_baseline_qcd.pheno.followup[,c(129,131)] %>% group_by(Age_bin_10yr) \n",
    "\n",
    "ch_base <- as.data.frame(prop.table(table(ch_base),margin = 2)*100)\n",
    "\n",
    "# ch_base$Gene_cat<- ifelse(ch_base$Gene %in% \n",
    "  #                            c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "   #                             \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "    #                            \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "     #                           \"ZRSR2\"), \n",
    "      #                      as.character(ch_base$Gene), \"Other\")\n",
    "\n",
    "## order: https://stackoverflow.com/questions/9227389/ordering-stacks-by-size-in-a-ggplot2-stacked-bar-graph\n",
    "ch_base$Gene_cat <- reorder(ch_base$Gene_cat, ch_base$Freq)\n",
    "\n",
    "\n",
    "## legend(\"topright\", legend = c(\"DNMT3A\",   \"TET2\",  \"JAK2\",  \"ASXL1\",    \"CBL\",   \"TP53\",  \"PPM1D\",  \"SF3B1\",   \"GNAS\",   \"GNB1\"), \n",
    "#  fill = c(\"#A6CEE3\", \"#B2DF8A\", \"#CAB2D6\", \"#33A02C\", \"#1F78B4\", \"#FB9A99\", \"#E31A1C\", \"#FDBF6F\", \"#6A3D9A\",\"#FF7F00\"), \n",
    "#cex=1,\n",
    "# border = c(\"#A6CEE3\", \"#B2DF8A\", \"#CAB2D6\", \"#33A02C\", \"#1F78B4\", \"#FB9A99\", \"#E31A1C\", \"#FDBF6F\", \"#6A3D9A\",\"#FF7F00\"))\n",
    "\n",
    "\n",
    "  # ch_base$Gene_cat <- factor(ch_base$Gene_cat, \n",
    "#                            levels = rev(levels(ch_base$Gene_cat)))\n",
    "\n",
    "## plot c\n",
    "c <- ch_base %>% filter(Freq>0) %>% ggplot(., \n",
    "                                     aes(fill=Gene_cat, \n",
    "                                         y=Freq, x=Age_bin)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion (%)\") + xlab(\"Age at baseline (years)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "# scale_color_brewer(type = \"qual\",\n",
    "#                      palette = \"Paired\",\n",
    "#                      direction = -1,\n",
    "#                      aesthetics = \"fill\" ) +\n",
    "# # scale_color_brewer(type = \"div\", \n",
    "# #                      palette = \"Spectral\",\n",
    "# #                      direction = -1, \n",
    "# #                      aesthetics = \"fill\" ) +\n",
    "theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"c\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) + \n",
    "scale_fill_manual(values = c(\"#fb9a99\",\"#ffff99\",\"#33a02c\",\n",
    "                               \"#cab2d6\",\"#ff7f00\",\"#6a3d9a\",\n",
    "                               \"#e31a1c\",\"#fdbf6f\",\"#b2df8a\",\n",
    "                               \"#1f78b4\",\"#a6cee3\") )\n",
    "                      \n",
    "# c <- c + scale_y_break(breaks=c(5, 25), \n",
    "  #                  scales=c(10, 0.00001),\n",
    "   #                # ticklabels = c(0,10,20,25,35,50,\n",
    "    #                #               seq(60,100,20)), \n",
    "     #               expand = F,space = 0.5) + \n",
    " # theme(plot.title = element_text(hjust = 0.0, vjust = 2))\n",
    "\n",
    "# c <- c + scale_y_continuous(breaks = c(0,5,25,40,60,80,100)) + \n",
    "  # theme(axis.title.y.right = element_blank(),\n",
    "    #    axis.text.y.right = element_blank(), \n",
    "     #   axis.line.y.right = element_blank(), \n",
    "      #  axis.ticks.y.right = element_blank())\n",
    "\n",
    "c\n",
    "###\n",
    "## follow-up\n",
    "ch_var_in_v05_qcd.pheno$Age_bin <- ifelse(ch_var_in_v05_qcd.pheno$Age<70,\"<70\\n(n=91)\",\n",
    "                                          ifelse(ch_var_in_v05_qcd.pheno$Age>=70 &  \n",
    "                                                   ch_var_in_v05_qcd.pheno$Age<75,\n",
    "                                                 \"70-75\\n(n=376)\", \n",
    "                                                 ifelse(ch_var_in_v05_qcd.pheno$Age>=75 &\n",
    "                                                          ch_var_in_v05_qcd.pheno$Age<80,\n",
    "                                                        \"75-80\\n(n=368)\", \n",
    "                                                        ifelse(ch_var_in_v05_qcd.pheno$Age>=80 & \n",
    "                                                                 ch_var_in_v05_qcd.pheno$Age<85,\n",
    "                                                               \"80-85\\n(n=316)\",\"85-90\\n(n=151)\"))))\n",
    "# ## each 10 years\n",
    "# ch_var_in_v05_qcd.pheno$Age_bin_10yr <- ifelse(ch_var_in_v05_qcd.pheno$Age<70,\"<70\\n(n=91)\",\n",
    "#                                           ifelse(ch_var_in_v05_qcd.pheno$Age>=70 &  \n",
    "#                                                    ch_var_in_v05_qcd.pheno$Age<80,\n",
    "#                                                  \"70-80\\n(n=744)\",\n",
    "#                                                  \"80-90\\n(n=467)\"))\n",
    "# table(ch_var_in_v05_qcd.pheno$Age_bin_10yr)\n",
    "# ch_var_in_v05_qcd.pheno$Age_bin_10yr <- factor(ch_var_in_v05_qcd.pheno$Age_bin_10yr, \n",
    "#                                           levels = c(\"<70\\n(n=91)\", \n",
    "#                                                      \"70-80\\n(n=744)\", \n",
    "#                                                      \"80-90\\n(n=467)\"), \n",
    "#                                           ordered = T)\n",
    "\n",
    "ch_var_in_v05_qcd.pheno$Gene_cat <- ifelse(ch_var_in_v05_qcd.pheno$Gene %in% \n",
    "                            c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "                              \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "                              \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "                              \"ZRSR2\"), \n",
    "                          as.character(ch_var_in_v05_qcd.pheno$Gene), \"Other\")\n",
    "\n",
    "\n",
    "table(ch_var_in_v05_qcd.pheno$Age_bin)\n",
    "    # order\n",
    "ch_var_in_v05_qcd.pheno$Age_bin <- factor(ch_var_in_v05_qcd.pheno$Age_bin, \n",
    "                                          levels = c(\"<70\\n(n=91)\", \n",
    "                                                     \"70-75\\n(n=376)\", \n",
    "                                                     \"75-80\\n(n=368)\",\n",
    "                                                      \"80-85\\n(n=316)\", \n",
    "                                                     \"85-90\\n(n=151)\"), \n",
    "                                          ordered = T)\n",
    "\n",
    "# ch_v05 <- ch_var_in_v05_qcd.pheno[,c(9,127)] %>% group_by(Age_bin)\n",
    "ch_v05 <- ch_var_in_v05_qcd.pheno[,c(128,127)] %>% group_by(Age_bin) \n",
    "# # 10yrs\n",
    "# ch_v05 <- ch_var_in_v05_qcd.pheno[,c(128,127)] %>% group_by(Age_bin_10yr) \n",
    "\n",
    "ch_v05 <- as.data.frame(prop.table(table(ch_v05),margin = 2)*100)\n",
    "\n",
    "# ch_v05$Gene_cat <- ifelse(ch_v05$Gene %in% \n",
    "  #                         c(\"DNMT3A\",   \"TET2\",  \"ASXL1\",  \n",
    "   #                          \"SF3B1\",  \"PPM1D\",\"TP53\",  \n",
    "    #                         \"SRSF2\", \"JAK2\", \"U2AF1\",\n",
    "     #                         \"ZRSR2\"), \n",
    "      #                   as.character(ch_v05$Gene), \"Other\")\n",
    "ch_v05$Gene_cat <- reorder(ch_v05$Gene_cat, ch_v05$Freq)\n",
    "\n",
    "## plot g\n",
    "g <- ch_v05 %>% filter(Freq>0) %>% ggplot(., \n",
    "                                     aes(fill=Gene_cat, \n",
    "                                         y=Freq, x=Age_bin)) + \n",
    "  geom_bar(position=\"stack\", stat=\"identity\") + \n",
    "  ylab(\"Proportion (%)\") + xlab(\"Age at follow-up (years)\") + \n",
    "#  scale_fill_viridis(discrete = T) +\n",
    "# scale_color_brewer(type = \"qual\", \n",
    "#                      palette = \"Paired\",\n",
    "#                      direction = -1, \n",
    "#                      aesthetics = \"fill\" ) +\n",
    "scale_fill_manual(values = c(\"#fb9a99\",\"#6a3d9a\",\"#ffff99\",\n",
    "                             \"#ff7f00\",\"#cab2d6\",\"#e31a1c\",\n",
    "                             \"#33a02c\",\"#fdbf6f\",\"#b2df8a\",\n",
    "                             \"#1f78b4\",\"#a6cee3\") ) +\n",
    "theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"g\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  \n",
    "\n",
    "# g <- g + scale_y_break(breaks=c(5, 25), \n",
    "  #                     scales=c(10, 0.00001),\n",
    "   #                    # ticklabels = c(0,10,20,25,35,50,\n",
    "    #                   #                seq(60,100,20)), \n",
    "     #                  expand = F,space = 0.5) + \n",
    " # theme(plot.title = element_text(hjust = 0.0, vjust = 2))\n",
    "\n",
    "# g <- g + scale_y_continuous(breaks = c(0,5,25,40,60,80,100)) + \n",
    " # theme(axis.title.y.right = element_blank(),\n",
    "  #      axis.text.y.right = element_blank(), \n",
    "   #     axis.line.y.right = element_blank(), \n",
    "    #    axis.ticks.y.right = element_blank())\n",
    "g\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a0406f55",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cbd0d467",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "624f1e2d",
   "metadata": {},
   "outputs": [],
   "source": [
    "##### Fig 1d,h\n",
    "\n",
    "## remove every thing after \"\\n\"\n",
    "## gsub(\"stringto_find..*\", \"relaplcewith\", my_input_string)\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_Bin <- gsub(pattern = \"\\n..*\",\n",
    "                                                      replacement = \"\", \n",
    "                                                      x = ch_var_in_baseline_qcd.pheno.followup$Age_bin)\n",
    "\n",
    "ch_var_in_v05_qcd.pheno$Age_Bin <- gsub(pattern = \"\\n..*\",\n",
    "                                                      replacement = \"\", \n",
    "                                                      x = ch_var_in_v05_qcd.pheno$Age_bin)\n",
    "\n",
    "table(ch_var_in_baseline_qcd.pheno.followup$Age_Bin)\n",
    "table(ch_var_in_v05_qcd.pheno$Age_Bin)\n",
    "table(ch_var_in_baseline_qcd.pheno.followup$Age_bin)\n",
    "table(ch_var_in_v05_qcd.pheno$Age_bin)\n",
    "\n",
    "ch_var_in_baseline_qcd.pheno.followup$Age_bin <- factor(ch_var_in_baseline_qcd.pheno.followup$Age_bin,\n",
    "                                          levels = c(\"<50\\n(n=45)\", \n",
    "                                                     \"50-55\\n(n=166)\", \n",
    "                                                     \"55-60\\n(n=162)\",\n",
    "                                                      \"60-65\\n(n=122)\", \n",
    "                                                     \"65-70\\n(n=67)\", \n",
    "                                                     \">=70\\n(n=14)\"), \n",
    "                                          ordered = T)\n",
    "\n",
    "ch_var_in_v05_qcd.pheno$Age_bin <- factor(ch_var_in_v05_qcd.pheno$Age_bin,\n",
    "                                          levels = c(\"<70\\n(n=91)\", \n",
    "                                                     \"70-75\\n(n=376)\", \n",
    "                                                     \"75-80\\n(n=368)\",\n",
    "                                                      \"80-85\\n(n=316)\", \n",
    "                                                     \"85-90\\n(n=151)\"), \n",
    "                                          ordered = T)\n",
    "\n",
    "## VAF distribution by Age bin\n",
    "ch_var_in_baseline_qcd.pheno.followup[,c(128,11)] %>% \n",
    "  group_by(Age_bin) %>% \n",
    "  summarise(VAF=median(VAF))\n",
    "\n",
    "ch_var_in_v05_qcd.pheno[,c(127,11)] %>% \n",
    "  group_by(Age_bin) %>% \n",
    "  summarise(VAF=median(VAF))\n",
    "\n",
    "## d\n",
    "d <- ggplot(data=ch_var_in_baseline_qcd.pheno.followup, \n",
    "            aes(x=Age_bin, y=VAF)) + \n",
    "  xlab(\"Age at baseline (years)\") + ylab(\"Variant Allele Fraction\") + \n",
    "  geom_boxplot() + \n",
    "  scale_y_log10(breaks = c(0.02,0.1,.2,.3, 0.5,1)) +\n",
    "  theme(# axis.text.x = element_text(angle = 40, vjust = 1,  hjust=1),\n",
    "        legend.title = element_blank(), legend.position = \"\",\n",
    "        plot.title = element_text(size = 20, face = \"bold\"),\n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) + \n",
    "# scale_color_brewer(type = \"seq\", \n",
    "#                      palette = \"Blues\",\n",
    "#                      direction = 1, \n",
    "#                      aesthetics = \"fill\" ) +\n",
    "  ggtitle(\"d\") \n",
    "\n",
    "d\n",
    "## h\n",
    "h <- ggplot(data=ch_var_in_v05_qcd.pheno, \n",
    "            aes(x=Age_bin, y=VAF)) + \n",
    "  xlab(\"Age at follow-up (years)\") + ylab(\"Variant Allele Fraction\") + \n",
    "  geom_boxplot() + \n",
    "  scale_y_log10(breaks = c(0.02,0.1,.2,.3, 0.5,1)) +\n",
    "  theme( # axis.text.x = element_text(angle = 40, vjust = 1,  hjust=1),\n",
    "        legend.title = element_blank(), legend.position = \"\",\n",
    "        plot.title = element_text(size = 20, face = \"bold\"),\n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\")) + \n",
    "# scale_color_brewer(type = \"seq\", \n",
    "#                      palette = \"Blues\",\n",
    "#                      direction = 1, \n",
    "#                      aesthetics = \"fill\" ) +\n",
    "  ggtitle(\"h\") \n",
    "\n",
    "d\n",
    "h\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "57155c9a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6f8a6a90",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7e282831",
   "metadata": {},
   "outputs": [],
   "source": [
    "## singlet plots\n",
    "# setwd(\"~/Documents/Project/Baylor_ARIC_Exomes/Paper1_ARIC/Display/\")\n",
    "pdf(\"Fig.1a.pdf\", width=8, height=8)\n",
    "a\n",
    "dev.off()\n",
    "\n",
    "pdf(\"Fig.1b.pdf\", width=8, height=8)\n",
    "b\n",
    "dev.off()\n",
    "\n",
    "# png(\"fig1c.png\", width=6, height=3, units= \"in\", res=300, pointsize = 4)\n",
    "# c\n",
    "# dev.off()\n",
    "\n",
    "pdf(\"Fig.1c.pdf\", width=8, height=8)\n",
    "c\n",
    "dev.off()\n",
    "\n",
    "pdf(\"Fig.1d.pdf\", width=8, height=8)\n",
    "d\n",
    "dev.off()\n",
    "\n",
    "pdf(\"Fig.1e.pdf\", width=8, height=8)\n",
    "e\n",
    "dev.off()\n",
    "\n",
    "pdf(\"Fig.1f.pdf\", width=8, height=8)\n",
    "f\n",
    "dev.off()\n",
    "\n",
    "pdf(\"Fig.1g.pdf\", width=8, height=8)\n",
    "g\n",
    "dev.off()\n",
    "\n",
    "pdf(\"Fig.1h.pdf\", width=8, height=8)\n",
    "h\n",
    "dev.off()\n",
    "\n",
    "# png(\"fig1d.png\", width=6, height=3, units= \"in\", res=300, pointsize = 4)\n",
    "# d\n",
    "# dev.off()\n",
    "# \n",
    "# png(\"fig1e.png\", width=6, height=3, units= \"in\", res=300, pointsize = 4)\n",
    "# e\n",
    "# dev.off()\n",
    "# \n",
    "# png(\"fig1f.png\", width=6, height=3, units= \"in\", res=300, pointsize = 4)\n",
    "# f\n",
    "# dev.off()\n",
    "# \n",
    "# png(\"fig1g.png\", width=6, height=3, units= \"in\", res=300, pointsize = 4)\n",
    "# g\n",
    "# dev.off()\n",
    "# \n",
    "# png(\"fig1h.png\", width=6, height=3, units= \"in\", res=300, pointsize = 4)\n",
    "# h\n",
    "# dev.off()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3fc46a7e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ecbe1319",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ea103191",
   "metadata": {},
   "outputs": [],
   "source": [
    "## multi-panel plot\n",
    "# png(\"fig1.CH_base_n_follow-up.noHemeCa_15Jun23.png\",\n",
    "#    width=26, height=12, units= \"in\", res=300, pointsize = 4)\n",
    "# library(patchwork)\n",
    "# # https://patchwork.data-imaginist.com/articles/guides/layout.html\n",
    "# # (a | b | c | d) / (e | f | g | h)\n",
    "# # ( (wrap_elements(a) / wrap_elements(e)) | (wrap_elements(b) / wrap_elements(f)) | (wrap_elements(c) / wrap_elements(g)) | (wrap_elements(d) / wrap_elements(h)) )\n",
    "# \n",
    "# ( (a / e) | (b / f)  |   (c / g) | (d / h) )\n",
    "# \n",
    "# # ggarrange(a, b, c, d, \n",
    "# #           e, f, g, h,\n",
    "# #           ncol = 4, \n",
    "# #           nrow = 2 )\n",
    "# \n",
    "# dev.off()\n",
    "png(\"fig1.CH_base_n_follow-up.noHemeCa_5Jul23.patchwork.v1.png\",\n",
    "    width=26, height=12, units= \"in\", res=300, pointsize = 4)\n",
    "( (a / e) | (b / f)  |   (c / g) | (d / h)) + \n",
    "  plot_layout(widths = c(1.2,1.5,1.5,1)\n",
    "# # https://patchwork.data-imaginist.com/articles/guides/layout.html\n",
    " #             , \n",
    "  #            heights = c(1,1.2,1.2,1)\n",
    ")\n",
    "dev.off()\n",
    "####\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8e6d2aa4",
   "metadata": {},
   "outputs": [],
   "source": [
    "## VAF distribution by Age bin\n",
    "\n",
    "## d\n",
    "# d <- ggplot(data=ch_var_in_baseline_qcd.pheno.followup, aes(x=Age_bin, y=VAF, fill=Age_bin)) + \n",
    "  # xlab(\"Age (baseline)\") + ylab(\"Variant Allele Fraction\") + \n",
    "  # geom_boxplot() + scale_y_log10(breaks = c(0.02,0.1,.2,.3, 0.5,1)) +\n",
    "  # theme(# axis.text.x = element_text(angle = 40, vjust = 1,  hjust=1),\n",
    "    #    legend.title = element_blank(), legend.position = \"\",\n",
    "     #   plot.title = element_text(size = 20, face = \"bold\"),\n",
    "      #  axis.text=element_text(size=12, face=\"bold\"),\n",
    "       # axis.title=element_text(size=14,face=\"bold\")) + \n",
    "# scale_color_brewer(type = \"seq\", \n",
    "  #                   palette = \"Blues\",\n",
    "   #                  direction = 1, \n",
    "    #                 aesthetics = \"fill\" ) +\n",
    "#  ggtitle(\"d\") \n",
    "\n",
    "\n",
    "## h\n",
    "# h <- ggplot(data=ch_var_in_v05_qcd.pheno, aes(x=Age_bin, y=VAF, fill=Age_bin)) + \n",
    " # xlab(\"Age (Follow-up)\") + ylab(\"Variant Allele Fraction\") + \n",
    " # geom_boxplot() + scale_y_log10(breaks = c(0.02,0.1,.2,.3, 0.5,1)) +\n",
    " # theme( # axis.text.x = element_text(angle = 40, vjust = 1,  hjust=1),\n",
    "  #      legend.title = element_blank(), legend.position = \"\",\n",
    "   #     plot.title = element_text(size = 20, face = \"bold\"),\n",
    "    #    axis.text=element_text(size=12, face=\"bold\"),\n",
    "     #   axis.title=element_text(size=14,face=\"bold\")) + \n",
    "# scale_color_brewer(type = \"seq\", \n",
    "  #                   palette = \"Blues\",\n",
    "   #                  direction = 1, \n",
    "    #                 aesthetics = \"fill\" ) +\n",
    "#  ggtitle(\"h\") \n",
    "\n",
    "# d\n",
    "# h"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "01519c0a",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3d32e688",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "a31c2cc6",
   "metadata": {},
   "outputs": [],
   "source": [
    "##########################"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e8d5edc7",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig a ####\n",
    "a <- ggplot(data=aric_baseline_full_noPrevHeme, aes(x=age_base, y=CH_baseline, group=Follow_up) ) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "  # geom_smooth(aes(colour=Follow_up), \n",
    "    #          method =\"glm\", method.args= list(family = \"binomial\")) +\n",
    "  geom_smooth(aes(colour=Follow_up), \n",
    "              method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  +\n",
    "scale_y_continuous(breaks = c(seq(from = 0, to = 1, by = 0.1)) ) \n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b99e5a70",
   "metadata": {},
   "outputs": [],
   "source": [
    "#### Fig a ####\n",
    "ggplot(data=aric_baseline_full_noPrevHeme, aes(x=age_base, y=CH_baseline, group=Follow_up) ) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "   geom_smooth(aes(colour=Follow_up), \n",
    "              method =\"glm\", method.args= list(family = \"binomial\")) +\n",
    "  #geom_smooth(aes(colour=Follow_up), \n",
    "   #           method =\"gam\") +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  +\n",
    "scale_y_continuous(breaks = c(seq(from = 0, to = 0.5, by = 0.05)) ) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "b954f2ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "a = ggplot(data=aric_baseline_n_v05_noPrevHeme, aes(x=age_base), group=1) + \n",
    "  xlab(\"Age (baseline)\") + ylab(\"Prevalence\") +\n",
    "    geom_smooth(aes(y=CH_baseline, colour=\"VAF>=2%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "    geom_smooth(aes(y=CHvaf10_baseline, colour=\"VAF>=10%\"), \n",
    "              method =\"glm\", method.args = list(family = \"binomial\")) +\n",
    "  theme(legend.title = element_blank(), \n",
    "        legend.position = \"\") +\n",
    "  ggtitle(\"a\") + \n",
    "  theme(legend.position = \"right\", \n",
    "        plot.title = element_text(size = 20, face = \"bold\"), \n",
    "        axis.text=element_text(size=12, face=\"bold\"),\n",
    "        axis.title=element_text(size=14,face=\"bold\"))  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "60ac5471",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d34227ab",
   "metadata": {},
   "outputs": [],
   "source": [
    "names(aric_baseline_n_v05_noPrevHeme)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "aa165ddd",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(maftools)\n",
    "maf.aric_v05 <- annovarToMaf(annovar = \"maftools.ch_var_in_v05_qcd_pheno.23Mar2023.txt\", \n",
    "                             refBuild = 'hg38', ens2hugo = TRUE, \n",
    "                             MAFobj = TRUE, tsbCol=\"GWAS_ID\")\n",
    "\n",
    "\n",
    "plotmafSummary(maf = maf.aric_v05, rmOutlier = TRUE, addStat = 'median', dashboard = TRUE, titvRaw = FALSE)\n",
    "\n",
    "oncoplot(maf = maf.aric_v05, top = 15)\n",
    "\n",
    "plotVaf(maf = maf.aric_v05, vafCol = 't_vaf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c093036d",
   "metadata": {},
   "outputs": [],
   "source": [
    "## Somatic Interactions\n",
    "# Mutually exclusive or co-occurring set of genes can be detected using somaticInteractions function, which performs pair-wise Fisher’s Exact test to detect such significant pair of genes.\n",
    "# exclusive/co-occurance event analysis on top 10 mutated genes. \n",
    "somaticInteractions(maf = maf.aric_v05, \n",
    "                    top = 10, \n",
    "                    pvalue = c(0.01, 0.05))"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
