{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "ef6929c0",
   "metadata": {},
   "source": [
    "### Prepare SNPs for PRS"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "800b5992",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "########## SNP weight for GRS\n",
    "library(readxl)\n",
    "library(data.table)\n",
    "library(stringr)\n",
    "  # SNP list in the plink file\n",
    "# aric_eur_snp <- fread(\"/Volumes/mesbah/dataset/dbgaps/aric_phs000090_GENEVA/geneva_qsub/aric_pgen/qcd_hapmap3/ARIC_EurAm_chr1_22.hapmap3.bim\", \n",
    "#                       header = F)\n",
    "# aric_afr_snp <- fread(\"/Volumes/mesbah/dataset/dbgaps/aric_phs000090_GENEVA/geneva_qsub/aric_pgen/qcd_hapmap3/ARIC_AfrAm_chr1_22.hapmap3.bim\", \n",
    "#                       header = F)\n",
    "# # cat ARIC_AfrAm_chr1.1kg_imp.pvar > ARIC.chr1_22.AA.tsv; for chr in {2..22}; do awk 'NR>1' ARIC_AfrAm_chr${chr}.1kg_imp.pvar >> ARIC.chr1_22.AA.tsv; done &\n",
    "# # cat ARIC_EurAm_chr1.1kg_imp.pvar > ARIC.chr1_22.EA.tsv; for chr in {2..22}; do awk 'NR>1' ARIC_EurAm_chr${chr}.1kg_imp.pvar >> ARIC.chr1_22.EA.tsv; done &\n",
    "aric_eur_snp <- fread(\"/Volumes/mesbah/dataset/dbgaps/aric_phs000090_GENEVA/geneva_qsub/aric_pgen/ARIC.chr1_22.EA.tsv\", \n",
    "                      header = T)\n",
    "names(aric_eur_snp) <- c(\"CHROM\",\t\"POS\",\t\"ID\",\t\"REF\",\t\"ALT\")\n",
    "\n",
    "aric_eur_snp$varID <- paste(aric_eur_snp$CHROM, \n",
    "                            aric_eur_snp$POS, \n",
    "                            aric_eur_snp$REF, \n",
    "                            aric_eur_snp$ALT, \n",
    "                            sep=\":\")\n",
    "\n",
    "aric_afr_snp <- fread(\"/Volumes/mesbah/dataset/dbgaps/aric_phs000090_GENEVA/geneva_qsub/aric_pgen/ARIC.chr1_22.AA.tsv\", \n",
    "                      header = T)\n",
    "\n",
    "names(aric_afr_snp) <- c(\"CHROM\",\t\"POS\",\t\"ID\",\t\"REF\",\t\"ALT\")\n",
    "\n",
    "aric_afr_snp$varID <- paste(aric_afr_snp$CHROM, \n",
    "                            aric_afr_snp$POS, \n",
    "                            aric_afr_snp$REF, \n",
    "                            aric_afr_snp$ALT, \n",
    "                            sep=\":\")\n",
    "\n",
    "table(ch_hg19_cojo$varID_hg19 %in% aric_afr_snp$varID )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "ba278a59",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "70904cd6",
   "metadata": {},
   "outputs": [],
   "source": [
    "## GWAS SNPs from REGENERON GWAS\n",
    "ch_rg22_cojo <- read_excel(\"~/Documents/Project/Baylor_ARIC_Exomes/Paper1_ARIC/GWAS/41586_2022_5448_MOESM5_ESM.xlsx\", \n",
    "                      sheet = 4)\n",
    "\n",
    "ch_rg22_cojo$REF_Allele <- str_split_fixed(string = ch_rg22_cojo$SNP, \n",
    "                                    pattern = \"[:]\", n = 4)[,3]\n",
    "ch_rg22_cojo$ALT_Allele <- str_split_fixed(string = ch_rg22_cojo$SNP, \n",
    "                                           pattern = \"[:]\", n = 4)[,4]\n",
    "  # only SNPs with P<5e-8 and AF>5%\n",
    "ch_rg22_cojo <- subset(ch_rg22_cojo, \n",
    "                       ch_rg22_cojo$P_COJO<5e-8 & \n",
    "                         ch_rg22_cojo$AAF>=0.05 &\n",
    "                         ch_rg22_cojo$REF_Allele %in% \n",
    "                         c(\"A\", \"C\", \"G\", \"T\") & \n",
    "                         ch_rg22_cojo$ALT_Allele %in% \n",
    "                         c(\"A\", \"C\", \"G\", \"T\"))\n",
    "\n",
    "\n",
    "  # DNMT3A\n",
    "dnmt3a_rg22_cojo <- read_excel(\"~/Documents/Project/Baylor_ARIC_Exomes/Paper1_ARIC/GWAS/41586_2022_5448_MOESM5_ESM.xlsx\", \n",
    "                          sheet = 6)\n",
    "\n",
    "dnmt3a_rg22_cojo$REF_Allele <- str_split_fixed(string = dnmt3a_rg22_cojo$SNP, \n",
    "                                           pattern = \"[:]\", n = 4)[,3]\n",
    "dnmt3a_rg22_cojo$ALT_Allele <- str_split_fixed(string = dnmt3a_rg22_cojo$SNP, \n",
    "                                           pattern = \"[:]\", n = 4)[,4]\n",
    "# only SNPs with P<5e-8 and AF>5%\n",
    "dnmt3a_rg22_cojo <- subset(dnmt3a_rg22_cojo, \n",
    "                           dnmt3a_rg22_cojo$P_COJO<5e-8 & \n",
    "                             dnmt3a_rg22_cojo$AAF>=0.05 &\n",
    "                             dnmt3a_rg22_cojo$REF_Allele %in% \n",
    "                         c(\"A\", \"C\", \"G\", \"T\") & \n",
    "                           dnmt3a_rg22_cojo$ALT_Allele %in% \n",
    "                         c(\"A\", \"C\", \"G\", \"T\"))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "35f2b5fd",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "## lift0ver hg38 to hg19\n",
    "library(liftOver)\n",
    "library(rtracklayer)\n",
    "library(TxDb.Hsapiens.UCSC.hg38.knownGene)\n",
    "library(GenomicRanges)\n",
    "# Load Chain file\n",
    "hg38_to_19_chain = import.chain( \"~/Documents/Project/Baylor_ARIC_Exomes/Paper1_ARIC/GWAS/hg38ToHg19.over.chain\")\n",
    "\n",
    "# CH\n",
    "ch_rg22_cojo$CHROM <- paste0(\"chr\",ch_rg22_cojo$CHR)\n",
    "ch_rg22_cojo_df <- ch_rg22_cojo[,c(19, 3, 3,1)]\n",
    "names(ch_rg22_cojo_df) <- c(\"chrom\", \"start\", \"end\",\"varID\")\n",
    "gr.ch.hg38 <- as(ch_rg22_cojo_df, \"GenomicRanges\")\n",
    "\n",
    "ch_hg38tohg19 <- liftOver(gr.ch.hg38, hg38_to_19_chain)\n",
    "ch_hg38tohg19 <- as.data.frame(ch_hg38tohg19)\n",
    "\n",
    "ch_hg38tohg19$chr_hg19 <- gsub(pattern = \"chr\",replacement = \"\", \n",
    "                                   x = as.character(ch_hg38tohg19$seqnames))\n",
    "\n",
    "ch_hg38tohg19$POS_hg19 <- ch_hg38tohg19$start\n",
    "\n",
    "ch_hg19_cojo <- merge(ch_hg38tohg19[,c(8,9,10)], \n",
    "                       ch_rg22_cojo, \n",
    "                           by.x=\"varID\",\n",
    "                       by.y=\"SNP\")\n",
    "\n",
    "ch_hg19_cojo$varID_hg19 <- paste(ch_hg19_cojo$chr_hg19, \n",
    "                                 ch_hg19_cojo$POS_hg19, \n",
    "                                 ch_hg19_cojo$REF_Allele, \n",
    "                                 ch_hg19_cojo$ALT_Allele, \n",
    "                                 sep=\":\")\n",
    "\n",
    "table(ch_hg19_cojo$varID_hg19 %in% aric_afr_snp$varID )\n",
    "# ch_hg19_cojo$varID_hg19[!(ch_hg19_cojo$varID_hg19 %in% aric_afr_snp$varID)]\n",
    "table(ch_hg19_cojo$varID_hg19 %in% aric_eur_snp$varID )\n",
    "# ch_hg19_cojo$varID_hg19[!(ch_hg19_cojo$varID_hg19 %in% aric_eur_snp$varID)]\n",
    "# 21 SNPs\n",
    "aa_aric_snvs_ch_hg19_cojo <- merge(ch_hg19_cojo[,c(1,2,3,19,20,\n",
    "                                                           15,22,17,10)],\n",
    "                                       aric_afr_snp, \n",
    "                                       by.x=\"varID_hg19\",\n",
    "                                       by.y=\"varID\")\n",
    "\n",
    "ea_aric_snvs_ch_hg19_cojo <- merge(ch_hg19_cojo[,c(1,2,3,19,20,\n",
    "                                                           15,22,17,10)],\n",
    "                                       aric_eur_snp, \n",
    "                                       by.x=\"varID_hg19\",\n",
    "                                       by.y=\"varID\")\n",
    "write.csv(ea_aric_snvs_ch_hg19_cojo, \n",
    "          \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/ea_aric_21snvs_ch_hg19_cojo.csv\", \n",
    "          row.names = F)\n",
    "write.csv(aa_aric_snvs_ch_hg19_cojo, \n",
    "          \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/aa_aric_21snvs_ch_hg19_cojo.csv\", \n",
    "          row.names = F)\n",
    "## DNMT3A\n",
    "dnmt3a_rg22_cojo$CHROM <- paste0(\"chr\",dnmt3a_rg22_cojo$CHR)\n",
    "dnmt3a_rg22_cojo_df <- dnmt3a_rg22_cojo[,c(19, 3, 3,1)]\n",
    "names(dnmt3a_rg22_cojo_df) <- c(\"chrom\", \"start\", \"end\",\"varID\")\n",
    "gr.dnmt3a.hg38 <- as(dnmt3a_rg22_cojo_df, \"GenomicRanges\")\n",
    "\n",
    "dnmt3a_hg38tohg19 <- liftOver(gr.dnmt3a.hg38, \n",
    "                              hg38_to_19_chain)\n",
    "dnmt3a_hg38tohg19 <- as.data.frame(dnmt3a_hg38tohg19)\n",
    "\n",
    "dnmt3a_hg38tohg19$chr_hg19 <- gsub(pattern = \"chr\",replacement = \"\", \n",
    "                               x = as.character(dnmt3a_hg38tohg19$seqnames))\n",
    "\n",
    "dnmt3a_hg38tohg19$POS_hg19 <- dnmt3a_hg38tohg19$start\n",
    "\n",
    "dnmt3a_hg19_cojo <- merge(dnmt3a_hg38tohg19[,c(8,9,10)], \n",
    "                          dnmt3a_rg22_cojo, \n",
    "                          by.x=\"varID\",\n",
    "                          by.y=\"SNP\")\n",
    "\n",
    "dnmt3a_hg19_cojo$varID_hg19 <- paste(dnmt3a_hg19_cojo$chr_hg19, \n",
    "                                     dnmt3a_hg19_cojo$POS_hg19, \n",
    "                                     dnmt3a_hg19_cojo$REF_Allele, \n",
    "                                     dnmt3a_hg19_cojo$ALT_Allele, \n",
    "                                 sep=\":\")\n",
    "\n",
    "\n",
    "\n",
    "# 22 SNPs\n",
    "table(dnmt3a_hg19_cojo$varID_hg19 %in% aric_afr_snp$varID )\n",
    "\n",
    "table(dnmt3a_hg19_cojo$varID_hg19 %in% aric_eur_snp$varID )\n",
    "\n",
    "aa_aric_snvs_dnmt3a_hg19_cojo <- merge(dnmt3a_hg19_cojo[,c(1,2,3,19,20,\n",
    "                                                           15,22,17,10)],\n",
    "                                     aric_afr_snp, \n",
    "                                     by.x=\"varID_hg19\",\n",
    "                                     by.y=\"varID\")\n",
    "\n",
    "ea_aric_snvs_dnmt3a_hg19_cojo <- merge(dnmt3a_hg19_cojo[,c(1,2,3,19,20,\n",
    "                                                         15,22,17,10)],\n",
    "                                     aric_eur_snp, \n",
    "                                     by.x=\"varID_hg19\",\n",
    "                                     by.y=\"varID\")\n",
    "write.csv(ea_aric_snvs_dnmt3a_hg19_cojo, \n",
    "          \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/ea_aric_22snvs_dnmt3a_hg19_cojo.csv\", \n",
    "          row.names = F)\n",
    "write.csv(aa_aric_snvs_dnmt3a_hg19_cojo, \n",
    "          \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/aa_aric_22snvs_dnmt3a_hg19_cojo.csv\", \n",
    "          row.names = F)\n",
    "## TET2\n",
    "## TET2\n",
    "tet2_rg22_indx <- read_excel(\"~/Documents/Project/Baylor_ARIC_Exomes/Paper1_ARIC/GWAS/41586_2022_5448_MOESM5_ESM.xlsx\", \n",
    "                             sheet = 15)\n",
    "\n",
    "tet2_rg22_indx$REF_Allele <- str_split_fixed(string = tet2_rg22_indx$name, \n",
    "                                             pattern = \"[:]\", n = 4)[,3]\n",
    "tet2_rg22_indx$ALT_Allele <- str_split_fixed(string = tet2_rg22_indx$name, \n",
    "                                             pattern = \"[:]\", n = 4)[,4]\n",
    "\n",
    "tet2_rg22_indx$Beta <- log(tet2_rg22_indx$effect_UKB)\n",
    "\n",
    "# only SNPs with P<5e-8 and AF>5%\n",
    "tet2_rg22_indx <- subset(tet2_rg22_indx, \n",
    "                         tet2_rg22_indx$pval_UKB<5e-8 & \n",
    "                           tet2_rg22_indx$aaf_UKB>=0.05 &\n",
    "                           tet2_rg22_indx$REF_Allele %in% \n",
    "                           c(\"A\", \"C\", \"G\", \"T\") & \n",
    "                           tet2_rg22_indx$ALT_Allele %in% \n",
    "                           c(\"A\", \"C\", \"G\", \"T\"))\n",
    "\n",
    "tet2_rg22_indx$CHROM <- paste0(\"chr\",tet2_rg22_indx$chr)\n",
    "\n",
    "tet2_rg22_cojo_df <- tet2_rg22_indx[,c(45, 3, 3,1)]\n",
    "names(tet2_rg22_cojo_df) <- c(\"chrom\", \"start\", \"end\",\"varID\")\n",
    "gr.tet2.hg38 <- as(tet2_rg22_cojo_df, \"GenomicRanges\")\n",
    "\n",
    "tet2_hg38tohg19 <- liftOver(gr.tet2.hg38, \n",
    "                              hg38_to_19_chain)\n",
    "\n",
    "tet2_hg38tohg19 <- as.data.frame(tet2_hg38tohg19)\n",
    "\n",
    "tet2_hg38tohg19$chr_hg19 <- gsub(pattern = \"chr\",replacement = \"\", \n",
    "                                   x = as.character(tet2_hg38tohg19$seqnames))\n",
    "\n",
    "tet2_hg38tohg19$POS_hg19 <- tet2_hg38tohg19$start\n",
    "\n",
    "tet2_hg19_cojo <- merge(tet2_hg38tohg19[,c(8,9,10)], \n",
    "                          tet2_rg22_indx, \n",
    "                          by.x=\"varID\",\n",
    "                          by.y=\"name\")\n",
    "\n",
    "tet2_hg19_cojo$varID_hg19 <- paste(tet2_hg19_cojo$chr_hg19, \n",
    "                                   tet2_hg19_cojo$POS_hg19, \n",
    "                                   tet2_hg19_cojo$REF_Allele, \n",
    "                                   tet2_hg19_cojo$ALT_Allele, \n",
    "                                     sep=\":\")\n",
    "# 6 SNPs\n",
    "table(tet2_hg19_cojo$varID_hg19 %in% aric_afr_snp$varID )\n",
    "\n",
    "table(tet2_hg19_cojo$varID_hg19 %in% aric_eur_snp$varID )\n",
    "\n",
    "aa_aric_snvs_tet2_hg19_cojo <- merge(tet2_hg19_cojo[,c(1,2,3,44,45,46,48,13,14)],\n",
    "                                     aric_afr_snp, \n",
    "                                     by.x=\"varID_hg19\",\n",
    "                                     by.y=\"varID\")\n",
    "\n",
    "ea_aric_snvs_tet2_hg19_cojo <- merge(tet2_hg19_cojo[,c(1,2,3,44,45,46,48,13,14)],\n",
    "                                     aric_eur_snp, \n",
    "                                     by.x=\"varID_hg19\",\n",
    "                                     by.y=\"varID\" \n",
    "                                     )\n",
    "write.csv(ea_aric_snvs_tet2_hg19_cojo, \n",
    "          \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/ea_aric_6snvs_tet2_hg19_cojo.csv\", \n",
    "          row.names = F)\n",
    "write.csv(aa_aric_snvs_tet2_hg19_cojo, \n",
    "          \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/aa_aric_6snvs_tet2_hg19_cojo.csv\", \n",
    "          row.names = F)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "71d354d2",
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "### modified csv\n",
    "aa_aric_rg22ch_beta <- fread(\"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/aa_aric_21snvs_ch_hg19_cojo.csv\", header = T)\n",
    "ea_aric_rg22ch_beta <- fread(\"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/ea_aric_21snvs_ch_hg19_cojo.csv\", header = T)\n",
    "fwrite(ea_aric_rg22ch_beta, \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/ea_aric_rg22ch_beta.21SNV.txt\", \n",
    "       row.names = F, \n",
    "       col.names = T, \n",
    "       quote = F, \n",
    "       sep=\" \")\n",
    "fwrite(aa_aric_rg22ch_beta, \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/aa_aric_rg22ch_beta.21SNV.txt\", \n",
    "       row.names = F, \n",
    "       col.names = T, \n",
    "       quote = F, \n",
    "       sep=\" \")\n",
    "\n",
    "aa_aric_rg22dnmt3a_beta <- fread(\"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/aa_aric_22snvs_dnmt3a_hg19_cojo.csv\", header = T)\n",
    "ea_aric_rg22dnmt3a_beta <- fread(\"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/ea_aric_22snvs_dnmt3a_hg19_cojo.csv\", header = T)\n",
    "fwrite(ea_aric_rg22dnmt3a_beta, \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/ea_aric_rg22dnmt3a_beta.22SNV.txt\", \n",
    "       row.names = F, \n",
    "       col.names = T, \n",
    "       quote = F, \n",
    "       sep=\" \")\n",
    "fwrite(aa_aric_rg22dnmt3a_beta, \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/aa_aric_rg22dnmt3a_beta.22SNV.txt\", \n",
    "       row.names = F, \n",
    "       col.names = T, \n",
    "       quote = F, \n",
    "       sep=\" \")\n",
    "aa_aric_rg22tet2_beta <- fread(\"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/aa_aric_6snvs_tet2_hg19_cojo.csv\", header = T)\n",
    "ea_aric_rg22tet2_beta <- fread(\"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/ea_aric_6snvs_tet2_hg19_cojo.csv\", header = T)\n",
    "fwrite(ea_aric_rg22tet2_beta, \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/ea_aric_rg22tet2_beta.6SNV.txt\", \n",
    "       row.names = F, \n",
    "       col.names = T, \n",
    "       quote = F, \n",
    "       sep=\" \")\n",
    "\n",
    "fwrite(aa_aric_rg22tet2_beta, \"/Volumes/medpop_esp2/mesbah/projects/ch_progression/aric/gwas/PRS/aa_aric_rg22tet2_beta.6SNV.txt\", \n",
    "       row.names = F, \n",
    "       col.names = T, \n",
    "       quote = F, \n",
    "       sep=\" \")\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.1.0"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
